<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.262">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Evgenii O. Tretiakov">

<title>Dynamics of the hypothalamic arcuate nucleus response to hfd and signalling between astrocytic groups and Agrp or POMC neurons</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="02-velocity-signalling_hfd_files/libs/clipboard/clipboard.min.js"></script>
<script src="02-velocity-signalling_hfd_files/libs/quarto-html/quarto.js"></script>
<script src="02-velocity-signalling_hfd_files/libs/quarto-html/popper.min.js"></script>
<script src="02-velocity-signalling_hfd_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="02-velocity-signalling_hfd_files/libs/quarto-html/anchor.min.js"></script>
<link href="02-velocity-signalling_hfd_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="02-velocity-signalling_hfd_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="02-velocity-signalling_hfd_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="02-velocity-signalling_hfd_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="02-velocity-signalling_hfd_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script>
<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>
<script src="https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js" crossorigin="anonymous"></script>


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#genes" id="toc-genes" class="nav-link active" data-scroll-target="#genes">genes</a></li>
  <li><a href="#parameters" id="toc-parameters" class="nav-link" data-scroll-target="#parameters">parameters</a></li>
  <li><a href="#load-data" id="toc-load-data" class="nav-link" data-scroll-target="#load-data">Load data</a></li>
  <li><a href="#signalling-analysis" id="toc-signalling-analysis" class="nav-link" data-scroll-target="#signalling-analysis">Signalling analysis</a></li>
  <li><a href="#preprocess-data-for-velocity" id="toc-preprocess-data-for-velocity" class="nav-link" data-scroll-target="#preprocess-data-for-velocity">Preprocess data for velocity</a></li>
  <li><a href="#rna-velocity" id="toc-rna-velocity" class="nav-link" data-scroll-target="#rna-velocity">RNA velocity</a></li>
  <li><a href="#cellrank-analysis" id="toc-cellrank-analysis" class="nav-link" data-scroll-target="#cellrank-analysis">CellRank analysis</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Dynamics of the hypothalamic arcuate nucleus response to hfd and signalling between astrocytic groups and Agrp or POMC neurons</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Evgenii O. Tretiakov </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<div class="cell" data-execution_count="1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> random</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> warnings</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> watermark <span class="im">import</span> watermark</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> sklearn</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> scipy</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> sklearn.utils <span class="im">import</span> check_random_state</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> liana.mt <span class="im">import</span> rank_aggregate</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> gprofiler <span class="im">import</span> GProfiler</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pacmap</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> umap</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> scvelo <span class="im">as</span> scv</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> scanpy <span class="im">as</span> sc</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cellrank <span class="im">as</span> cr</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> anndata <span class="im">as</span> ad</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> plotnine <span class="im">as</span> p9</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> liana <span class="im">as</span> li</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>warnings.simplefilter(<span class="st">"ignore"</span>, category<span class="op">=</span><span class="pp">UserWarning</span>)</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>warnings.simplefilter(<span class="st">"ignore"</span>, category<span class="op">=</span><span class="pp">FutureWarning</span>)</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>warnings.simplefilter(<span class="st">"ignore"</span>, category<span class="op">=</span><span class="pp">DeprecationWarning</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="genes" class="level3">
<h3 class="anchored" data-anchor-id="genes">genes</h3>
<div class="cell" data-execution_count="2">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>npr <span class="op">=</span> [</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adcyap1r1"</span>,</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Avpr1a"</span>,</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Calcr"</span>,</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Calcrl"</span>,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cckar"</span>,</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cckbr"</span>,</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cntfr"</span>,</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Crhr1"</span>,</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Crhr2"</span>,</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Esr1"</span>,</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Galr1"</span>,</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Galr2"</span>,</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Galr3"</span>,</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ghr"</span>,</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ghrhr"</span>,</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ghsr"</span>,</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Glp1r"</span>,</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gpr55"</span>,</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gpr83"</span>,</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gpr149"</span>,</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grpr"</span>,</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Hcrtr1"</span>,</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Hcrtr2"</span>,</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Igf1r"</span>,</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Insr"</span>,</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Insrr"</span>,</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Kiss1r"</span>,</span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Lepr"</span>,</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mc1r"</span>,</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mc3r"</span>,</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mc4r"</span>,</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mchr1"</span>,</span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Nmbr"</span>,</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Nmur1"</span>,</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Nmur2"</span>,</span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Npffr1"</span>,</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Npffr2"</span>,</span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Npr1"</span>,</span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Npr2"</span>,</span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Npr3"</span>,</span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Npsr1"</span>,</span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Npsr2"</span>,</span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Npy1r"</span>,</span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Npy2r"</span>,</span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Npy5r"</span>,</span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ntrk2"</span>,</span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ntsr1"</span>,</span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ntsr2"</span>,</span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Oprd1"</span>,</span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Oprk1"</span>,</span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Oprl1"</span>,</span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Oprm1"</span>,</span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Oxtr"</span>,</span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Prlhr"</span>,</span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Prlr"</span>,</span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Prokr2"</span>,</span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Qrfpr"</span>,</span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Rxfp1"</span>,</span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Rxfp2"</span>,</span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Sstr1"</span>,</span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Sstr2"</span>,</span>
<span id="cb2-63"><a href="#cb2-63" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Sstr3"</span>,</span>
<span id="cb2-64"><a href="#cb2-64" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Tacr1"</span>,</span>
<span id="cb2-65"><a href="#cb2-65" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Tacr3"</span>,</span>
<span id="cb2-66"><a href="#cb2-66" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Trhr"</span>,</span>
<span id="cb2-67"><a href="#cb2-67" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Trhr2"</span>,</span>
<span id="cb2-68"><a href="#cb2-68" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Tshr"</span>,</span>
<span id="cb2-69"><a href="#cb2-69" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Vipr1"</span>,</span>
<span id="cb2-70"><a href="#cb2-70" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Vipr2"</span>,</span>
<span id="cb2-71"><a href="#cb2-71" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb2-72"><a href="#cb2-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-73"><a href="#cb2-73" aria-hidden="true" tabindex="-1"></a>npep <span class="op">=</span> [</span>
<span id="cb2-74"><a href="#cb2-74" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adcyap1"</span>,</span>
<span id="cb2-75"><a href="#cb2-75" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Agrp"</span>,</span>
<span id="cb2-76"><a href="#cb2-76" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Avp"</span>,</span>
<span id="cb2-77"><a href="#cb2-77" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Bdnf"</span>,</span>
<span id="cb2-78"><a href="#cb2-78" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cartpt"</span>,</span>
<span id="cb2-79"><a href="#cb2-79" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cck"</span>,</span>
<span id="cb2-80"><a href="#cb2-80" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cntf"</span>,</span>
<span id="cb2-81"><a href="#cb2-81" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Crh"</span>,</span>
<span id="cb2-82"><a href="#cb2-82" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gal"</span>,</span>
<span id="cb2-83"><a href="#cb2-83" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ghrh"</span>,</span>
<span id="cb2-84"><a href="#cb2-84" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ghrl"</span>,</span>
<span id="cb2-85"><a href="#cb2-85" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grp"</span>,</span>
<span id="cb2-86"><a href="#cb2-86" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Hcrt"</span>,</span>
<span id="cb2-87"><a href="#cb2-87" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Kiss1"</span>,</span>
<span id="cb2-88"><a href="#cb2-88" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Lep"</span>,</span>
<span id="cb2-89"><a href="#cb2-89" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Nmb"</span>,</span>
<span id="cb2-90"><a href="#cb2-90" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Nms"</span>,</span>
<span id="cb2-91"><a href="#cb2-91" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Nmu"</span>,</span>
<span id="cb2-92"><a href="#cb2-92" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Npvf"</span>,</span>
<span id="cb2-93"><a href="#cb2-93" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Npw"</span>,</span>
<span id="cb2-94"><a href="#cb2-94" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Npy"</span>,</span>
<span id="cb2-95"><a href="#cb2-95" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Nts"</span>,</span>
<span id="cb2-96"><a href="#cb2-96" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Oxt"</span>,</span>
<span id="cb2-97"><a href="#cb2-97" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pdyn"</span>,</span>
<span id="cb2-98"><a href="#cb2-98" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Penk"</span>,</span>
<span id="cb2-99"><a href="#cb2-99" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pmch"</span>,</span>
<span id="cb2-100"><a href="#cb2-100" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pnoc"</span>,</span>
<span id="cb2-101"><a href="#cb2-101" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pomc"</span>,</span>
<span id="cb2-102"><a href="#cb2-102" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Qrfp"</span>,</span>
<span id="cb2-103"><a href="#cb2-103" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Reln"</span>,</span>
<span id="cb2-104"><a href="#cb2-104" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Rln1"</span>,</span>
<span id="cb2-105"><a href="#cb2-105" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Rln3"</span>,</span>
<span id="cb2-106"><a href="#cb2-106" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Sst"</span>,</span>
<span id="cb2-107"><a href="#cb2-107" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Tac1"</span>,</span>
<span id="cb2-108"><a href="#cb2-108" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Tac2"</span>,</span>
<span id="cb2-109"><a href="#cb2-109" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Trh"</span>,</span>
<span id="cb2-110"><a href="#cb2-110" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb2-111"><a href="#cb2-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-112"><a href="#cb2-112" aria-hidden="true" tabindex="-1"></a>genes_embed <span class="op">=</span> [</span>
<span id="cb2-113"><a href="#cb2-113" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Abcd1"</span>,</span>
<span id="cb2-114"><a href="#cb2-114" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Abcd2"</span>,</span>
<span id="cb2-115"><a href="#cb2-115" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Abcd3"</span>,</span>
<span id="cb2-116"><a href="#cb2-116" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Acaa1"</span>,</span>
<span id="cb2-117"><a href="#cb2-117" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Acaa2"</span>,</span>
<span id="cb2-118"><a href="#cb2-118" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Acox1"</span>,</span>
<span id="cb2-119"><a href="#cb2-119" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Agrn"</span>,</span>
<span id="cb2-120"><a href="#cb2-120" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Agt"</span>,</span>
<span id="cb2-121"><a href="#cb2-121" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Alcam"</span>,</span>
<span id="cb2-122"><a href="#cb2-122" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Aldh1a1"</span>,</span>
<span id="cb2-123"><a href="#cb2-123" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Aldh1l1"</span>,</span>
<span id="cb2-124"><a href="#cb2-124" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Aldoc"</span>,</span>
<span id="cb2-125"><a href="#cb2-125" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Angpt1"</span>,</span>
<span id="cb2-126"><a href="#cb2-126" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Apoe"</span>,</span>
<span id="cb2-127"><a href="#cb2-127" aria-hidden="true" tabindex="-1"></a>    <span class="st">"App"</span>,</span>
<span id="cb2-128"><a href="#cb2-128" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Aqp4"</span>,</span>
<span id="cb2-129"><a href="#cb2-129" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Arf1"</span>,</span>
<span id="cb2-130"><a href="#cb2-130" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Bmp7"</span>,</span>
<span id="cb2-131"><a href="#cb2-131" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Bsg"</span>,</span>
<span id="cb2-132"><a href="#cb2-132" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Caf4"</span>,</span>
<span id="cb2-133"><a href="#cb2-133" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ccl25"</span>,</span>
<span id="cb2-134"><a href="#cb2-134" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ckb"</span>,</span>
<span id="cb2-135"><a href="#cb2-135" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cnr1"</span>,</span>
<span id="cb2-136"><a href="#cb2-136" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cnr2"</span>,</span>
<span id="cb2-137"><a href="#cb2-137" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Col4a5"</span>,</span>
<span id="cb2-138"><a href="#cb2-138" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cst3"</span>,</span>
<span id="cb2-139"><a href="#cb2-139" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Dagla"</span>,</span>
<span id="cb2-140"><a href="#cb2-140" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Daglb"</span>,</span>
<span id="cb2-141"><a href="#cb2-141" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Decr2"</span>,</span>
<span id="cb2-142"><a href="#cb2-142" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Dnm1"</span>,</span>
<span id="cb2-143"><a href="#cb2-143" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Drp1"</span>,</span>
<span id="cb2-144"><a href="#cb2-144" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ech1"</span>,</span>
<span id="cb2-145"><a href="#cb2-145" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Efna5"</span>,</span>
<span id="cb2-146"><a href="#cb2-146" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Egfr"</span>,</span>
<span id="cb2-147"><a href="#cb2-147" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Enho"</span>,</span>
<span id="cb2-148"><a href="#cb2-148" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Eno1"</span>,</span>
<span id="cb2-149"><a href="#cb2-149" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Faah"</span>,</span>
<span id="cb2-150"><a href="#cb2-150" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fgf1"</span>,</span>
<span id="cb2-151"><a href="#cb2-151" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fgfr3"</span>,</span>
<span id="cb2-152"><a href="#cb2-152" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fis1"</span>,</span>
<span id="cb2-153"><a href="#cb2-153" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fos"</span>,</span>
<span id="cb2-154"><a href="#cb2-154" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fth1"</span>,</span>
<span id="cb2-155"><a href="#cb2-155" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ftl1"</span>,</span>
<span id="cb2-156"><a href="#cb2-156" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gfap"</span>,</span>
<span id="cb2-157"><a href="#cb2-157" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gja1"</span>,</span>
<span id="cb2-158"><a href="#cb2-158" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gli1"</span>,</span>
<span id="cb2-159"><a href="#cb2-159" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Glul"</span>,</span>
<span id="cb2-160"><a href="#cb2-160" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gnai2"</span>,</span>
<span id="cb2-161"><a href="#cb2-161" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gnas"</span>,</span>
<span id="cb2-162"><a href="#cb2-162" aria-hidden="true" tabindex="-1"></a>    <span class="st">"H2-K1"</span>,</span>
<span id="cb2-163"><a href="#cb2-163" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Hacd2"</span>,</span>
<span id="cb2-164"><a href="#cb2-164" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Hadhb"</span>,</span>
<span id="cb2-165"><a href="#cb2-165" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Hbegf"</span>,</span>
<span id="cb2-166"><a href="#cb2-166" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Hepacam"</span>,</span>
<span id="cb2-167"><a href="#cb2-167" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Hif1"</span>,</span>
<span id="cb2-168"><a href="#cb2-168" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Htra1"</span>,</span>
<span id="cb2-169"><a href="#cb2-169" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Igsf1"</span>,</span>
<span id="cb2-170"><a href="#cb2-170" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Il18"</span>,</span>
<span id="cb2-171"><a href="#cb2-171" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Il1rapl1"</span>,</span>
<span id="cb2-172"><a href="#cb2-172" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Itgav"</span>,</span>
<span id="cb2-173"><a href="#cb2-173" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Jam2"</span>,</span>
<span id="cb2-174"><a href="#cb2-174" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Lama2"</span>,</span>
<span id="cb2-175"><a href="#cb2-175" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Lamb2"</span>,</span>
<span id="cb2-176"><a href="#cb2-176" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Lcat"</span>,</span>
<span id="cb2-177"><a href="#cb2-177" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Lgi1"</span>,</span>
<span id="cb2-178"><a href="#cb2-178" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Lgi4"</span>,</span>
<span id="cb2-179"><a href="#cb2-179" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Lpcat3"</span>,</span>
<span id="cb2-180"><a href="#cb2-180" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Lrpap1"</span>,</span>
<span id="cb2-181"><a href="#cb2-181" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Lrrc4b"</span>,</span>
<span id="cb2-182"><a href="#cb2-182" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Lxn"</span>,</span>
<span id="cb2-183"><a href="#cb2-183" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mdk"</span>,</span>
<span id="cb2-184"><a href="#cb2-184" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mdv1"</span>,</span>
<span id="cb2-185"><a href="#cb2-185" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mfn1"</span>,</span>
<span id="cb2-186"><a href="#cb2-186" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mfn2"</span>,</span>
<span id="cb2-187"><a href="#cb2-187" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mgll"</span>,</span>
<span id="cb2-188"><a href="#cb2-188" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mief1"</span>,</span>
<span id="cb2-189"><a href="#cb2-189" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Napepld"</span>,</span>
<span id="cb2-190"><a href="#cb2-190" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ncam1"</span>,</span>
<span id="cb2-191"><a href="#cb2-191" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ncan"</span>,</span>
<span id="cb2-192"><a href="#cb2-192" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ndrg2"</span>,</span>
<span id="cb2-193"><a href="#cb2-193" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Nfasc"</span>,</span>
<span id="cb2-194"><a href="#cb2-194" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Nfia"</span>,</span>
<span id="cb2-195"><a href="#cb2-195" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Nlgn3"</span>,</span>
<span id="cb2-196"><a href="#cb2-196" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Nrxn1"</span>,</span>
<span id="cb2-197"><a href="#cb2-197" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Nrxn2"</span>,</span>
<span id="cb2-198"><a href="#cb2-198" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ntn1"</span>,</span>
<span id="cb2-199"><a href="#cb2-199" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ntrk3"</span>,</span>
<span id="cb2-200"><a href="#cb2-200" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Opa1"</span>,</span>
<span id="cb2-201"><a href="#cb2-201" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Otp"</span>,</span>
<span id="cb2-202"><a href="#cb2-202" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pex1"</span>,</span>
<span id="cb2-203"><a href="#cb2-203" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pex10"</span>,</span>
<span id="cb2-204"><a href="#cb2-204" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pex12"</span>,</span>
<span id="cb2-205"><a href="#cb2-205" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pex13"</span>,</span>
<span id="cb2-206"><a href="#cb2-206" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pex14"</span>,</span>
<span id="cb2-207"><a href="#cb2-207" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pex16"</span>,</span>
<span id="cb2-208"><a href="#cb2-208" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pex2"</span>,</span>
<span id="cb2-209"><a href="#cb2-209" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pex26"</span>,</span>
<span id="cb2-210"><a href="#cb2-210" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pex3"</span>,</span>
<span id="cb2-211"><a href="#cb2-211" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pex6"</span>,</span>
<span id="cb2-212"><a href="#cb2-212" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pkm"</span>,</span>
<span id="cb2-213"><a href="#cb2-213" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pla2g7"</span>,</span>
<span id="cb2-214"><a href="#cb2-214" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Plcb1"</span>,</span>
<span id="cb2-215"><a href="#cb2-215" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Psap"</span>,</span>
<span id="cb2-216"><a href="#cb2-216" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ptn"</span>,</span>
<span id="cb2-217"><a href="#cb2-217" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pygb"</span>,</span>
<span id="cb2-218"><a href="#cb2-218" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Rgma"</span>,</span>
<span id="cb2-219"><a href="#cb2-219" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Rtn4"</span>,</span>
<span id="cb2-220"><a href="#cb2-220" aria-hidden="true" tabindex="-1"></a>    <span class="st">"S100a1"</span>,</span>
<span id="cb2-221"><a href="#cb2-221" aria-hidden="true" tabindex="-1"></a>    <span class="st">"S100a6"</span>,</span>
<span id="cb2-222"><a href="#cb2-222" aria-hidden="true" tabindex="-1"></a>    <span class="st">"S100b"</span>,</span>
<span id="cb2-223"><a href="#cb2-223" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Scd2"</span>,</span>
<span id="cb2-224"><a href="#cb2-224" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Sdc2"</span>,</span>
<span id="cb2-225"><a href="#cb2-225" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Sema6a"</span>,</span>
<span id="cb2-226"><a href="#cb2-226" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Sema6d"</span>,</span>
<span id="cb2-227"><a href="#cb2-227" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Sgcd"</span>,</span>
<span id="cb2-228"><a href="#cb2-228" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Sirpa"</span>,</span>
<span id="cb2-229"><a href="#cb2-229" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Slc1a2"</span>,</span>
<span id="cb2-230"><a href="#cb2-230" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Slc1a3"</span>,</span>
<span id="cb2-231"><a href="#cb2-231" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Slc38a1"</span>,</span>
<span id="cb2-232"><a href="#cb2-232" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Slc4a4"</span>,</span>
<span id="cb2-233"><a href="#cb2-233" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Slc6a11"</span>,</span>
<span id="cb2-234"><a href="#cb2-234" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Slc7a10"</span>,</span>
<span id="cb2-235"><a href="#cb2-235" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Slit1"</span>,</span>
<span id="cb2-236"><a href="#cb2-236" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Slit2"</span>,</span>
<span id="cb2-237"><a href="#cb2-237" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Slitrk2"</span>,</span>
<span id="cb2-238"><a href="#cb2-238" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Sorbs1"</span>,</span>
<span id="cb2-239"><a href="#cb2-239" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Sox9"</span>,</span>
<span id="cb2-240"><a href="#cb2-240" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Sparc"</span>,</span>
<span id="cb2-241"><a href="#cb2-241" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Spon1"</span>,</span>
<span id="cb2-242"><a href="#cb2-242" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Tafa1"</span>,</span>
<span id="cb2-243"><a href="#cb2-243" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Timp3"</span>,</span>
<span id="cb2-244"><a href="#cb2-244" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Tkt"</span>,</span>
<span id="cb2-245"><a href="#cb2-245" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Trpv1"</span>,</span>
<span id="cb2-246"><a href="#cb2-246" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Vcam1"</span>,</span>
<span id="cb2-247"><a href="#cb2-247" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Vegfa"</span>,</span>
<span id="cb2-248"><a href="#cb2-248" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb2-249"><a href="#cb2-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-250"><a href="#cb2-250" aria-hidden="true" tabindex="-1"></a>nmr <span class="op">=</span> [</span>
<span id="cb2-251"><a href="#cb2-251" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adra1a"</span>,</span>
<span id="cb2-252"><a href="#cb2-252" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adra1b"</span>,</span>
<span id="cb2-253"><a href="#cb2-253" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adra1d"</span>,</span>
<span id="cb2-254"><a href="#cb2-254" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adra2a"</span>,</span>
<span id="cb2-255"><a href="#cb2-255" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adra2b"</span>,</span>
<span id="cb2-256"><a href="#cb2-256" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adra2c"</span>,</span>
<span id="cb2-257"><a href="#cb2-257" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adrb1"</span>,</span>
<span id="cb2-258"><a href="#cb2-258" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adrb2"</span>,</span>
<span id="cb2-259"><a href="#cb2-259" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adrb3"</span>,</span>
<span id="cb2-260"><a href="#cb2-260" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adrbk1"</span>,</span>
<span id="cb2-261"><a href="#cb2-261" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adrbk2"</span>,</span>
<span id="cb2-262"><a href="#cb2-262" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adrm1"</span>,  <span class="co"># adrenergic receptors</span></span>
<span id="cb2-263"><a href="#cb2-263" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adora1"</span>,</span>
<span id="cb2-264"><a href="#cb2-264" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adora2a"</span>,</span>
<span id="cb2-265"><a href="#cb2-265" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adora2b"</span>,</span>
<span id="cb2-266"><a href="#cb2-266" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Adora3"</span>,  <span class="co"># adenosine receptors</span></span>
<span id="cb2-267"><a href="#cb2-267" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrm1"</span>,</span>
<span id="cb2-268"><a href="#cb2-268" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrm2"</span>,</span>
<span id="cb2-269"><a href="#cb2-269" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrm3"</span>,</span>
<span id="cb2-270"><a href="#cb2-270" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrm4"</span>,</span>
<span id="cb2-271"><a href="#cb2-271" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrm5"</span>,</span>
<span id="cb2-272"><a href="#cb2-272" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrna1"</span>,</span>
<span id="cb2-273"><a href="#cb2-273" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrna2"</span>,</span>
<span id="cb2-274"><a href="#cb2-274" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrna3"</span>,</span>
<span id="cb2-275"><a href="#cb2-275" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrna4"</span>,</span>
<span id="cb2-276"><a href="#cb2-276" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrna5"</span>,</span>
<span id="cb2-277"><a href="#cb2-277" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrna6"</span>,</span>
<span id="cb2-278"><a href="#cb2-278" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrna7"</span>,</span>
<span id="cb2-279"><a href="#cb2-279" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrna9"</span>,</span>
<span id="cb2-280"><a href="#cb2-280" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrna10"</span>,</span>
<span id="cb2-281"><a href="#cb2-281" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrnb1"</span>,</span>
<span id="cb2-282"><a href="#cb2-282" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrnb2"</span>,</span>
<span id="cb2-283"><a href="#cb2-283" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrnb3"</span>,</span>
<span id="cb2-284"><a href="#cb2-284" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrnd"</span>,</span>
<span id="cb2-285"><a href="#cb2-285" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Chrng"</span>,  <span class="co"># cholinergic receptors</span></span>
<span id="cb2-286"><a href="#cb2-286" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gria1"</span>,</span>
<span id="cb2-287"><a href="#cb2-287" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gria2"</span>,</span>
<span id="cb2-288"><a href="#cb2-288" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gria3"</span>,</span>
<span id="cb2-289"><a href="#cb2-289" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gria4"</span>,  <span class="co"># iGlu AMPA receptors</span></span>
<span id="cb2-290"><a href="#cb2-290" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grid1"</span>,</span>
<span id="cb2-291"><a href="#cb2-291" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grid2"</span>,  <span class="co"># iGlu delta receptors</span></span>
<span id="cb2-292"><a href="#cb2-292" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grik1"</span>,</span>
<span id="cb2-293"><a href="#cb2-293" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grik2"</span>,</span>
<span id="cb2-294"><a href="#cb2-294" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grik3"</span>,</span>
<span id="cb2-295"><a href="#cb2-295" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grik4"</span>,</span>
<span id="cb2-296"><a href="#cb2-296" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grik5"</span>,  <span class="co"># iGlu kainate receptors</span></span>
<span id="cb2-297"><a href="#cb2-297" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grin1"</span>,</span>
<span id="cb2-298"><a href="#cb2-298" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grin2a"</span>,</span>
<span id="cb2-299"><a href="#cb2-299" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grin2b"</span>,</span>
<span id="cb2-300"><a href="#cb2-300" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grin2c"</span>,</span>
<span id="cb2-301"><a href="#cb2-301" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grin2d"</span>,</span>
<span id="cb2-302"><a href="#cb2-302" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grin3a"</span>,</span>
<span id="cb2-303"><a href="#cb2-303" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grin3b"</span>,  <span class="co"># iGlu NMDA receptors</span></span>
<span id="cb2-304"><a href="#cb2-304" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grm1"</span>,</span>
<span id="cb2-305"><a href="#cb2-305" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grm5"</span>,  <span class="co"># mGluRs 1</span></span>
<span id="cb2-306"><a href="#cb2-306" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grm2"</span>,</span>
<span id="cb2-307"><a href="#cb2-307" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grm3"</span>,  <span class="co"># mGluRs 2</span></span>
<span id="cb2-308"><a href="#cb2-308" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grm4"</span>,</span>
<span id="cb2-309"><a href="#cb2-309" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grm6"</span>,</span>
<span id="cb2-310"><a href="#cb2-310" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grm7"</span>,</span>
<span id="cb2-311"><a href="#cb2-311" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Grm8"</span>,  <span class="co"># mGluRs 3</span></span>
<span id="cb2-312"><a href="#cb2-312" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabra1"</span>,</span>
<span id="cb2-313"><a href="#cb2-313" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabra2"</span>,</span>
<span id="cb2-314"><a href="#cb2-314" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabra3"</span>,</span>
<span id="cb2-315"><a href="#cb2-315" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabra4"</span>,</span>
<span id="cb2-316"><a href="#cb2-316" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabra5"</span>,</span>
<span id="cb2-317"><a href="#cb2-317" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabra6"</span>,</span>
<span id="cb2-318"><a href="#cb2-318" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabrb1"</span>,</span>
<span id="cb2-319"><a href="#cb2-319" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabrb2"</span>,</span>
<span id="cb2-320"><a href="#cb2-320" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabrb3"</span>,</span>
<span id="cb2-321"><a href="#cb2-321" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabrg1"</span>,</span>
<span id="cb2-322"><a href="#cb2-322" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabrg2"</span>,</span>
<span id="cb2-323"><a href="#cb2-323" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabrg3"</span>,</span>
<span id="cb2-324"><a href="#cb2-324" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabrd"</span>,</span>
<span id="cb2-325"><a href="#cb2-325" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabre"</span>,</span>
<span id="cb2-326"><a href="#cb2-326" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabrp"</span>,</span>
<span id="cb2-327"><a href="#cb2-327" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabrq"</span>,</span>
<span id="cb2-328"><a href="#cb2-328" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabrr1"</span>,</span>
<span id="cb2-329"><a href="#cb2-329" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabrr2"</span>,</span>
<span id="cb2-330"><a href="#cb2-330" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabrr3"</span>,</span>
<span id="cb2-331"><a href="#cb2-331" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabbr1"</span>,</span>
<span id="cb2-332"><a href="#cb2-332" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gabbr2"</span>,  <span class="co"># GABA receptors</span></span>
<span id="cb2-333"><a href="#cb2-333" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Drd1"</span>,</span>
<span id="cb2-334"><a href="#cb2-334" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Drd2"</span>,</span>
<span id="cb2-335"><a href="#cb2-335" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Drd3"</span>,</span>
<span id="cb2-336"><a href="#cb2-336" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Drd4"</span>,</span>
<span id="cb2-337"><a href="#cb2-337" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Drd5"</span>,  <span class="co"># dopamine receptors</span></span>
<span id="cb2-338"><a href="#cb2-338" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Htr1a"</span>,</span>
<span id="cb2-339"><a href="#cb2-339" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Htr1b"</span>,</span>
<span id="cb2-340"><a href="#cb2-340" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Htr1d"</span>,</span>
<span id="cb2-341"><a href="#cb2-341" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Htr1f"</span>,</span>
<span id="cb2-342"><a href="#cb2-342" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Htr2a"</span>,</span>
<span id="cb2-343"><a href="#cb2-343" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Htr2b"</span>,</span>
<span id="cb2-344"><a href="#cb2-344" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Htr2c"</span>,</span>
<span id="cb2-345"><a href="#cb2-345" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Htr3a"</span>,</span>
<span id="cb2-346"><a href="#cb2-346" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Htr3b"</span>,</span>
<span id="cb2-347"><a href="#cb2-347" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Htr4"</span>,</span>
<span id="cb2-348"><a href="#cb2-348" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Htr5a"</span>,</span>
<span id="cb2-349"><a href="#cb2-349" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Htr5b"</span>,</span>
<span id="cb2-350"><a href="#cb2-350" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Htr6"</span>,</span>
<span id="cb2-351"><a href="#cb2-351" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Htr7"</span>,  <span class="co"># serotonin receptors</span></span>
<span id="cb2-352"><a href="#cb2-352" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gnas"</span>,</span>
<span id="cb2-353"><a href="#cb2-353" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gnai1"</span>,</span>
<span id="cb2-354"><a href="#cb2-354" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gnai2"</span>,</span>
<span id="cb2-355"><a href="#cb2-355" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gnai3"</span>,</span>
<span id="cb2-356"><a href="#cb2-356" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gnao1"</span>,</span>
<span id="cb2-357"><a href="#cb2-357" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gnao2"</span>,</span>
<span id="cb2-358"><a href="#cb2-358" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gnaq"</span>,</span>
<span id="cb2-359"><a href="#cb2-359" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gna11"</span>,</span>
<span id="cb2-360"><a href="#cb2-360" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gna12"</span>,</span>
<span id="cb2-361"><a href="#cb2-361" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gna13"</span>,</span>
<span id="cb2-362"><a href="#cb2-362" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gnal"</span>,</span>
<span id="cb2-363"><a href="#cb2-363" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gnasxl"</span>,  <span class="co"># G protein alpha subunit</span></span>
<span id="cb2-364"><a href="#cb2-364" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gnb1"</span>,</span>
<span id="cb2-365"><a href="#cb2-365" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gnb2"</span>,</span>
<span id="cb2-366"><a href="#cb2-366" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gnb3"</span>,</span>
<span id="cb2-367"><a href="#cb2-367" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gnb4"</span>,</span>
<span id="cb2-368"><a href="#cb2-368" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gnb5"</span>,  <span class="co"># G protein beta subunit</span></span>
<span id="cb2-369"><a href="#cb2-369" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gng2"</span>,</span>
<span id="cb2-370"><a href="#cb2-370" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gng3"</span>,</span>
<span id="cb2-371"><a href="#cb2-371" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gng4"</span>,</span>
<span id="cb2-372"><a href="#cb2-372" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gng5"</span>,</span>
<span id="cb2-373"><a href="#cb2-373" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gng7"</span>,</span>
<span id="cb2-374"><a href="#cb2-374" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gng8"</span>,</span>
<span id="cb2-375"><a href="#cb2-375" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gng10"</span>,</span>
<span id="cb2-376"><a href="#cb2-376" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gng11"</span>,</span>
<span id="cb2-377"><a href="#cb2-377" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gng12"</span>,</span>
<span id="cb2-378"><a href="#cb2-378" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gng13"</span>,</span>
<span id="cb2-379"><a href="#cb2-379" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gngt1"</span>,</span>
<span id="cb2-380"><a href="#cb2-380" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gngt2"</span>,  <span class="co"># G protein gamma subunit</span></span>
<span id="cb2-381"><a href="#cb2-381" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P2rx1"</span>,</span>
<span id="cb2-382"><a href="#cb2-382" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P2rx2"</span>,</span>
<span id="cb2-383"><a href="#cb2-383" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P2rx3"</span>,</span>
<span id="cb2-384"><a href="#cb2-384" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P2rx4"</span>,</span>
<span id="cb2-385"><a href="#cb2-385" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P2rx5"</span>,</span>
<span id="cb2-386"><a href="#cb2-386" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P2rx6"</span>,</span>
<span id="cb2-387"><a href="#cb2-387" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P2rx7"</span>,</span>
<span id="cb2-388"><a href="#cb2-388" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P2ry1"</span>,</span>
<span id="cb2-389"><a href="#cb2-389" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P2ry2"</span>,</span>
<span id="cb2-390"><a href="#cb2-390" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P2ry4"</span>,</span>
<span id="cb2-391"><a href="#cb2-391" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P2ry6"</span>,</span>
<span id="cb2-392"><a href="#cb2-392" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P2ry12"</span>,</span>
<span id="cb2-393"><a href="#cb2-393" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P2ry13"</span>,</span>
<span id="cb2-394"><a href="#cb2-394" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P2ry14"</span>,  <span class="co"># purinergic receptors</span></span>
<span id="cb2-395"><a href="#cb2-395" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ryr1"</span>,</span>
<span id="cb2-396"><a href="#cb2-396" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ryr2"</span>,</span>
<span id="cb2-397"><a href="#cb2-397" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ryr3"</span>,  <span class="co"># ryanodine receptors</span></span>
<span id="cb2-398"><a href="#cb2-398" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb2-399"><a href="#cb2-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-400"><a href="#cb2-400" aria-hidden="true" tabindex="-1"></a>mitochondrial <span class="op">=</span> [</span>
<span id="cb2-401"><a href="#cb2-401" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mfn1"</span>,  <span class="co"># Mitofusin 1: A GTPase located on the outer mitochondrial membrane that plays a crucial role in mitochondrial fusion. Mfn1 mediates tethering and fusion of the outer mitochondrial membranes, helping to maintain the integrity and function of the mitochondrial network.</span></span>
<span id="cb2-402"><a href="#cb2-402" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mfn2"</span>,  <span class="co"># Mitofusin 2: Another GTPase on the outer mitochondrial membrane that is involved in mitochondrial fusion. Mfn2 works in concert with Mfn1 to mediate the fusion of outer mitochondrial membranes, maintaining proper mitochondrial function and dynamics.</span></span>
<span id="cb2-403"><a href="#cb2-403" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Opa1"</span>,  <span class="co"># Optic atrophy 1: A dynamin-related GTPase located in the inner mitochondrial membrane that is crucial for the regulation of mitochondrial fusion. Opa1 mediates the fusion of inner mitochondrial membranes, contributing to the maintenance of mitochondrial morphology, cristae structure, and overall mitochondrial function. Additionally, Opa1 plays a role in the regulation of apoptosis and is involved in the proper functioning of the electron transport chain.</span></span>
<span id="cb2-404"><a href="#cb2-404" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Dnm1"</span>,  <span class="co"># Dynamin 1: A GTPase involved in the regulation of membrane remodeling and trafficking, including vesicle formation and endocytosis. It has also been implicated in the regulation of mitochondrial morphology and function.</span></span>
<span id="cb2-405"><a href="#cb2-405" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Drp1"</span>,  <span class="co"># Dynamin-related protein 1: Another GTPase, which plays a central role in mitochondrial fission. It is recruited to the outer mitochondrial membrane during fission and is crucial for the proper division and maintenance of mitochondrial networks.</span></span>
<span id="cb2-406"><a href="#cb2-406" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Dnm1l"</span>,  <span class="co"># Dynamin 1-like, involved in mitochondrial fission</span></span>
<span id="cb2-407"><a href="#cb2-407" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fis1"</span>,  <span class="co"># Fission 1, involved in mitochondrial fission</span></span>
<span id="cb2-408"><a href="#cb2-408" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mff"</span>,  <span class="co"># Mitochondrial fission factor, involved in mitochondrial fission</span></span>
<span id="cb2-409"><a href="#cb2-409" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mid49"</span>,  <span class="co"># Mitochondrial dynamics protein of 49 kDa, involved in mitochondrial fission</span></span>
<span id="cb2-410"><a href="#cb2-410" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mid51"</span>,  <span class="co"># Mitochondrial dynamics protein of 51 kDa, involved in mitochondrial fission</span></span>
<span id="cb2-411"><a href="#cb2-411" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ppargc1a"</span>,  <span class="co"># Peroxisome proliferator-activated receptor gamma coactivator 1-alpha, involved in mitochondrial biogenesis</span></span>
<span id="cb2-412"><a href="#cb2-412" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ppargc1b"</span>,  <span class="co"># Peroxisome proliferator-activated receptor gamma coactivator 1-beta, involved in mitochondrial biogenesis</span></span>
<span id="cb2-413"><a href="#cb2-413" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Nrf1"</span>,  <span class="co"># Nuclear respiratory factor 1, involved in mitochondrial biogenesis</span></span>
<span id="cb2-414"><a href="#cb2-414" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Nrf2"</span>,  <span class="co"># Nuclear respiratory factor 2, involved in mitochondrial biogenesis</span></span>
<span id="cb2-415"><a href="#cb2-415" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Tfam"</span>,  <span class="co"># Mitochondrial transcription factor A, involved in mitochondrial biogenesis</span></span>
<span id="cb2-416"><a href="#cb2-416" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Sod2"</span>,  <span class="co"># Superoxide dismutase 2, involved in mitochondrial antioxidant defense</span></span>
<span id="cb2-417"><a href="#cb2-417" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Prdx3"</span>,  <span class="co"># Peroxiredoxin 3, involved in mitochondrial antioxidant defense</span></span>
<span id="cb2-418"><a href="#cb2-418" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Slc25a1"</span>,  <span class="co"># Solute carrier family 25 member 1, involved in mitochondrial metabolism</span></span>
<span id="cb2-419"><a href="#cb2-419" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Slc25a4"</span>,  <span class="co"># Solute carrier family 25 member 4, involved in mitochondrial metabolism</span></span>
<span id="cb2-420"><a href="#cb2-420" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Slc25a5"</span>,  <span class="co"># Solute carrier family 25 member 5, involved in mitochondrial metabolism</span></span>
<span id="cb2-421"><a href="#cb2-421" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Uqcrc1"</span>,  <span class="co"># Ubiquinol-cytochrome c reductase core protein 1, involved in mitochondrial electron transport chain</span></span>
<span id="cb2-422"><a href="#cb2-422" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Uqcrc2"</span>,  <span class="co"># Ubiquinol-cytochrome c reductase core protein 2, involved in mitochondrial electron transport chain</span></span>
<span id="cb2-423"><a href="#cb2-423" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cox4i1"</span>,  <span class="co"># Cytochrome c oxidase subunit 4 isoform 1, involved in mitochondrial electron transport chain</span></span>
<span id="cb2-424"><a href="#cb2-424" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cox4i2"</span>,  <span class="co"># Cytochrome c oxidase subunit 4 isoform 2, involved in mitochondrial electron transport chain</span></span>
<span id="cb2-425"><a href="#cb2-425" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Atp5a1"</span>,  <span class="co"># ATP synthase, H+ transporting, mitochondrial F1 complex, alpha subunit 1, involved in mitochondrial ATP production</span></span>
<span id="cb2-426"><a href="#cb2-426" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Atp5b"</span>,  <span class="co"># ATP synthase, H+ transporting, mitochondrial F1 complex, beta subunit, involved in mitochondrial ATP production</span></span>
<span id="cb2-427"><a href="#cb2-427" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Bak1"</span>,  <span class="co"># BCL2 antagonist/killer 1, involved in mitochondrial-mediated apoptosis</span></span>
<span id="cb2-428"><a href="#cb2-428" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Bax"</span>,  <span class="co"># BCL2 associated X, involved in mitochondrial-mediated apoptosis</span></span>
<span id="cb2-429"><a href="#cb2-429" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Bcl2"</span>,  <span class="co"># B-cell CLL/lymphoma 2, involved in mitochondrial-mediated apoptosis</span></span>
<span id="cb2-430"><a href="#cb2-430" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Bcl2l1"</span>,  <span class="co"># BCL2-like 1, involved in mitochondrial-mediated apoptosis</span></span>
<span id="cb2-431"><a href="#cb2-431" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Bnip3"</span>,  <span class="co"># BCL2/adenovirus E1B 19kDa interacting protein 3, involved in mitoptosis</span></span>
<span id="cb2-432"><a href="#cb2-432" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Bnip3l"</span>,  <span class="co"># BCL2/adenovirus E1B 19kDa interacting protein 3-like, involved in mitoptosis</span></span>
<span id="cb2-433"><a href="#cb2-433" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Casp9"</span>,  <span class="co"># Caspase 9, involved in mitochondrial-mediated apoptosis</span></span>
<span id="cb2-434"><a href="#cb2-434" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cybb"</span>,  <span class="co"># Cytochrome b-245, beta polypeptide, involved in oxidative stress</span></span>
<span id="cb2-435"><a href="#cb2-435" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gpx1"</span>,  <span class="co"># Glutathione peroxidase 1, involved in oxidative stress</span></span>
<span id="cb2-436"><a href="#cb2-436" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gpx4"</span>,  <span class="co"># Glutathione peroxidase 4, involved in oxidative stress</span></span>
<span id="cb2-437"><a href="#cb2-437" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mapk14"</span>,  <span class="co"># Mitogen-activated protein kinase 14, involved in ischemia and oxidative stress</span></span>
<span id="cb2-438"><a href="#cb2-438" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Nfe2l2"</span>,  <span class="co"># Nuclear factor, erythroid 2 like 2, involved in oxidative stress response</span></span>
<span id="cb2-439"><a href="#cb2-439" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Nox4"</span>,  <span class="co"># NADPH oxidase 4, involved in oxidative stress</span></span>
<span id="cb2-440"><a href="#cb2-440" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Polg"</span>,  <span class="co"># DNA polymerase gamma, involved in mitochondrial DNA replication</span></span>
<span id="cb2-441"><a href="#cb2-441" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Polg2"</span>,  <span class="co"># DNA polymerase gamma 2, involved in mitochondrial DNA replication</span></span>
<span id="cb2-442"><a href="#cb2-442" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pink1"</span>,  <span class="co"># PTEN-induced putative kinase 1, involved in mitophagy</span></span>
<span id="cb2-443"><a href="#cb2-443" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Park2"</span>,  <span class="co"># Parkin RBR E3 ubiquitin protein ligase, involved in mitophagy</span></span>
<span id="cb2-444"><a href="#cb2-444" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Aifm1"</span>,  <span class="co"># Apoptosis-inducing factor, mitochondrion-associated, 1, involved in DNA damage and apoptosis</span></span>
<span id="cb2-445"><a href="#cb2-445" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ogg1"</span>,  <span class="co"># 8-oxoguanine DNA glycosylase 1, involved in DNA damage repair</span></span>
<span id="cb2-446"><a href="#cb2-446" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Mutyh"</span>,  <span class="co"># MutY DNA glycosylase, involved in DNA damage repair</span></span>
<span id="cb2-447"><a href="#cb2-447" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Sod1"</span>,  <span class="co"># Superoxide dismutase 1, involved in oxidative stress</span></span>
<span id="cb2-448"><a href="#cb2-448" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Sod3"</span>,  <span class="co"># Superoxide dismutase 3, involved in oxidative stress</span></span>
<span id="cb2-449"><a href="#cb2-449" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ucp1"</span>,  <span class="co"># Uncoupling protein 1, involved in thermogenesis</span></span>
<span id="cb2-450"><a href="#cb2-450" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ucp2"</span>,  <span class="co"># Uncoupling protein 2, involved in thermogenesis</span></span>
<span id="cb2-451"><a href="#cb2-451" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ucp3"</span>,  <span class="co"># Uncoupling protein 3, involved in thermogenesis</span></span>
<span id="cb2-452"><a href="#cb2-452" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ucp4"</span>,  <span class="co"># Uncoupling protein 4, involved in thermogenesis</span></span>
<span id="cb2-453"><a href="#cb2-453" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ucp5"</span>,  <span class="co"># Uncoupling protein 5, involved in thermogenesis</span></span>
<span id="cb2-454"><a href="#cb2-454" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fgf21"</span>,  <span class="co"># Fibroblast growth factor 21, involved in metabolic regulation</span></span>
<span id="cb2-455"><a href="#cb2-455" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Klb"</span>,  <span class="co"># Klotho beta (beta-klotho), co-receptor for Fgf21</span></span>
<span id="cb2-456"><a href="#cb2-456" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Additional genes related to brain metabolism switch and different catabolic/anabolic modes</span></span>
<span id="cb2-457"><a href="#cb2-457" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Bdh1"</span>,  <span class="co"># 3-hydroxybutyrate dehydrogenase 1, involved in ketone body metabolism</span></span>
<span id="cb2-458"><a href="#cb2-458" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Bdh2"</span>,  <span class="co"># 3-hydroxybutyrate dehydrogenase 2, involved in ketone body metabolism</span></span>
<span id="cb2-459"><a href="#cb2-459" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Hmgcs2"</span>,  <span class="co"># 3-hydroxy-3-methylglutaryl-CoA synthase 2, involved in ketone body synthesis</span></span>
<span id="cb2-460"><a href="#cb2-460" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Slc16a1"</span>,  <span class="co"># Solute carrier family 16 member 1 (monocarboxylate transporter 1), involved in ketone body transport</span></span>
<span id="cb2-461"><a href="#cb2-461" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Slc16a7"</span>,  <span class="co"># Solute carrier family 16 member 7 (monocarboxylate transporter 2), involved in ketone body transport</span></span>
<span id="cb2-462"><a href="#cb2-462" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Slc2a1"</span>,  <span class="co"># Solute carrier family 2 member 1 (glucose transporter type 1), involved in glucose transport</span></span>
<span id="cb2-463"><a href="#cb2-463" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Slc2a3"</span>,  <span class="co"># Solute carrier family 2 member 3 (glucose transporter type 3), involved in glucose transport</span></span>
<span id="cb2-464"><a href="#cb2-464" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Gck"</span>,  <span class="co"># Glucokinase, involved in glycolysis</span></span>
<span id="cb2-465"><a href="#cb2-465" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pfkp"</span>,  <span class="co"># Phosphofructokinase, platelet, involved in glycolysis</span></span>
<span id="cb2-466"><a href="#cb2-466" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pkm"</span>,  <span class="co"># Pyruvate kinase, muscle, involved in glycolysis</span></span>
<span id="cb2-467"><a href="#cb2-467" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pdha1"</span>,  <span class="co"># Pyruvate dehydrogenase E1 alpha 1, involved in the conversion of pyruvate to acetyl-CoA</span></span>
<span id="cb2-468"><a href="#cb2-468" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pdha2"</span>,  <span class="co"># Pyruvate dehydrogenase E1 alpha 2, involved in the conversion of pyruvate to acetyl-CoA</span></span>
<span id="cb2-469"><a href="#cb2-469" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pdk1"</span>,  <span class="co"># Pyruvate dehydrogenase kinase 1, involved in the regulation of pyruvate dehydrogenase complex</span></span>
<span id="cb2-470"><a href="#cb2-470" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pdk2"</span>,  <span class="co"># Pyruvate dehydrogenase kinase 2, involved in the regulation of pyruvate dehydrogenase complex</span></span>
<span id="cb2-471"><a href="#cb2-471" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pdk3"</span>,  <span class="co"># Pyruvate dehydrogenase kinase 3, involved in the regulation of pyruvate dehydrogenase complex</span></span>
<span id="cb2-472"><a href="#cb2-472" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pdk4"</span>,  <span class="co"># Pyruvate dehydrogenase kinase 4, involved in the regulation of pyruvate dehydrogenase complex</span></span>
<span id="cb2-473"><a href="#cb2-473" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Acox1"</span>,  <span class="co"># Acyl-CoA oxidase 1, involved in fatty acid oxidation</span></span>
<span id="cb2-474"><a href="#cb2-474" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cpt1a"</span>,  <span class="co"># Carnitine palmitoyltransferase 1A, involved in fatty acid oxidation</span></span>
<span id="cb2-475"><a href="#cb2-475" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cpt1b"</span>,  <span class="co"># Carnitine palmitoyltransferase 1B, involved in fatty acid oxidation</span></span>
<span id="cb2-476"><a href="#cb2-476" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cpt1c"</span>,  <span class="co"># Carnitine palmitoyltransferase 1C, involved in fatty acid oxidation</span></span>
<span id="cb2-477"><a href="#cb2-477" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Cpt2"</span>,  <span class="co"># Carnitine palmitoyltransferase 2, involved in fatty acid oxidation</span></span>
<span id="cb2-478"><a href="#cb2-478" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Acsl1"</span>,  <span class="co"># Acyl-CoA synthetase long-chain family member 1, involved in fatty acid activation</span></span>
<span id="cb2-479"><a href="#cb2-479" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Acsl3"</span>,  <span class="co"># Acyl-CoA synthetase long-chain family member 3, involved in fatty acid activation</span></span>
<span id="cb2-480"><a href="#cb2-480" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Acsl4"</span>,  <span class="co"># Acyl-CoA synthetase long-chain family member 4, involved in fatty acid activation</span></span>
<span id="cb2-481"><a href="#cb2-481" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Acsl5"</span>,  <span class="co"># Acyl-CoA synthetase long-chain family member 5, involved in fatty acid activation</span></span>
<span id="cb2-482"><a href="#cb2-482" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Acsl6"</span>,  <span class="co"># Acyl-CoA synthetase long-chain family member 6, involved in fatty acid activation</span></span>
<span id="cb2-483"><a href="#cb2-483" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Acly"</span>,  <span class="co"># ATP citrate lyase, involved in fatty acid synthesis</span></span>
<span id="cb2-484"><a href="#cb2-484" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fasn"</span>,  <span class="co"># Fatty acid synthase, involved in fatty acid synthesis</span></span>
<span id="cb2-485"><a href="#cb2-485" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Scd1"</span>,  <span class="co"># Stearoyl-CoA desaturase 1, involved in fatty acid desaturation</span></span>
<span id="cb2-486"><a href="#cb2-486" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Scd2"</span>,  <span class="co"># Stearoyl-CoA desaturase 2, involved in fatty acid desaturation</span></span>
<span id="cb2-487"><a href="#cb2-487" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Acaa1"</span>,  <span class="co"># Acetyl-CoA acyltransferase 1: Involved in the beta-oxidation of fatty acids in the mitochondria. Catalyzes the last step in the breakdown of fatty acids, converting 3-ketoacyl-CoA to acetyl-CoA.</span></span>
<span id="cb2-488"><a href="#cb2-488" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Acaa2"</span>,  <span class="co"># Acetyl-CoA acyltransferase 2: Also involved in beta-oxidation of fatty acids in the mitochondria. Similar to Acaa1, it catalyzes the last step of fatty acid breakdown but has different substrate specificity.</span></span>
<span id="cb2-489"><a href="#cb2-489" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Aldh1a1"</span>,  <span class="co"># Aldehyde dehydrogenase 1 family, member A1: An enzyme involved in the detoxification of aldehydes (generated from alcohol metabolism and lipid peroxidation) by converting them to carboxylic acids.</span></span>
<span id="cb2-490"><a href="#cb2-490" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Eno1"</span>,  <span class="co"># Enolase 1: A glycolytic enzyme that catalyzes the conversion of 2-phosphoglycerate to phosphoenolpyruvate in the glycolysis pathway, which generates ATP through the breakdown of glucose.</span></span>
<span id="cb2-491"><a href="#cb2-491" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Hadhb"</span>,  <span class="co"># Hydroxyacyl-CoA dehydrogenase/3-ketoacyl-CoA thiolase/enoyl-CoA hydratase, beta subunit: A mitochondrial enzyme that participates in fatty acid beta-oxidation, playing a role in multiple steps of the process.</span></span>
<span id="cb2-492"><a href="#cb2-492" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pygb"</span>,  <span class="co"># Glycogen phosphorylase, brain form: An enzyme that breaks down glycogen into glucose-1-phosphate, which is then converted to glucose-6-phosphate for entry into glycolysis. This enzyme plays a critical role in energy production, particularly during periods of increased energy demand.</span></span>
<span id="cb2-493"><a href="#cb2-493" aria-hidden="true" tabindex="-1"></a>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="parameters" class="level3">
<h3 class="anchored" data-anchor-id="parameters">parameters</h3>
<div class="cell" data-execution_count="3">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>bioproject <span class="op">=</span> <span class="st">"PRJNA604055"</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Distance Metric to Use</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># metric = snakemake.params["metric"]</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>metric <span class="op">=</span> <span class="st">"euclidean"</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co"># connectivity_model = snakemake.params["connectivity_model"]</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>connectivity_model <span class="op">=</span> <span class="st">"min_tree"</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>PLOTS_DIR <span class="op">=</span> os.path.join(<span class="st">"../output/figures/velocity-signalling/"</span>)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>os.makedirs(PLOTS_DIR, exist_ok<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>OUT_DIR <span class="op">=</span> os.path.join(<span class="st">"../data/velocity-signalling/"</span>)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>os.makedirs(OUT_DIR, exist_ok<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="co"># plotting params</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>pt_size <span class="op">=</span> <span class="dv">200</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>dnsty <span class="op">=</span> <span class="fl">0.8</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>plt_size <span class="op">=</span> [<span class="dv">5</span>, <span class="dv">5</span>]</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>SMALL_SIZE <span class="op">=</span> <span class="dv">7</span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>MEDIUM_SIZE <span class="op">=</span> <span class="dv">8</span></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>BIGGER_SIZE <span class="op">=</span> <span class="dv">12</span></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>plt.rc(<span class="st">"font"</span>, size<span class="op">=</span>SMALL_SIZE)  <span class="co"># controls default text sizes</span></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>plt.rc(<span class="st">"axes"</span>, titlesize<span class="op">=</span>SMALL_SIZE)  <span class="co"># fontsize of the axes title</span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>plt.rc(<span class="st">"axes"</span>, labelsize<span class="op">=</span>MEDIUM_SIZE)  <span class="co"># fontsize of the x and y labels</span></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>plt.rc(<span class="st">"xtick"</span>, labelsize<span class="op">=</span>SMALL_SIZE)  <span class="co"># fontsize of the tick labels</span></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>plt.rc(<span class="st">"ytick"</span>, labelsize<span class="op">=</span>SMALL_SIZE)  <span class="co"># fontsize of the tick labels</span></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>plt.rc(<span class="st">"legend"</span>, fontsize<span class="op">=</span>SMALL_SIZE)  <span class="co"># legend fontsize</span></span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>plt.rc(<span class="st">"figure"</span>, titlesize<span class="op">=</span>BIGGER_SIZE)  <span class="co"># fontsize of the figure title</span></span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>matplotlib.rcParams[<span class="st">"font.size"</span>] <span class="op">=</span> <span class="dv">11</span></span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>matplotlib.rcParams[<span class="st">"pdf.use14corefonts"</span>] <span class="op">=</span> <span class="va">True</span></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>matplotlib.rcParams[<span class="st">"pdf.fonttype"</span>] <span class="op">=</span> <span class="dv">42</span></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>matplotlib.rcParams[<span class="st">"ps.fonttype"</span>] <span class="op">=</span> <span class="dv">42</span></span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>matplotlib.rcParams[<span class="st">"font.family"</span>] <span class="op">=</span> <span class="st">"sans-serif"</span></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>matplotlib.rcParams[<span class="st">"font.sans-serif"</span>] <span class="op">=</span> [<span class="st">"Helvetica"</span>]</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>matplotlib.rcParams[<span class="st">"figure.max_open_warning"</span>] <span class="op">=</span> <span class="dv">20000</span></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>reseed <span class="op">=</span> <span class="dv">42</span></span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>random.seed(reseed)</span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>np.random.seed(reseed)</span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>random_state <span class="op">=</span> check_random_state(reseed)</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>verbose <span class="op">=</span> <span class="va">True</span></span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>sc.settings.verbosity <span class="op">=</span> <span class="dv">2</span>  <span class="co"># verbosity: errors (0), warnings (1), info (2), hints (3)</span></span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>scv.settings.set_figure_params(<span class="st">"scvelo"</span>)</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>sc.settings.figdir <span class="op">=</span> PLOTS_DIR</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>sc.settings.writedir <span class="op">=</span> OUT_DIR</span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>sc.settings.set_figure_params(</span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>    dpi<span class="op">=</span><span class="dv">120</span>,</span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>    dpi_save<span class="op">=</span><span class="dv">300</span>,</span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>    vector_friendly<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>    <span class="bu">format</span><span class="op">=</span><span class="st">"pdf"</span>,</span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>    transparent<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>    color_map<span class="op">=</span><span class="st">"gnuplot_r"</span>,</span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>sc.settings.autoshow <span class="op">=</span> <span class="va">False</span></span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>sc.settings.autosave <span class="op">=</span> <span class="va">True</span></span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a><span class="co"># k = int(snakemake.params["k"])</span></span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a>k <span class="op">=</span> <span class="dv">20</span></span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a>ks <span class="op">=</span> k <span class="op">+</span> <span class="dv">10</span></span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a>npcs <span class="op">=</span> <span class="dv">50</span></span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a>npcs_selected <span class="op">=</span> npcs <span class="op">-</span> <span class="dv">20</span></span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a>min_shared_umis <span class="op">=</span> <span class="dv">15</span></span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a>n_top_hvg <span class="op">=</span> <span class="dv">3000</span></span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a>cb_fpr <span class="op">=</span> <span class="fl">0.001</span></span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a>paga_threshold <span class="op">=</span> <span class="fl">1.0</span></span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a>spread <span class="op">=</span> <span class="dv">2</span></span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a>threads <span class="op">=</span> <span class="dv">80</span></span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a>svm_fit_strength <span class="op">=</span> <span class="dv">100</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="load-data" class="level3">
<h3 class="anchored" data-anchor-id="load-data">Load data</h3>
<div class="cell" data-execution_count="4">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>housekeeping <span class="op">=</span> <span class="ss">f"/data/data/housekeeping_mouse.tsv"</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>hk_genes1 <span class="op">=</span> []</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="cf">with</span> <span class="bu">open</span>(housekeeping) <span class="im">as</span> <span class="bu">file</span>:</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>    <span class="cf">while</span> hk_genes :<span class="op">=</span> <span class="bu">file</span>.readline():</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>        hk_genes1.append(hk_genes.rstrip())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>adata <span class="op">=</span> sc.read_h5ad(</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>    <span class="ss">f"../data/class_cello/PRJNA604055-astrocytes_dataset-</span><span class="sc">{</span>cb_fpr<span class="sc">}</span><span class="ss">-initial_selection.h5ad"</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>adata.obs[<span class="st">"cell_id"</span>] <span class="op">=</span> adata.obs_names</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>alldata <span class="op">=</span> sc.read_h5ad(</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>    <span class="ss">f"../data/class_cello/PRJNA604055-whole_dataset-</span><span class="sc">{</span>cb_fpr<span class="sc">}</span><span class="ss">-cello_annotation.h5ad"</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>alldata <span class="op">=</span> alldata[alldata.obs[<span class="st">"ora_celltype"</span>].isin([<span class="st">"Astrocytes"</span>, <span class="st">"Neurons"</span>])]</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>adata_norm <span class="op">=</span> alldata[alldata.obs[<span class="st">"condit"</span>] <span class="op">==</span> <span class="dv">0</span>]</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>adata_hfd <span class="op">=</span> alldata[alldata.obs[<span class="st">"condit"</span>] <span class="op">==</span> <span class="dv">1</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="signalling-analysis" class="level3">
<h3 class="anchored" data-anchor-id="signalling-analysis">Signalling analysis</h3>
<section id="normal-diet" class="level4">
<h4 class="anchored" data-anchor-id="normal-diet">Normal diet</h4>
<div class="cell" data-execution_count="6">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>adata_norm.obs[<span class="st">"k_tree"</span>] <span class="op">=</span> adata_norm.obs[<span class="st">"k_tree"</span>].astype(<span class="st">"category"</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>adata_norm.raw.X[:<span class="dv">10</span>, :<span class="dv">10</span>].todense()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="6">
<pre><code>matrix([[ 2.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  1.,  1.],
        [ 1.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  1.,  0.],
        [ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  1.,  0.],
        [26.,  0.,  1.,  0.,  0.,  0.,  0.,  0.,  2.,  0.],
        [16.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.],
        [12.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  1.,  0.],
        [16.,  0.,  1.,  0.,  0.,  0.,  0.,  0.,  0.,  0.],
        [ 3.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  1.],
        [ 3.,  1.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.],
        [26.,  1.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  1.]], dtype=float32)</code></pre>
</div>
</div>
<p>subset groups of interest</p>
<div class="cell" data-execution_count="7">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>sc.pl.umap(</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>    adata_norm,</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[<span class="st">"k_tree"</span>],</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span>pt_size,</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    legend_loc<span class="op">=</span><span class="st">"on data"</span>,</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-8-output-1.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="8">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>sc.pl.umap(</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>    adata_norm,</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Rbfox3"</span>,</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Pomc"</span>,</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Agrp"</span>,</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy"</span>,</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy1r"</span>,</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy2r"</span>,</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Gfap"</span>,</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Lxn"</span>,</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Slc1a3"</span>,</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Htra1"</span>,</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Mfn2"</span>,</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Fos"</span>,</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>    ],</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.6</span>,</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span>pt_size,</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-9-output-1.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="9">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>sc.pl.embedding(</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>    adata_norm,</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"X_pacmap"</span>,</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Rbfox3"</span>,</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Pomc"</span>,</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Agrp"</span>,</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy"</span>,</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy1r"</span>,</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy2r"</span>,</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Gfap"</span>,</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Lxn"</span>,</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Slc1a3"</span>,</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Htra1"</span>,</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Mfn2"</span>,</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Fos"</span>,</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>    ],</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.6</span>,</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>    add_outline<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>    legend_fontoutline<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"nipy_spectral"</span>,</span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-pacmap-int-normal-diet.pdf"</span>,</span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../output/figures/velocity-signalling/X_pacmap-pacmap-int-normal-diet.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-10-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="10">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>sc.pl.embedding(</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>    adata_norm,</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"X_pacmap"</span>,</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> genes_embed <span class="cf">if</span> g <span class="kw">in</span> adata_norm.var_names],</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.6</span>,</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>    add_outline<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>    legend_fontoutline<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"nipy_spectral"</span>,</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-pacmap-adgene-normal-diet.pdf"</span>,</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../output/figures/velocity-signalling/X_pacmap-pacmap-adgene-normal-diet.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-11-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="11">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="cf">with</span> matplotlib.rc_context({<span class="st">"figure.figsize"</span>: (<span class="fl">5.5</span>, <span class="dv">4</span>)}):</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>    sc.pl.violin(</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>        adata_norm,</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>        [<span class="st">"Pomc"</span>, <span class="st">"Agrp"</span>, <span class="st">"Npy"</span>, <span class="st">"Lxn"</span>, <span class="st">"Gfap"</span>, <span class="st">"Fos"</span>],</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>        groupby<span class="op">=</span><span class="st">"k_tree"</span>,</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>        stripplot<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>        frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>        palette<span class="op">=</span><span class="st">"nipy_spectral"</span>,</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>        show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>        save<span class="op">=</span><span class="st">"-violin-adgene-normal-diet.pdf"</span>,</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../output/figures/velocity-signalling/violin-violin-adgene-normal-diet.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-12-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="12">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>pomc_neurons <span class="op">=</span> adata_norm[</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>    (</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>        (adata_norm[:, <span class="st">"Pomc"</span>].to_df() <span class="op">&gt;</span> <span class="dv">20</span>).stack().values</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>        <span class="op">&amp;</span> (adata_norm[:, <span class="st">"Rbfox3"</span>].to_df() <span class="op">&gt;</span> <span class="dv">10</span>).stack().values</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>    <span class="op">&amp;</span> (adata_norm.obs[<span class="st">"ora_celltype"</span>].isin([<span class="st">"Neurons"</span>])),</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>    :,</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>npy_neurons <span class="op">=</span> adata_norm[</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>    (</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>        (</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>            (adata_norm[:, <span class="st">"Agrp"</span>].to_df() <span class="op">&gt;</span> <span class="dv">5</span>).stack().values</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>            <span class="op">|</span> (adata_norm[:, <span class="st">"Npy"</span>].to_df() <span class="op">&gt;</span> <span class="dv">5</span>).stack().values</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>        <span class="op">&amp;</span> (adata_norm[:, <span class="st">"Rbfox3"</span>].to_df() <span class="op">&gt;</span> <span class="dv">5</span>).stack().values</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>    <span class="op">&amp;</span> (adata_norm.obs[<span class="st">"ora_celltype"</span>].isin([<span class="st">"Neurons"</span>])),</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>    :,</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>lxn_pos_astro <span class="op">=</span> adata_norm[</span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>    (</span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>        (adata_norm[:, <span class="st">"Lxn"</span>].to_df() <span class="op">&gt;</span> <span class="dv">0</span>).stack().values</span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>        <span class="op">&amp;</span> (adata_norm[:, <span class="st">"Rbfox3"</span>].to_df() <span class="op">&lt;</span> <span class="dv">5</span>).stack().values</span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>    <span class="op">&amp;</span> (adata_norm.obs_names.isin(adata.obs_names)),</span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>    :,</span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>lxn_neg_astro <span class="op">=</span> adata_norm[</span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>    (</span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>        (adata_norm[:, <span class="st">"Lxn"</span>].to_df() <span class="op">==</span> <span class="dv">0</span>).stack().values</span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>        <span class="op">&amp;</span> (adata_norm[:, <span class="st">"Rbfox3"</span>].to_df() <span class="op">&lt;</span> <span class="dv">5</span>).stack().values</span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>    <span class="op">&amp;</span> (adata_norm.obs_names.isin(adata.obs_names)),</span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>    :,</span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="13">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>adata_norm.obs[<span class="st">"goi"</span>] <span class="op">=</span> adata_norm.obs[<span class="st">"k_tree"</span>]</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>adata_norm.obs[<span class="st">"goi"</span>] <span class="op">=</span> adata_norm.obs[<span class="st">"goi"</span>].cat.add_categories(<span class="st">"POMC neurons"</span>)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>adata_norm.obs.loc[</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>    <span class="kw">lambda</span> df: df.index.isin(pomc_neurons.obs.index), <span class="st">"goi"</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>] <span class="op">=</span> <span class="st">"POMC neurons"</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>adata_norm.obs[<span class="st">"goi"</span>] <span class="op">=</span> adata_norm.obs[<span class="st">"goi"</span>].cat.add_categories(<span class="st">"AGRP/NPY neurons"</span>)</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>adata_norm.obs.loc[</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>    <span class="kw">lambda</span> df: df.index.isin(npy_neurons.obs.index), <span class="st">"goi"</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>] <span class="op">=</span> <span class="st">"AGRP/NPY neurons"</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>adata_norm.obs[<span class="st">"goi"</span>] <span class="op">=</span> adata_norm.obs[<span class="st">"goi"</span>].cat.add_categories(<span class="st">"Astro_LXN+"</span>)</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>adata_norm.obs.loc[</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>    <span class="kw">lambda</span> df: df.index.isin(lxn_pos_astro.obs.index), <span class="st">"goi"</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>] <span class="op">=</span> <span class="st">"Astro_LXN+"</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>adata_norm.obs[<span class="st">"goi"</span>] <span class="op">=</span> adata_norm.obs[<span class="st">"goi"</span>].cat.add_categories(<span class="st">"Astro_LXN-"</span>)</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>adata_norm.obs.loc[</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>    <span class="kw">lambda</span> df: df.index.isin(lxn_neg_astro.obs.index), <span class="st">"goi"</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>] <span class="op">=</span> <span class="st">"Astro_LXN-"</span></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>adata_norm <span class="op">=</span> adata_norm[</span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>    adata_norm.obs[<span class="st">"goi"</span>].isin(</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>        [<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>, <span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>]</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>adata_norm.obs[<span class="st">"goi"</span>].cat.remove_unused_categories(inplace<span class="op">=</span><span class="va">True</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="14">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>sc.pl.embedding(</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>    adata_norm,</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"X_pacmap"</span>,</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Rbfox3"</span>,</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Pomc"</span>,</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Agrp"</span>,</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy"</span>,</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy1r"</span>,</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy2r"</span>,</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Gfap"</span>,</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Lxn"</span>,</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Slc1a3"</span>,</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Htra1"</span>,</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Mfn2"</span>,</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Fos"</span>,</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>    ],</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.6</span>,</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>    add_outline<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>    legend_fontoutline<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"nipy_spectral"</span>,</span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-pacmap-subset-int-normal-diet.pdf"</span>,</span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../output/figures/velocity-signalling/X_pacmap-pacmap-subset-int-normal-diet.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-15-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="15">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>sc.pl.embedding(</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>    adata_norm,</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"X_pacmap"</span>,</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[<span class="st">"goi"</span>],</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.6</span>,</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>    add_outline<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>    legend_fontoutline<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"nipy_spectral"</span>,</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-pacmap-subset-goi-normal-diet.pdf"</span>,</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../output/figures/velocity-signalling/X_pacmap-pacmap-subset-goi-normal-diet.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-16-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="16">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>sc.pl.embedding(</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>    adata_norm,</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"X_pacmap"</span>,</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> genes_embed <span class="cf">if</span> g <span class="kw">in</span> adata_norm.var_names],</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.6</span>,</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>    add_outline<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>    legend_fontoutline<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"nipy_spectral"</span>,</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-pacmap-subset-adgene-normal-diet.pdf"</span>,</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../output/figures/velocity-signalling/X_pacmap-pacmap-subset-adgene-normal-diet.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-17-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="17">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>sc.pp.normalize_total(adata_norm, target_sum<span class="op">=</span><span class="fl">1e4</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>sc.pp.log1p(adata_norm)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>adata_norm.raw <span class="op">=</span> adata_norm</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>li.mt.rank_aggregate(</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>    adata_norm,</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>    groupby<span class="op">=</span><span class="st">"goi"</span>,</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>    de_method<span class="op">=</span><span class="st">"logreg"</span>,</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>    expr_prop<span class="op">=</span><span class="dv">0</span>,</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>    min_cells<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>    return_all_lrs<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>    verbose<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>    resource_name<span class="op">=</span><span class="st">"mouseconsensus"</span>,</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>    n_perms<span class="op">=</span><span class="dv">100000</span>,</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>    seed<span class="op">=</span>reseed,</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>normalizing counts per cell</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    finished (0:00:00)</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>... as `zero_center=True`, sparse input is densified and may lead to large memory consumption</code></pre>
</div>
</div>
<div class="cell" data-execution_count="18">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>adata_norm.uns[<span class="st">"liana_res"</span>].to_csv(<span class="st">"arc_astro-with_npy_pomc-liana-res.csv"</span>)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>adata_norm.uns[<span class="st">"liana_res"</span>].head()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="18">

<div>

<table class="dataframe table table-sm table-striped">
  <thead>
    <tr>
      <th></th>
      <th>source</th>
      <th>target</th>
      <th>ligand_complex</th>
      <th>receptor_complex</th>
      <th>lr_means</th>
      <th>cellphone_pvals</th>
      <th>expr_prod</th>
      <th>scaled_weight</th>
      <th>lr_logfc</th>
      <th>spec_weight</th>
      <th>lrscore</th>
      <th>lr_probs</th>
      <th>cellchat_pvals</th>
      <th>steady_rank</th>
      <th>specificity_rank</th>
      <th>magnitude_rank</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>20941</th>
      <td>Astro_LXN-</td>
      <td>AGRP/NPY neurons</td>
      <td>Nrxn1</td>
      <td>Nlgn1</td>
      <td>3.715075</td>
      <td>0.00055</td>
      <td>13.451365</td>
      <td>0.257168</td>
      <td>0.329536</td>
      <td>0.072446</td>
      <td>0.954388</td>
      <td>0.315478</td>
      <td>0.00440</td>
      <td>0.001446</td>
      <td>0.022157</td>
      <td>1.628633e-11</td>
    </tr>
    <tr>
      <th>37593</th>
      <td>AGRP/NPY neurons</td>
      <td>Astro_LXN-</td>
      <td>Nlgn1</td>
      <td>Nrxn1</td>
      <td>3.715075</td>
      <td>0.00055</td>
      <td>13.451365</td>
      <td>0.257168</td>
      <td>0.329536</td>
      <td>0.072446</td>
      <td>0.954388</td>
      <td>0.315478</td>
      <td>0.00440</td>
      <td>0.001446</td>
      <td>0.022157</td>
      <td>1.628633e-11</td>
    </tr>
    <tr>
      <th>26561</th>
      <td>AGRP/NPY neurons</td>
      <td>Astro_LXN+</td>
      <td>Nlgn1</td>
      <td>Nrxn1</td>
      <td>3.701165</td>
      <td>0.00813</td>
      <td>13.364478</td>
      <td>0.227609</td>
      <td>0.110424</td>
      <td>0.071978</td>
      <td>0.954247</td>
      <td>0.313704</td>
      <td>0.02264</td>
      <td>0.003374</td>
      <td>0.027768</td>
      <td>9.497614e-11</td>
    </tr>
    <tr>
      <th>18183</th>
      <td>Astro_LXN+</td>
      <td>AGRP/NPY neurons</td>
      <td>Nrxn1</td>
      <td>Nlgn1</td>
      <td>3.701165</td>
      <td>0.00813</td>
      <td>13.364478</td>
      <td>0.227609</td>
      <td>0.110424</td>
      <td>0.071978</td>
      <td>0.954247</td>
      <td>0.313704</td>
      <td>0.02264</td>
      <td>0.003374</td>
      <td>0.027768</td>
      <td>9.497614e-11</td>
    </tr>
    <tr>
      <th>43109</th>
      <td>Astro_LXN-</td>
      <td>Astro_LXN-</td>
      <td>Nlgn1</td>
      <td>Nrxn1</td>
      <td>3.675779</td>
      <td>0.00000</td>
      <td>13.112859</td>
      <td>0.181901</td>
      <td>0.336546</td>
      <td>0.070623</td>
      <td>0.953830</td>
      <td>0.308481</td>
      <td>0.00000</td>
      <td>0.001025</td>
      <td>0.009624</td>
      <td>2.862140e-10</td>
    </tr>
  </tbody>
</table>
</div>
</div>
</div>
<div class="cell" data-execution_count="19">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>n2a_norm <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>    adata<span class="op">=</span>adata_norm,</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>    top_n<span class="op">=</span><span class="dv">20</span>,</span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"lr_logfc"</span>,</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>    n2a_norm</span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a>n2a_norm.save(filename<span class="op">=</span><span class="st">"liana-dotplot-logfc-normal-diet-n2a.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family ['Dejavu Sans'] not found. Falling back to Helvetica.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family ['Dejavu Sans'] not found. Falling back to Helvetica.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family ['Dejavu Sans'] not found. Falling back to Helvetica.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family ['Dejavu Sans'] not found. Falling back to Helvetica.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family ['Dejavu Sans'] not found. Falling back to Helvetica.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family ['Dejavu Sans'] not found. Falling back to Helvetica.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="20">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>n2a_norm <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>    adata<span class="op">=</span>adata_norm,</span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb75-7"><a href="#cb75-7" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb75-8"><a href="#cb75-8" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb75-9"><a href="#cb75-9" aria-hidden="true" tabindex="-1"></a>    filterby<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb75-10"><a href="#cb75-10" aria-hidden="true" tabindex="-1"></a>    filter_lambda<span class="op">=</span><span class="kw">lambda</span> x: x <span class="op">&lt;=</span> <span class="fl">0.01</span>,</span>
<span id="cb75-11"><a href="#cb75-11" aria-hidden="true" tabindex="-1"></a>    top_n<span class="op">=</span><span class="dv">20</span>,</span>
<span id="cb75-12"><a href="#cb75-12" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb75-13"><a href="#cb75-13" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb75-14"><a href="#cb75-14" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb75-15"><a href="#cb75-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb75-16"><a href="#cb75-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-17"><a href="#cb75-17" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb75-18"><a href="#cb75-18" aria-hidden="true" tabindex="-1"></a>    n2a_norm</span>
<span id="cb75-19"><a href="#cb75-19" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb75-20"><a href="#cb75-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb75-21"><a href="#cb75-21" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb75-22"><a href="#cb75-22" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb75-23"><a href="#cb75-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb75-24"><a href="#cb75-24" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb75-25"><a href="#cb75-25" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb75-26"><a href="#cb75-26" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb75-27"><a href="#cb75-27" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb75-28"><a href="#cb75-28" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb75-29"><a href="#cb75-29" aria-hidden="true" tabindex="-1"></a>n2a_norm.save(filename<span class="op">=</span><span class="st">"liana-dotplot-specificity_rank-normal-diet-n2a.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="21">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a>n2a_norm <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>    adata<span class="op">=</span>adata_norm,</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb114-9"><a href="#cb114-9" aria-hidden="true" tabindex="-1"></a>    filterby<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb114-10"><a href="#cb114-10" aria-hidden="true" tabindex="-1"></a>    filter_lambda<span class="op">=</span><span class="kw">lambda</span> x: x <span class="op">&lt;=</span> <span class="fl">0.01</span>,</span>
<span id="cb114-11"><a href="#cb114-11" aria-hidden="true" tabindex="-1"></a>    top_n<span class="op">=</span><span class="dv">20</span>,</span>
<span id="cb114-12"><a href="#cb114-12" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb114-13"><a href="#cb114-13" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb114-14"><a href="#cb114-14" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb114-15"><a href="#cb114-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb114-16"><a href="#cb114-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-17"><a href="#cb114-17" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb114-18"><a href="#cb114-18" aria-hidden="true" tabindex="-1"></a>    n2a_norm</span>
<span id="cb114-19"><a href="#cb114-19" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb114-20"><a href="#cb114-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb114-21"><a href="#cb114-21" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb114-22"><a href="#cb114-22" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb114-23"><a href="#cb114-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb114-24"><a href="#cb114-24" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb114-25"><a href="#cb114-25" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb114-26"><a href="#cb114-26" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb114-27"><a href="#cb114-27" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb114-28"><a href="#cb114-28" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb114-29"><a href="#cb114-29" aria-hidden="true" tabindex="-1"></a>n2a_norm.save(filename<span class="op">=</span><span class="st">"liana-dotplot-magnitude_rank-normal-diet-n2a.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="22">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb155"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="co"># First, filter the DataFrame</span></span>
<span id="cb155-2"><a href="#cb155-2" aria-hidden="true" tabindex="-1"></a>filtered_df <span class="op">=</span> adata_norm.uns[<span class="st">"liana_res"</span>][</span>
<span id="cb155-3"><a href="#cb155-3" aria-hidden="true" tabindex="-1"></a>    adata_norm.uns[<span class="st">"liana_res"</span>][<span class="st">"receptor_complex"</span>] <span class="op">==</span> <span class="st">"Insr"</span></span>
<span id="cb155-4"><a href="#cb155-4" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb155-5"><a href="#cb155-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-6"><a href="#cb155-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Then, pass the filtered DataFrame to the dotplot function</span></span>
<span id="cb155-7"><a href="#cb155-7" aria-hidden="true" tabindex="-1"></a>n2a_norm <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb155-8"><a href="#cb155-8" aria-hidden="true" tabindex="-1"></a>    liana_res<span class="op">=</span>filtered_df,</span>
<span id="cb155-9"><a href="#cb155-9" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb155-10"><a href="#cb155-10" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb155-11"><a href="#cb155-11" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb155-12"><a href="#cb155-12" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb155-13"><a href="#cb155-13" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb155-14"><a href="#cb155-14" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb155-15"><a href="#cb155-15" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"lr_logfc"</span>,</span>
<span id="cb155-16"><a href="#cb155-16" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb155-17"><a href="#cb155-17" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb155-18"><a href="#cb155-18" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb155-19"><a href="#cb155-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-20"><a href="#cb155-20" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb155-21"><a href="#cb155-21" aria-hidden="true" tabindex="-1"></a>    n2a_norm</span>
<span id="cb155-22"><a href="#cb155-22" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb155-23"><a href="#cb155-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb155-24"><a href="#cb155-24" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb155-25"><a href="#cb155-25" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb155-26"><a href="#cb155-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb155-27"><a href="#cb155-27" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb155-28"><a href="#cb155-28" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb155-29"><a href="#cb155-29" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb155-30"><a href="#cb155-30" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb155-31"><a href="#cb155-31" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb155-32"><a href="#cb155-32" aria-hidden="true" tabindex="-1"></a>n2a_norm.save(filename<span class="op">=</span><span class="st">"liana-dotplot-insr-logfc-normal-diet-n2a.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="23">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb180"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb180-1"><a href="#cb180-1" aria-hidden="true" tabindex="-1"></a>n2a_norm <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb180-2"><a href="#cb180-2" aria-hidden="true" tabindex="-1"></a>    liana_res<span class="op">=</span>filtered_df,</span>
<span id="cb180-3"><a href="#cb180-3" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb180-4"><a href="#cb180-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb180-5"><a href="#cb180-5" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb180-6"><a href="#cb180-6" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb180-7"><a href="#cb180-7" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb180-8"><a href="#cb180-8" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb180-9"><a href="#cb180-9" aria-hidden="true" tabindex="-1"></a>    filterby<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb180-10"><a href="#cb180-10" aria-hidden="true" tabindex="-1"></a>    filter_lambda<span class="op">=</span><span class="kw">lambda</span> x: x <span class="op">&lt;=</span> <span class="fl">0.01</span>,</span>
<span id="cb180-11"><a href="#cb180-11" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb180-12"><a href="#cb180-12" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb180-13"><a href="#cb180-13" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb180-14"><a href="#cb180-14" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb180-15"><a href="#cb180-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb180-16"><a href="#cb180-16" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb180-17"><a href="#cb180-17" aria-hidden="true" tabindex="-1"></a>    n2a_norm</span>
<span id="cb180-18"><a href="#cb180-18" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb180-19"><a href="#cb180-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb180-20"><a href="#cb180-20" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb180-21"><a href="#cb180-21" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb180-22"><a href="#cb180-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb180-23"><a href="#cb180-23" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb180-24"><a href="#cb180-24" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb180-25"><a href="#cb180-25" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb180-26"><a href="#cb180-26" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb180-27"><a href="#cb180-27" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb180-28"><a href="#cb180-28" aria-hidden="true" tabindex="-1"></a>n2a_norm.save(filename<span class="op">=</span><span class="st">"liana-dotplot-insr-specificity_rank-normal-diet-n2a.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="24">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb201"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb201-1"><a href="#cb201-1" aria-hidden="true" tabindex="-1"></a>n2a_norm <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb201-2"><a href="#cb201-2" aria-hidden="true" tabindex="-1"></a>    liana_res<span class="op">=</span>filtered_df,</span>
<span id="cb201-3"><a href="#cb201-3" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb201-4"><a href="#cb201-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb201-5"><a href="#cb201-5" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb201-6"><a href="#cb201-6" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb201-7"><a href="#cb201-7" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb201-8"><a href="#cb201-8" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb201-9"><a href="#cb201-9" aria-hidden="true" tabindex="-1"></a>    filterby<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb201-10"><a href="#cb201-10" aria-hidden="true" tabindex="-1"></a>    filter_lambda<span class="op">=</span><span class="kw">lambda</span> x: x <span class="op">&lt;=</span> <span class="fl">0.01</span>,</span>
<span id="cb201-11"><a href="#cb201-11" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb201-12"><a href="#cb201-12" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb201-13"><a href="#cb201-13" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb201-14"><a href="#cb201-14" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb201-15"><a href="#cb201-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb201-16"><a href="#cb201-16" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb201-17"><a href="#cb201-17" aria-hidden="true" tabindex="-1"></a>    n2a_norm</span>
<span id="cb201-18"><a href="#cb201-18" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb201-19"><a href="#cb201-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb201-20"><a href="#cb201-20" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb201-21"><a href="#cb201-21" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb201-22"><a href="#cb201-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb201-23"><a href="#cb201-23" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb201-24"><a href="#cb201-24" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb201-25"><a href="#cb201-25" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb201-26"><a href="#cb201-26" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb201-27"><a href="#cb201-27" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb201-28"><a href="#cb201-28" aria-hidden="true" tabindex="-1"></a>n2a_norm.save(filename<span class="op">=</span><span class="st">"liana-dotplot-insr-magnitude_rank-normal-diet-n2a.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="25">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb221"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb221-1"><a href="#cb221-1" aria-hidden="true" tabindex="-1"></a>a2n_norm <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb221-2"><a href="#cb221-2" aria-hidden="true" tabindex="-1"></a>    adata<span class="op">=</span>adata_norm,</span>
<span id="cb221-3"><a href="#cb221-3" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb221-4"><a href="#cb221-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb221-5"><a href="#cb221-5" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb221-6"><a href="#cb221-6" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb221-7"><a href="#cb221-7" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb221-8"><a href="#cb221-8" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb221-9"><a href="#cb221-9" aria-hidden="true" tabindex="-1"></a>    top_n<span class="op">=</span><span class="dv">20</span>,</span>
<span id="cb221-10"><a href="#cb221-10" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"lr_logfc"</span>,</span>
<span id="cb221-11"><a href="#cb221-11" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb221-12"><a href="#cb221-12" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb221-13"><a href="#cb221-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb221-14"><a href="#cb221-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb221-15"><a href="#cb221-15" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb221-16"><a href="#cb221-16" aria-hidden="true" tabindex="-1"></a>    a2n_norm</span>
<span id="cb221-17"><a href="#cb221-17" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb221-18"><a href="#cb221-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb221-19"><a href="#cb221-19" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb221-20"><a href="#cb221-20" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb221-21"><a href="#cb221-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb221-22"><a href="#cb221-22" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb221-23"><a href="#cb221-23" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb221-24"><a href="#cb221-24" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb221-25"><a href="#cb221-25" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb221-26"><a href="#cb221-26" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb221-27"><a href="#cb221-27" aria-hidden="true" tabindex="-1"></a>a2n_norm.save(filename<span class="op">=</span><span class="st">"liana-dotplot-logfc-normal-diet-a2n.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="26">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb262"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb262-1"><a href="#cb262-1" aria-hidden="true" tabindex="-1"></a>a2n_norm <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb262-2"><a href="#cb262-2" aria-hidden="true" tabindex="-1"></a>    adata<span class="op">=</span>adata_norm,</span>
<span id="cb262-3"><a href="#cb262-3" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb262-4"><a href="#cb262-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb262-5"><a href="#cb262-5" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb262-6"><a href="#cb262-6" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb262-7"><a href="#cb262-7" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb262-8"><a href="#cb262-8" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb262-9"><a href="#cb262-9" aria-hidden="true" tabindex="-1"></a>    filterby<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb262-10"><a href="#cb262-10" aria-hidden="true" tabindex="-1"></a>    filter_lambda<span class="op">=</span><span class="kw">lambda</span> x: x <span class="op">&lt;=</span> <span class="fl">0.01</span>,</span>
<span id="cb262-11"><a href="#cb262-11" aria-hidden="true" tabindex="-1"></a>    top_n<span class="op">=</span><span class="dv">20</span>,</span>
<span id="cb262-12"><a href="#cb262-12" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb262-13"><a href="#cb262-13" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb262-14"><a href="#cb262-14" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb262-15"><a href="#cb262-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb262-16"><a href="#cb262-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb262-17"><a href="#cb262-17" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb262-18"><a href="#cb262-18" aria-hidden="true" tabindex="-1"></a>    a2n_norm</span>
<span id="cb262-19"><a href="#cb262-19" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb262-20"><a href="#cb262-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb262-21"><a href="#cb262-21" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb262-22"><a href="#cb262-22" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb262-23"><a href="#cb262-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb262-24"><a href="#cb262-24" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb262-25"><a href="#cb262-25" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb262-26"><a href="#cb262-26" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb262-27"><a href="#cb262-27" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb262-28"><a href="#cb262-28" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb262-29"><a href="#cb262-29" aria-hidden="true" tabindex="-1"></a>a2n_norm.save(filename<span class="op">=</span><span class="st">"liana-dotplot-specificity_rank-normal-diet-a2n.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="27">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb302"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb302-1"><a href="#cb302-1" aria-hidden="true" tabindex="-1"></a>a2n_norm <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb302-2"><a href="#cb302-2" aria-hidden="true" tabindex="-1"></a>    adata<span class="op">=</span>adata_norm,</span>
<span id="cb302-3"><a href="#cb302-3" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb302-4"><a href="#cb302-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb302-5"><a href="#cb302-5" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb302-6"><a href="#cb302-6" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb302-7"><a href="#cb302-7" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb302-8"><a href="#cb302-8" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb302-9"><a href="#cb302-9" aria-hidden="true" tabindex="-1"></a>    filterby<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb302-10"><a href="#cb302-10" aria-hidden="true" tabindex="-1"></a>    filter_lambda<span class="op">=</span><span class="kw">lambda</span> x: x <span class="op">&lt;=</span> <span class="fl">0.01</span>,</span>
<span id="cb302-11"><a href="#cb302-11" aria-hidden="true" tabindex="-1"></a>    top_n<span class="op">=</span><span class="dv">20</span>,</span>
<span id="cb302-12"><a href="#cb302-12" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb302-13"><a href="#cb302-13" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb302-14"><a href="#cb302-14" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb302-15"><a href="#cb302-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb302-16"><a href="#cb302-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb302-17"><a href="#cb302-17" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb302-18"><a href="#cb302-18" aria-hidden="true" tabindex="-1"></a>    a2n_norm</span>
<span id="cb302-19"><a href="#cb302-19" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb302-20"><a href="#cb302-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb302-21"><a href="#cb302-21" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb302-22"><a href="#cb302-22" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb302-23"><a href="#cb302-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb302-24"><a href="#cb302-24" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb302-25"><a href="#cb302-25" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb302-26"><a href="#cb302-26" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb302-27"><a href="#cb302-27" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb302-28"><a href="#cb302-28" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb302-29"><a href="#cb302-29" aria-hidden="true" tabindex="-1"></a>a2n_norm.save(filename<span class="op">=</span><span class="st">"liana-dotplot-magnitude_rank-normal-diet-a2n.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<p>Save data</p>
<div class="cell" data-execution_count="28">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb343"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb343-1"><a href="#cb343-1" aria-hidden="true" tabindex="-1"></a>adata_norm.write(</span>
<span id="cb343-2"><a href="#cb343-2" aria-hidden="true" tabindex="-1"></a>    <span class="ss">f"/data/</span><span class="sc">{</span>bioproject<span class="sc">}</span><span class="ss">/data/velocity-signalling/</span><span class="sc">{</span>bioproject<span class="sc">}</span><span class="ss">-signalling-norm.h5ad"</span></span>
<span id="cb343-3"><a href="#cb343-3" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="high-fat-diet" class="level4">
<h4 class="anchored" data-anchor-id="high-fat-diet">High Fat Diet</h4>
<div class="cell" data-execution_count="29">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb344"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb344-1"><a href="#cb344-1" aria-hidden="true" tabindex="-1"></a>adata_hfd.obs[<span class="st">"k_tree"</span>] <span class="op">=</span> adata_hfd.obs[<span class="st">"k_tree"</span>].astype(<span class="st">"category"</span>)</span>
<span id="cb344-2"><a href="#cb344-2" aria-hidden="true" tabindex="-1"></a>adata_hfd.raw.X</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="29">
<pre><code>&lt;8257x25600 sparse matrix of type '&lt;class 'numpy.float32'&gt;'
    with 39315361 stored elements in Compressed Sparse Column format&gt;</code></pre>
</div>
</div>
<p>subset groups of interest</p>
<div class="cell" data-execution_count="30">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb346"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb346-1"><a href="#cb346-1" aria-hidden="true" tabindex="-1"></a>sc.pl.umap(</span>
<span id="cb346-2"><a href="#cb346-2" aria-hidden="true" tabindex="-1"></a>    adata_hfd,</span>
<span id="cb346-3"><a href="#cb346-3" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[<span class="st">"k_tree"</span>],</span>
<span id="cb346-4"><a href="#cb346-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span>pt_size,</span>
<span id="cb346-5"><a href="#cb346-5" aria-hidden="true" tabindex="-1"></a>    legend_loc<span class="op">=</span><span class="st">"on data"</span>,</span>
<span id="cb346-6"><a href="#cb346-6" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb346-7"><a href="#cb346-7" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb346-8"><a href="#cb346-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-31-output-1.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="31">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb347"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb347-1"><a href="#cb347-1" aria-hidden="true" tabindex="-1"></a>sc.pl.umap(</span>
<span id="cb347-2"><a href="#cb347-2" aria-hidden="true" tabindex="-1"></a>    adata_hfd,</span>
<span id="cb347-3"><a href="#cb347-3" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[</span>
<span id="cb347-4"><a href="#cb347-4" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Rbfox3"</span>,</span>
<span id="cb347-5"><a href="#cb347-5" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Pomc"</span>,</span>
<span id="cb347-6"><a href="#cb347-6" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Agrp"</span>,</span>
<span id="cb347-7"><a href="#cb347-7" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy"</span>,</span>
<span id="cb347-8"><a href="#cb347-8" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy1r"</span>,</span>
<span id="cb347-9"><a href="#cb347-9" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy2r"</span>,</span>
<span id="cb347-10"><a href="#cb347-10" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Gfap"</span>,</span>
<span id="cb347-11"><a href="#cb347-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Lxn"</span>,</span>
<span id="cb347-12"><a href="#cb347-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Slc1a3"</span>,</span>
<span id="cb347-13"><a href="#cb347-13" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Htra1"</span>,</span>
<span id="cb347-14"><a href="#cb347-14" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Mfn2"</span>,</span>
<span id="cb347-15"><a href="#cb347-15" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Fos"</span>,</span>
<span id="cb347-16"><a href="#cb347-16" aria-hidden="true" tabindex="-1"></a>    ],</span>
<span id="cb347-17"><a href="#cb347-17" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.6</span>,</span>
<span id="cb347-18"><a href="#cb347-18" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span>pt_size,</span>
<span id="cb347-19"><a href="#cb347-19" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb347-20"><a href="#cb347-20" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb347-21"><a href="#cb347-21" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-32-output-1.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="32">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb348"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb348-1"><a href="#cb348-1" aria-hidden="true" tabindex="-1"></a>sc.pl.embedding(</span>
<span id="cb348-2"><a href="#cb348-2" aria-hidden="true" tabindex="-1"></a>    adata_hfd,</span>
<span id="cb348-3"><a href="#cb348-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"X_pacmap"</span>,</span>
<span id="cb348-4"><a href="#cb348-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[</span>
<span id="cb348-5"><a href="#cb348-5" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Rbfox3"</span>,</span>
<span id="cb348-6"><a href="#cb348-6" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Pomc"</span>,</span>
<span id="cb348-7"><a href="#cb348-7" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Agrp"</span>,</span>
<span id="cb348-8"><a href="#cb348-8" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy"</span>,</span>
<span id="cb348-9"><a href="#cb348-9" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy1r"</span>,</span>
<span id="cb348-10"><a href="#cb348-10" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy2r"</span>,</span>
<span id="cb348-11"><a href="#cb348-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Gfap"</span>,</span>
<span id="cb348-12"><a href="#cb348-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Lxn"</span>,</span>
<span id="cb348-13"><a href="#cb348-13" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Slc1a3"</span>,</span>
<span id="cb348-14"><a href="#cb348-14" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Htra1"</span>,</span>
<span id="cb348-15"><a href="#cb348-15" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Mfn2"</span>,</span>
<span id="cb348-16"><a href="#cb348-16" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Fos"</span>,</span>
<span id="cb348-17"><a href="#cb348-17" aria-hidden="true" tabindex="-1"></a>    ],</span>
<span id="cb348-18"><a href="#cb348-18" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.6</span>,</span>
<span id="cb348-19"><a href="#cb348-19" aria-hidden="true" tabindex="-1"></a>    add_outline<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb348-20"><a href="#cb348-20" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb348-21"><a href="#cb348-21" aria-hidden="true" tabindex="-1"></a>    legend_fontoutline<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb348-22"><a href="#cb348-22" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb348-23"><a href="#cb348-23" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"nipy_spectral"</span>,</span>
<span id="cb348-24"><a href="#cb348-24" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb348-25"><a href="#cb348-25" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-pacmap-int-HF-diet.pdf"</span>,</span>
<span id="cb348-26"><a href="#cb348-26" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../output/figures/velocity-signalling/X_pacmap-pacmap-int-HF-diet.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-33-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="33">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb350"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb350-1"><a href="#cb350-1" aria-hidden="true" tabindex="-1"></a>sc.pl.embedding(</span>
<span id="cb350-2"><a href="#cb350-2" aria-hidden="true" tabindex="-1"></a>    adata_hfd,</span>
<span id="cb350-3"><a href="#cb350-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"X_pacmap"</span>,</span>
<span id="cb350-4"><a href="#cb350-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> genes_embed <span class="cf">if</span> g <span class="kw">in</span> adata_hfd.var_names],</span>
<span id="cb350-5"><a href="#cb350-5" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.6</span>,</span>
<span id="cb350-6"><a href="#cb350-6" aria-hidden="true" tabindex="-1"></a>    add_outline<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb350-7"><a href="#cb350-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb350-8"><a href="#cb350-8" aria-hidden="true" tabindex="-1"></a>    legend_fontoutline<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb350-9"><a href="#cb350-9" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb350-10"><a href="#cb350-10" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"nipy_spectral"</span>,</span>
<span id="cb350-11"><a href="#cb350-11" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb350-12"><a href="#cb350-12" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-pacmap-adgene-HF-diet.pdf"</span>,</span>
<span id="cb350-13"><a href="#cb350-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../output/figures/velocity-signalling/X_pacmap-pacmap-adgene-HF-diet.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-34-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="34">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb352"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb352-1"><a href="#cb352-1" aria-hidden="true" tabindex="-1"></a><span class="cf">with</span> matplotlib.rc_context({<span class="st">"figure.figsize"</span>: (<span class="fl">5.5</span>, <span class="dv">4</span>)}):</span>
<span id="cb352-2"><a href="#cb352-2" aria-hidden="true" tabindex="-1"></a>    sc.pl.violin(</span>
<span id="cb352-3"><a href="#cb352-3" aria-hidden="true" tabindex="-1"></a>        adata_hfd,</span>
<span id="cb352-4"><a href="#cb352-4" aria-hidden="true" tabindex="-1"></a>        [<span class="st">"Pomc"</span>, <span class="st">"Agrp"</span>, <span class="st">"Npy"</span>, <span class="st">"Lxn"</span>, <span class="st">"Gfap"</span>, <span class="st">"Fos"</span>],</span>
<span id="cb352-5"><a href="#cb352-5" aria-hidden="true" tabindex="-1"></a>        groupby<span class="op">=</span><span class="st">"k_tree"</span>,</span>
<span id="cb352-6"><a href="#cb352-6" aria-hidden="true" tabindex="-1"></a>        stripplot<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb352-7"><a href="#cb352-7" aria-hidden="true" tabindex="-1"></a>        frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb352-8"><a href="#cb352-8" aria-hidden="true" tabindex="-1"></a>        palette<span class="op">=</span><span class="st">"nipy_spectral"</span>,</span>
<span id="cb352-9"><a href="#cb352-9" aria-hidden="true" tabindex="-1"></a>        show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb352-10"><a href="#cb352-10" aria-hidden="true" tabindex="-1"></a>        save<span class="op">=</span><span class="st">"-violin-adgene-HF-diet.pdf"</span>,</span>
<span id="cb352-11"><a href="#cb352-11" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../output/figures/velocity-signalling/violin-violin-adgene-HF-diet.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-35-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="35">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb354"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb354-1"><a href="#cb354-1" aria-hidden="true" tabindex="-1"></a>pomc_neurons <span class="op">=</span> adata_hfd[</span>
<span id="cb354-2"><a href="#cb354-2" aria-hidden="true" tabindex="-1"></a>    (</span>
<span id="cb354-3"><a href="#cb354-3" aria-hidden="true" tabindex="-1"></a>        (adata_hfd[:, <span class="st">"Pomc"</span>].to_df() <span class="op">&gt;</span> <span class="dv">20</span>).stack().values</span>
<span id="cb354-4"><a href="#cb354-4" aria-hidden="true" tabindex="-1"></a>        <span class="op">&amp;</span> (adata_hfd[:, <span class="st">"Rbfox3"</span>].to_df() <span class="op">&gt;</span> <span class="dv">10</span>).stack().values</span>
<span id="cb354-5"><a href="#cb354-5" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb354-6"><a href="#cb354-6" aria-hidden="true" tabindex="-1"></a>    <span class="op">&amp;</span> (adata_hfd.obs[<span class="st">"ora_celltype"</span>].isin([<span class="st">"Neurons"</span>])),</span>
<span id="cb354-7"><a href="#cb354-7" aria-hidden="true" tabindex="-1"></a>    :,</span>
<span id="cb354-8"><a href="#cb354-8" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb354-9"><a href="#cb354-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb354-10"><a href="#cb354-10" aria-hidden="true" tabindex="-1"></a>npy_neurons <span class="op">=</span> adata_hfd[</span>
<span id="cb354-11"><a href="#cb354-11" aria-hidden="true" tabindex="-1"></a>    (</span>
<span id="cb354-12"><a href="#cb354-12" aria-hidden="true" tabindex="-1"></a>        (</span>
<span id="cb354-13"><a href="#cb354-13" aria-hidden="true" tabindex="-1"></a>            (adata_hfd[:, <span class="st">"Agrp"</span>].to_df() <span class="op">&gt;</span> <span class="dv">5</span>).stack().values</span>
<span id="cb354-14"><a href="#cb354-14" aria-hidden="true" tabindex="-1"></a>            <span class="op">|</span> (adata_hfd[:, <span class="st">"Npy"</span>].to_df() <span class="op">&gt;</span> <span class="dv">5</span>).stack().values</span>
<span id="cb354-15"><a href="#cb354-15" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb354-16"><a href="#cb354-16" aria-hidden="true" tabindex="-1"></a>        <span class="op">&amp;</span> (adata_hfd[:, <span class="st">"Rbfox3"</span>].to_df() <span class="op">&gt;</span> <span class="dv">5</span>).stack().values</span>
<span id="cb354-17"><a href="#cb354-17" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb354-18"><a href="#cb354-18" aria-hidden="true" tabindex="-1"></a>    <span class="op">&amp;</span> (adata_hfd.obs[<span class="st">"ora_celltype"</span>].isin([<span class="st">"Neurons"</span>])),</span>
<span id="cb354-19"><a href="#cb354-19" aria-hidden="true" tabindex="-1"></a>    :,</span>
<span id="cb354-20"><a href="#cb354-20" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb354-21"><a href="#cb354-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb354-22"><a href="#cb354-22" aria-hidden="true" tabindex="-1"></a>lxn_pos_astro <span class="op">=</span> adata_hfd[</span>
<span id="cb354-23"><a href="#cb354-23" aria-hidden="true" tabindex="-1"></a>    (</span>
<span id="cb354-24"><a href="#cb354-24" aria-hidden="true" tabindex="-1"></a>        (adata_hfd[:, <span class="st">"Lxn"</span>].to_df() <span class="op">&gt;</span> <span class="dv">0</span>).stack().values</span>
<span id="cb354-25"><a href="#cb354-25" aria-hidden="true" tabindex="-1"></a>        <span class="op">&amp;</span> (adata_hfd[:, <span class="st">"Rbfox3"</span>].to_df() <span class="op">&lt;</span> <span class="dv">5</span>).stack().values</span>
<span id="cb354-26"><a href="#cb354-26" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb354-27"><a href="#cb354-27" aria-hidden="true" tabindex="-1"></a>    <span class="op">&amp;</span> (adata_hfd.obs_names.isin(adata.obs_names)),</span>
<span id="cb354-28"><a href="#cb354-28" aria-hidden="true" tabindex="-1"></a>    :,</span>
<span id="cb354-29"><a href="#cb354-29" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb354-30"><a href="#cb354-30" aria-hidden="true" tabindex="-1"></a>lxn_neg_astro <span class="op">=</span> adata_hfd[</span>
<span id="cb354-31"><a href="#cb354-31" aria-hidden="true" tabindex="-1"></a>    (</span>
<span id="cb354-32"><a href="#cb354-32" aria-hidden="true" tabindex="-1"></a>        (adata_hfd[:, <span class="st">"Lxn"</span>].to_df() <span class="op">==</span> <span class="dv">0</span>).stack().values</span>
<span id="cb354-33"><a href="#cb354-33" aria-hidden="true" tabindex="-1"></a>        <span class="op">&amp;</span> (adata_hfd[:, <span class="st">"Rbfox3"</span>].to_df() <span class="op">&lt;</span> <span class="dv">5</span>).stack().values</span>
<span id="cb354-34"><a href="#cb354-34" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb354-35"><a href="#cb354-35" aria-hidden="true" tabindex="-1"></a>    <span class="op">&amp;</span> (adata_hfd.obs_names.isin(adata.obs_names)),</span>
<span id="cb354-36"><a href="#cb354-36" aria-hidden="true" tabindex="-1"></a>    :,</span>
<span id="cb354-37"><a href="#cb354-37" aria-hidden="true" tabindex="-1"></a>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="36">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb355"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb355-1"><a href="#cb355-1" aria-hidden="true" tabindex="-1"></a>adata_hfd.obs[<span class="st">"goi"</span>] <span class="op">=</span> adata_hfd.obs[<span class="st">"k_tree"</span>]</span>
<span id="cb355-2"><a href="#cb355-2" aria-hidden="true" tabindex="-1"></a>adata_hfd.obs[<span class="st">"goi"</span>] <span class="op">=</span> adata_hfd.obs[<span class="st">"goi"</span>].cat.add_categories(<span class="st">"POMC neurons"</span>)</span>
<span id="cb355-3"><a href="#cb355-3" aria-hidden="true" tabindex="-1"></a>adata_hfd.obs.loc[</span>
<span id="cb355-4"><a href="#cb355-4" aria-hidden="true" tabindex="-1"></a>    <span class="kw">lambda</span> df: df.index.isin(pomc_neurons.obs.index), <span class="st">"goi"</span></span>
<span id="cb355-5"><a href="#cb355-5" aria-hidden="true" tabindex="-1"></a>] <span class="op">=</span> <span class="st">"POMC neurons"</span></span>
<span id="cb355-6"><a href="#cb355-6" aria-hidden="true" tabindex="-1"></a>adata_hfd.obs[<span class="st">"goi"</span>] <span class="op">=</span> adata_hfd.obs[<span class="st">"goi"</span>].cat.add_categories(<span class="st">"AGRP/NPY neurons"</span>)</span>
<span id="cb355-7"><a href="#cb355-7" aria-hidden="true" tabindex="-1"></a>adata_hfd.obs.loc[</span>
<span id="cb355-8"><a href="#cb355-8" aria-hidden="true" tabindex="-1"></a>    <span class="kw">lambda</span> df: df.index.isin(npy_neurons.obs.index), <span class="st">"goi"</span></span>
<span id="cb355-9"><a href="#cb355-9" aria-hidden="true" tabindex="-1"></a>] <span class="op">=</span> <span class="st">"AGRP/NPY neurons"</span></span>
<span id="cb355-10"><a href="#cb355-10" aria-hidden="true" tabindex="-1"></a>adata_hfd.obs[<span class="st">"goi"</span>] <span class="op">=</span> adata_hfd.obs[<span class="st">"goi"</span>].cat.add_categories(<span class="st">"Astro_LXN+"</span>)</span>
<span id="cb355-11"><a href="#cb355-11" aria-hidden="true" tabindex="-1"></a>adata_hfd.obs.loc[</span>
<span id="cb355-12"><a href="#cb355-12" aria-hidden="true" tabindex="-1"></a>    <span class="kw">lambda</span> df: df.index.isin(lxn_pos_astro.obs.index), <span class="st">"goi"</span></span>
<span id="cb355-13"><a href="#cb355-13" aria-hidden="true" tabindex="-1"></a>] <span class="op">=</span> <span class="st">"Astro_LXN+"</span></span>
<span id="cb355-14"><a href="#cb355-14" aria-hidden="true" tabindex="-1"></a>adata_hfd.obs[<span class="st">"goi"</span>] <span class="op">=</span> adata_hfd.obs[<span class="st">"goi"</span>].cat.add_categories(<span class="st">"Astro_LXN-"</span>)</span>
<span id="cb355-15"><a href="#cb355-15" aria-hidden="true" tabindex="-1"></a>adata_hfd.obs.loc[</span>
<span id="cb355-16"><a href="#cb355-16" aria-hidden="true" tabindex="-1"></a>    <span class="kw">lambda</span> df: df.index.isin(lxn_neg_astro.obs.index), <span class="st">"goi"</span></span>
<span id="cb355-17"><a href="#cb355-17" aria-hidden="true" tabindex="-1"></a>] <span class="op">=</span> <span class="st">"Astro_LXN-"</span></span>
<span id="cb355-18"><a href="#cb355-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb355-19"><a href="#cb355-19" aria-hidden="true" tabindex="-1"></a>adata_hfd <span class="op">=</span> adata_hfd[</span>
<span id="cb355-20"><a href="#cb355-20" aria-hidden="true" tabindex="-1"></a>    adata_hfd.obs[<span class="st">"goi"</span>].isin(</span>
<span id="cb355-21"><a href="#cb355-21" aria-hidden="true" tabindex="-1"></a>        [<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>, <span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>]</span>
<span id="cb355-22"><a href="#cb355-22" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb355-23"><a href="#cb355-23" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb355-24"><a href="#cb355-24" aria-hidden="true" tabindex="-1"></a>adata_hfd.obs[<span class="st">"goi"</span>].cat.remove_unused_categories(inplace<span class="op">=</span><span class="va">True</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="37">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb356"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb356-1"><a href="#cb356-1" aria-hidden="true" tabindex="-1"></a>sc.pl.embedding(</span>
<span id="cb356-2"><a href="#cb356-2" aria-hidden="true" tabindex="-1"></a>    adata_hfd,</span>
<span id="cb356-3"><a href="#cb356-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"X_pacmap"</span>,</span>
<span id="cb356-4"><a href="#cb356-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[</span>
<span id="cb356-5"><a href="#cb356-5" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Rbfox3"</span>,</span>
<span id="cb356-6"><a href="#cb356-6" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Pomc"</span>,</span>
<span id="cb356-7"><a href="#cb356-7" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Agrp"</span>,</span>
<span id="cb356-8"><a href="#cb356-8" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy"</span>,</span>
<span id="cb356-9"><a href="#cb356-9" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy1r"</span>,</span>
<span id="cb356-10"><a href="#cb356-10" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Npy2r"</span>,</span>
<span id="cb356-11"><a href="#cb356-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Gfap"</span>,</span>
<span id="cb356-12"><a href="#cb356-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Lxn"</span>,</span>
<span id="cb356-13"><a href="#cb356-13" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Slc1a3"</span>,</span>
<span id="cb356-14"><a href="#cb356-14" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Htra1"</span>,</span>
<span id="cb356-15"><a href="#cb356-15" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Mfn2"</span>,</span>
<span id="cb356-16"><a href="#cb356-16" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Fos"</span>,</span>
<span id="cb356-17"><a href="#cb356-17" aria-hidden="true" tabindex="-1"></a>    ],</span>
<span id="cb356-18"><a href="#cb356-18" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.6</span>,</span>
<span id="cb356-19"><a href="#cb356-19" aria-hidden="true" tabindex="-1"></a>    add_outline<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb356-20"><a href="#cb356-20" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb356-21"><a href="#cb356-21" aria-hidden="true" tabindex="-1"></a>    legend_fontoutline<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb356-22"><a href="#cb356-22" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb356-23"><a href="#cb356-23" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"nipy_spectral"</span>,</span>
<span id="cb356-24"><a href="#cb356-24" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb356-25"><a href="#cb356-25" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-pacmap-subset-int-HF-diet.pdf"</span>,</span>
<span id="cb356-26"><a href="#cb356-26" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../output/figures/velocity-signalling/X_pacmap-pacmap-subset-int-HF-diet.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-38-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="38">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb358"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb358-1"><a href="#cb358-1" aria-hidden="true" tabindex="-1"></a>sc.pl.embedding(</span>
<span id="cb358-2"><a href="#cb358-2" aria-hidden="true" tabindex="-1"></a>    adata_hfd,</span>
<span id="cb358-3"><a href="#cb358-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"X_pacmap"</span>,</span>
<span id="cb358-4"><a href="#cb358-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> genes_embed <span class="cf">if</span> g <span class="kw">in</span> adata_hfd.var_names],</span>
<span id="cb358-5"><a href="#cb358-5" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.6</span>,</span>
<span id="cb358-6"><a href="#cb358-6" aria-hidden="true" tabindex="-1"></a>    add_outline<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb358-7"><a href="#cb358-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb358-8"><a href="#cb358-8" aria-hidden="true" tabindex="-1"></a>    legend_fontoutline<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb358-9"><a href="#cb358-9" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb358-10"><a href="#cb358-10" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"nipy_spectral"</span>,</span>
<span id="cb358-11"><a href="#cb358-11" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb358-12"><a href="#cb358-12" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-pacmap-subset-adgene-HF-diet.pdf"</span>,</span>
<span id="cb358-13"><a href="#cb358-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../output/figures/velocity-signalling/X_pacmap-pacmap-subset-adgene-HF-diet.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-39-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="39">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb360"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb360-1"><a href="#cb360-1" aria-hidden="true" tabindex="-1"></a>sc.pp.normalize_total(adata_hfd, target_sum<span class="op">=</span><span class="fl">1e4</span>)</span>
<span id="cb360-2"><a href="#cb360-2" aria-hidden="true" tabindex="-1"></a>sc.pp.log1p(adata_hfd)</span>
<span id="cb360-3"><a href="#cb360-3" aria-hidden="true" tabindex="-1"></a>adata_hfd.raw <span class="op">=</span> adata_hfd</span>
<span id="cb360-4"><a href="#cb360-4" aria-hidden="true" tabindex="-1"></a>li.mt.rank_aggregate(</span>
<span id="cb360-5"><a href="#cb360-5" aria-hidden="true" tabindex="-1"></a>    adata_hfd,</span>
<span id="cb360-6"><a href="#cb360-6" aria-hidden="true" tabindex="-1"></a>    groupby<span class="op">=</span><span class="st">"goi"</span>,</span>
<span id="cb360-7"><a href="#cb360-7" aria-hidden="true" tabindex="-1"></a>    de_method<span class="op">=</span><span class="st">"logreg"</span>,</span>
<span id="cb360-8"><a href="#cb360-8" aria-hidden="true" tabindex="-1"></a>    expr_prop<span class="op">=</span><span class="dv">0</span>,</span>
<span id="cb360-9"><a href="#cb360-9" aria-hidden="true" tabindex="-1"></a>    min_cells<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb360-10"><a href="#cb360-10" aria-hidden="true" tabindex="-1"></a>    return_all_lrs<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb360-11"><a href="#cb360-11" aria-hidden="true" tabindex="-1"></a>    verbose<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb360-12"><a href="#cb360-12" aria-hidden="true" tabindex="-1"></a>    resource_name<span class="op">=</span><span class="st">"mouseconsensus"</span>,</span>
<span id="cb360-13"><a href="#cb360-13" aria-hidden="true" tabindex="-1"></a>    n_perms<span class="op">=</span><span class="dv">100000</span>,</span>
<span id="cb360-14"><a href="#cb360-14" aria-hidden="true" tabindex="-1"></a>    seed<span class="op">=</span>reseed,</span>
<span id="cb360-15"><a href="#cb360-15" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>normalizing counts per cell</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    finished (0:00:00)</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>... as `zero_center=True`, sparse input is densified and may lead to large memory consumption</code></pre>
</div>
</div>
<div class="cell" data-execution_count="40">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb364"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb364-1"><a href="#cb364-1" aria-hidden="true" tabindex="-1"></a>adata_hfd.uns[<span class="st">"liana_res"</span>].to_csv(<span class="st">"hfd-arc_astro-with_npy_pomc-liana-res.csv"</span>)</span>
<span id="cb364-2"><a href="#cb364-2" aria-hidden="true" tabindex="-1"></a>adata_hfd.uns[<span class="st">"liana_res"</span>].head()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="40">

<div>

<table class="dataframe table table-sm table-striped">
  <thead>
    <tr>
      <th></th>
      <th>source</th>
      <th>target</th>
      <th>ligand_complex</th>
      <th>receptor_complex</th>
      <th>lr_means</th>
      <th>cellphone_pvals</th>
      <th>expr_prod</th>
      <th>scaled_weight</th>
      <th>lr_logfc</th>
      <th>spec_weight</th>
      <th>lrscore</th>
      <th>lr_probs</th>
      <th>cellchat_pvals</th>
      <th>steady_rank</th>
      <th>specificity_rank</th>
      <th>magnitude_rank</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>10222</th>
      <td>Astro_LXN-</td>
      <td>POMC neurons</td>
      <td>Nrxn1</td>
      <td>Nlgn1</td>
      <td>3.746233</td>
      <td>0.00569</td>
      <td>13.775460</td>
      <td>0.361921</td>
      <td>0.438989</td>
      <td>0.073659</td>
      <td>0.955289</td>
      <td>0.393909</td>
      <td>0.00666</td>
      <td>0.001437</td>
      <td>0.011857</td>
      <td>3.236681e-12</td>
    </tr>
    <tr>
      <th>35843</th>
      <td>POMC neurons</td>
      <td>Astro_LXN-</td>
      <td>Nlgn1</td>
      <td>Nrxn1</td>
      <td>3.746233</td>
      <td>0.00569</td>
      <td>13.775460</td>
      <td>0.361921</td>
      <td>0.438989</td>
      <td>0.073659</td>
      <td>0.955289</td>
      <td>0.393909</td>
      <td>0.00666</td>
      <td>0.001437</td>
      <td>0.011857</td>
      <td>3.236681e-12</td>
    </tr>
    <tr>
      <th>24491</th>
      <td>POMC neurons</td>
      <td>Astro_LXN+</td>
      <td>Nlgn1</td>
      <td>Nrxn1</td>
      <td>3.714226</td>
      <td>0.02740</td>
      <td>13.568213</td>
      <td>0.299596</td>
      <td>0.170498</td>
      <td>0.072551</td>
      <td>0.954964</td>
      <td>0.390915</td>
      <td>0.02020</td>
      <td>0.003352</td>
      <td>0.026571</td>
      <td>4.111543e-11</td>
    </tr>
    <tr>
      <th>7384</th>
      <td>Astro_LXN+</td>
      <td>POMC neurons</td>
      <td>Nrxn1</td>
      <td>Nlgn1</td>
      <td>3.714226</td>
      <td>0.02740</td>
      <td>13.568213</td>
      <td>0.299596</td>
      <td>0.170498</td>
      <td>0.072551</td>
      <td>0.954964</td>
      <td>0.390915</td>
      <td>0.02020</td>
      <td>0.003352</td>
      <td>0.026571</td>
      <td>4.111543e-11</td>
    </tr>
    <tr>
      <th>21574</th>
      <td>Astro_LXN-</td>
      <td>AGRP/NPY neurons</td>
      <td>Nrxn1</td>
      <td>Nlgn1</td>
      <td>3.678081</td>
      <td>0.00000</td>
      <td>13.195488</td>
      <td>0.240230</td>
      <td>0.358704</td>
      <td>0.070558</td>
      <td>0.954361</td>
      <td>0.379140</td>
      <td>0.00098</td>
      <td>0.005267</td>
      <td>0.027457</td>
      <td>1.595377e-10</td>
    </tr>
  </tbody>
</table>
</div>
</div>
</div>
<div class="cell" data-execution_count="41">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb365"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb365-1"><a href="#cb365-1" aria-hidden="true" tabindex="-1"></a>n2a_hfd <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb365-2"><a href="#cb365-2" aria-hidden="true" tabindex="-1"></a>    adata<span class="op">=</span>adata_hfd,</span>
<span id="cb365-3"><a href="#cb365-3" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb365-4"><a href="#cb365-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb365-5"><a href="#cb365-5" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb365-6"><a href="#cb365-6" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb365-7"><a href="#cb365-7" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb365-8"><a href="#cb365-8" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb365-9"><a href="#cb365-9" aria-hidden="true" tabindex="-1"></a>    top_n<span class="op">=</span><span class="dv">20</span>,</span>
<span id="cb365-10"><a href="#cb365-10" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"lr_logfc"</span>,</span>
<span id="cb365-11"><a href="#cb365-11" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb365-12"><a href="#cb365-12" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb365-13"><a href="#cb365-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb365-14"><a href="#cb365-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb365-15"><a href="#cb365-15" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb365-16"><a href="#cb365-16" aria-hidden="true" tabindex="-1"></a>    n2a_hfd</span>
<span id="cb365-17"><a href="#cb365-17" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb365-18"><a href="#cb365-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb365-19"><a href="#cb365-19" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb365-20"><a href="#cb365-20" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb365-21"><a href="#cb365-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb365-22"><a href="#cb365-22" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb365-23"><a href="#cb365-23" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb365-24"><a href="#cb365-24" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb365-25"><a href="#cb365-25" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb365-26"><a href="#cb365-26" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb365-27"><a href="#cb365-27" aria-hidden="true" tabindex="-1"></a>n2a_hfd.save(filename<span class="op">=</span><span class="st">"liana-dotplot-logfc-HF-diet-n2a.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="42">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb407"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb407-1"><a href="#cb407-1" aria-hidden="true" tabindex="-1"></a>n2a_hfd <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb407-2"><a href="#cb407-2" aria-hidden="true" tabindex="-1"></a>    adata<span class="op">=</span>adata_hfd,</span>
<span id="cb407-3"><a href="#cb407-3" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb407-4"><a href="#cb407-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb407-5"><a href="#cb407-5" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb407-6"><a href="#cb407-6" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb407-7"><a href="#cb407-7" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb407-8"><a href="#cb407-8" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb407-9"><a href="#cb407-9" aria-hidden="true" tabindex="-1"></a>    filterby<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb407-10"><a href="#cb407-10" aria-hidden="true" tabindex="-1"></a>    filter_lambda<span class="op">=</span><span class="kw">lambda</span> x: x <span class="op">&lt;=</span> <span class="fl">0.01</span>,</span>
<span id="cb407-11"><a href="#cb407-11" aria-hidden="true" tabindex="-1"></a>    top_n<span class="op">=</span><span class="dv">20</span>,</span>
<span id="cb407-12"><a href="#cb407-12" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb407-13"><a href="#cb407-13" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb407-14"><a href="#cb407-14" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb407-15"><a href="#cb407-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb407-16"><a href="#cb407-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb407-17"><a href="#cb407-17" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb407-18"><a href="#cb407-18" aria-hidden="true" tabindex="-1"></a>    n2a_hfd</span>
<span id="cb407-19"><a href="#cb407-19" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb407-20"><a href="#cb407-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb407-21"><a href="#cb407-21" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb407-22"><a href="#cb407-22" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb407-23"><a href="#cb407-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb407-24"><a href="#cb407-24" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb407-25"><a href="#cb407-25" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb407-26"><a href="#cb407-26" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb407-27"><a href="#cb407-27" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb407-28"><a href="#cb407-28" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb407-29"><a href="#cb407-29" aria-hidden="true" tabindex="-1"></a>n2a_hfd.save(filename<span class="op">=</span><span class="st">"liana-dotplot-specificity_rank-HF-diet-n2a.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="43">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb450"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb450-1"><a href="#cb450-1" aria-hidden="true" tabindex="-1"></a>n2a_hfd <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb450-2"><a href="#cb450-2" aria-hidden="true" tabindex="-1"></a>    adata<span class="op">=</span>adata_hfd,</span>
<span id="cb450-3"><a href="#cb450-3" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb450-4"><a href="#cb450-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb450-5"><a href="#cb450-5" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb450-6"><a href="#cb450-6" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb450-7"><a href="#cb450-7" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb450-8"><a href="#cb450-8" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb450-9"><a href="#cb450-9" aria-hidden="true" tabindex="-1"></a>    filterby<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb450-10"><a href="#cb450-10" aria-hidden="true" tabindex="-1"></a>    filter_lambda<span class="op">=</span><span class="kw">lambda</span> x: x <span class="op">&lt;=</span> <span class="fl">0.01</span>,</span>
<span id="cb450-11"><a href="#cb450-11" aria-hidden="true" tabindex="-1"></a>    top_n<span class="op">=</span><span class="dv">20</span>,</span>
<span id="cb450-12"><a href="#cb450-12" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb450-13"><a href="#cb450-13" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb450-14"><a href="#cb450-14" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb450-15"><a href="#cb450-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb450-16"><a href="#cb450-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb450-17"><a href="#cb450-17" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb450-18"><a href="#cb450-18" aria-hidden="true" tabindex="-1"></a>    n2a_hfd</span>
<span id="cb450-19"><a href="#cb450-19" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb450-20"><a href="#cb450-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb450-21"><a href="#cb450-21" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb450-22"><a href="#cb450-22" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb450-23"><a href="#cb450-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb450-24"><a href="#cb450-24" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb450-25"><a href="#cb450-25" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb450-26"><a href="#cb450-26" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb450-27"><a href="#cb450-27" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb450-28"><a href="#cb450-28" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb450-29"><a href="#cb450-29" aria-hidden="true" tabindex="-1"></a>n2a_hfd.save(filename<span class="op">=</span><span class="st">"liana-dotplot-magnitude_rank-HF-diet-n2a.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="44">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb491"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb491-1"><a href="#cb491-1" aria-hidden="true" tabindex="-1"></a><span class="co"># First, filter the DataFrame</span></span>
<span id="cb491-2"><a href="#cb491-2" aria-hidden="true" tabindex="-1"></a>filtered_df <span class="op">=</span> adata_hfd.uns[<span class="st">"liana_res"</span>][</span>
<span id="cb491-3"><a href="#cb491-3" aria-hidden="true" tabindex="-1"></a>    adata_hfd.uns[<span class="st">"liana_res"</span>][<span class="st">"receptor_complex"</span>] <span class="op">==</span> <span class="st">"Insr"</span></span>
<span id="cb491-4"><a href="#cb491-4" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb491-5"><a href="#cb491-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb491-6"><a href="#cb491-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Then, pass the filtered DataFrame to the dotplot function</span></span>
<span id="cb491-7"><a href="#cb491-7" aria-hidden="true" tabindex="-1"></a>n2a_hfd <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb491-8"><a href="#cb491-8" aria-hidden="true" tabindex="-1"></a>    liana_res<span class="op">=</span>filtered_df,</span>
<span id="cb491-9"><a href="#cb491-9" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb491-10"><a href="#cb491-10" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb491-11"><a href="#cb491-11" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb491-12"><a href="#cb491-12" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb491-13"><a href="#cb491-13" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb491-14"><a href="#cb491-14" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb491-15"><a href="#cb491-15" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"lr_logfc"</span>,</span>
<span id="cb491-16"><a href="#cb491-16" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb491-17"><a href="#cb491-17" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb491-18"><a href="#cb491-18" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb491-19"><a href="#cb491-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb491-20"><a href="#cb491-20" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb491-21"><a href="#cb491-21" aria-hidden="true" tabindex="-1"></a>    n2a_hfd</span>
<span id="cb491-22"><a href="#cb491-22" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb491-23"><a href="#cb491-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb491-24"><a href="#cb491-24" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb491-25"><a href="#cb491-25" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb491-26"><a href="#cb491-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb491-27"><a href="#cb491-27" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb491-28"><a href="#cb491-28" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb491-29"><a href="#cb491-29" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb491-30"><a href="#cb491-30" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb491-31"><a href="#cb491-31" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb491-32"><a href="#cb491-32" aria-hidden="true" tabindex="-1"></a>n2a_hfd.save(filename<span class="op">=</span><span class="st">"liana-dotplot-insr-logfc-HF-diet-n2a.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="45">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb519"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb519-1"><a href="#cb519-1" aria-hidden="true" tabindex="-1"></a>n2a_hfd <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb519-2"><a href="#cb519-2" aria-hidden="true" tabindex="-1"></a>    liana_res<span class="op">=</span>filtered_df,</span>
<span id="cb519-3"><a href="#cb519-3" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb519-4"><a href="#cb519-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb519-5"><a href="#cb519-5" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb519-6"><a href="#cb519-6" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb519-7"><a href="#cb519-7" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb519-8"><a href="#cb519-8" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb519-9"><a href="#cb519-9" aria-hidden="true" tabindex="-1"></a>    filterby<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb519-10"><a href="#cb519-10" aria-hidden="true" tabindex="-1"></a>    filter_lambda<span class="op">=</span><span class="kw">lambda</span> x: x <span class="op">&lt;=</span> <span class="fl">0.01</span>,</span>
<span id="cb519-11"><a href="#cb519-11" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb519-12"><a href="#cb519-12" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb519-13"><a href="#cb519-13" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb519-14"><a href="#cb519-14" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb519-15"><a href="#cb519-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb519-16"><a href="#cb519-16" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb519-17"><a href="#cb519-17" aria-hidden="true" tabindex="-1"></a>    n2a_hfd</span>
<span id="cb519-18"><a href="#cb519-18" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb519-19"><a href="#cb519-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb519-20"><a href="#cb519-20" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb519-21"><a href="#cb519-21" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb519-22"><a href="#cb519-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb519-23"><a href="#cb519-23" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb519-24"><a href="#cb519-24" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb519-25"><a href="#cb519-25" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb519-26"><a href="#cb519-26" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb519-27"><a href="#cb519-27" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb519-28"><a href="#cb519-28" aria-hidden="true" tabindex="-1"></a>n2a_hfd.save(filename<span class="op">=</span><span class="st">"liana-dotplot-insr-specificity_rank-HF-diet-n2a.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="46">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb540"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb540-1"><a href="#cb540-1" aria-hidden="true" tabindex="-1"></a>n2a_hfd <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb540-2"><a href="#cb540-2" aria-hidden="true" tabindex="-1"></a>    liana_res<span class="op">=</span>filtered_df,</span>
<span id="cb540-3"><a href="#cb540-3" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb540-4"><a href="#cb540-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb540-5"><a href="#cb540-5" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb540-6"><a href="#cb540-6" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb540-7"><a href="#cb540-7" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb540-8"><a href="#cb540-8" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb540-9"><a href="#cb540-9" aria-hidden="true" tabindex="-1"></a>    filterby<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb540-10"><a href="#cb540-10" aria-hidden="true" tabindex="-1"></a>    filter_lambda<span class="op">=</span><span class="kw">lambda</span> x: x <span class="op">&lt;=</span> <span class="fl">0.01</span>,</span>
<span id="cb540-11"><a href="#cb540-11" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb540-12"><a href="#cb540-12" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb540-13"><a href="#cb540-13" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb540-14"><a href="#cb540-14" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb540-15"><a href="#cb540-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb540-16"><a href="#cb540-16" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb540-17"><a href="#cb540-17" aria-hidden="true" tabindex="-1"></a>    n2a_hfd</span>
<span id="cb540-18"><a href="#cb540-18" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb540-19"><a href="#cb540-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb540-20"><a href="#cb540-20" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb540-21"><a href="#cb540-21" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb540-22"><a href="#cb540-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb540-23"><a href="#cb540-23" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb540-24"><a href="#cb540-24" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb540-25"><a href="#cb540-25" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb540-26"><a href="#cb540-26" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb540-27"><a href="#cb540-27" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb540-28"><a href="#cb540-28" aria-hidden="true" tabindex="-1"></a>n2a_hfd.save(filename<span class="op">=</span><span class="st">"liana-dotplot-insr-magnitude_rank-HF-diet-n2a.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="47">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb562"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb562-1"><a href="#cb562-1" aria-hidden="true" tabindex="-1"></a>a2n_hfd <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb562-2"><a href="#cb562-2" aria-hidden="true" tabindex="-1"></a>    adata<span class="op">=</span>adata_hfd,</span>
<span id="cb562-3"><a href="#cb562-3" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb562-4"><a href="#cb562-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb562-5"><a href="#cb562-5" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb562-6"><a href="#cb562-6" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb562-7"><a href="#cb562-7" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb562-8"><a href="#cb562-8" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb562-9"><a href="#cb562-9" aria-hidden="true" tabindex="-1"></a>    top_n<span class="op">=</span><span class="dv">20</span>,</span>
<span id="cb562-10"><a href="#cb562-10" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"lr_logfc"</span>,</span>
<span id="cb562-11"><a href="#cb562-11" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb562-12"><a href="#cb562-12" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb562-13"><a href="#cb562-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb562-14"><a href="#cb562-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb562-15"><a href="#cb562-15" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb562-16"><a href="#cb562-16" aria-hidden="true" tabindex="-1"></a>    a2n_hfd</span>
<span id="cb562-17"><a href="#cb562-17" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb562-18"><a href="#cb562-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb562-19"><a href="#cb562-19" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb562-20"><a href="#cb562-20" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb562-21"><a href="#cb562-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb562-22"><a href="#cb562-22" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb562-23"><a href="#cb562-23" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb562-24"><a href="#cb562-24" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb562-25"><a href="#cb562-25" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb562-26"><a href="#cb562-26" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb562-27"><a href="#cb562-27" aria-hidden="true" tabindex="-1"></a>a2n_hfd.save(filename<span class="op">=</span><span class="st">"liana-dotplot-logfc-HF-diet-a2n.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="48">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb601"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb601-1"><a href="#cb601-1" aria-hidden="true" tabindex="-1"></a>a2n_hfd <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb601-2"><a href="#cb601-2" aria-hidden="true" tabindex="-1"></a>    adata<span class="op">=</span>adata_hfd,</span>
<span id="cb601-3"><a href="#cb601-3" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb601-4"><a href="#cb601-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb601-5"><a href="#cb601-5" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb601-6"><a href="#cb601-6" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb601-7"><a href="#cb601-7" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb601-8"><a href="#cb601-8" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb601-9"><a href="#cb601-9" aria-hidden="true" tabindex="-1"></a>    filterby<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb601-10"><a href="#cb601-10" aria-hidden="true" tabindex="-1"></a>    filter_lambda<span class="op">=</span><span class="kw">lambda</span> x: x <span class="op">&lt;=</span> <span class="fl">0.01</span>,</span>
<span id="cb601-11"><a href="#cb601-11" aria-hidden="true" tabindex="-1"></a>    top_n<span class="op">=</span><span class="dv">20</span>,</span>
<span id="cb601-12"><a href="#cb601-12" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb601-13"><a href="#cb601-13" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb601-14"><a href="#cb601-14" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb601-15"><a href="#cb601-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb601-16"><a href="#cb601-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb601-17"><a href="#cb601-17" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb601-18"><a href="#cb601-18" aria-hidden="true" tabindex="-1"></a>    a2n_hfd</span>
<span id="cb601-19"><a href="#cb601-19" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb601-20"><a href="#cb601-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb601-21"><a href="#cb601-21" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb601-22"><a href="#cb601-22" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb601-23"><a href="#cb601-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb601-24"><a href="#cb601-24" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb601-25"><a href="#cb601-25" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb601-26"><a href="#cb601-26" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb601-27"><a href="#cb601-27" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb601-28"><a href="#cb601-28" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb601-29"><a href="#cb601-29" aria-hidden="true" tabindex="-1"></a>a2n_hfd.save(filename<span class="op">=</span><span class="st">"liana-dotplot-specificity_rank-HF-diet-a2n.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="49">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb640"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb640-1"><a href="#cb640-1" aria-hidden="true" tabindex="-1"></a>a2n_hfd <span class="op">=</span> li.pl.dotplot(</span>
<span id="cb640-2"><a href="#cb640-2" aria-hidden="true" tabindex="-1"></a>    adata<span class="op">=</span>adata_hfd,</span>
<span id="cb640-3"><a href="#cb640-3" aria-hidden="true" tabindex="-1"></a>    colour<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb640-4"><a href="#cb640-4" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb640-5"><a href="#cb640-5" aria-hidden="true" tabindex="-1"></a>    inverse_size<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb640-6"><a href="#cb640-6" aria-hidden="true" tabindex="-1"></a>    inverse_colour<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb640-7"><a href="#cb640-7" aria-hidden="true" tabindex="-1"></a>    source_labels<span class="op">=</span>[<span class="st">"Astro_LXN+"</span>, <span class="st">"Astro_LXN-"</span>],</span>
<span id="cb640-8"><a href="#cb640-8" aria-hidden="true" tabindex="-1"></a>    target_labels<span class="op">=</span>[<span class="st">"POMC neurons"</span>, <span class="st">"AGRP/NPY neurons"</span>],</span>
<span id="cb640-9"><a href="#cb640-9" aria-hidden="true" tabindex="-1"></a>    filterby<span class="op">=</span><span class="st">"specificity_rank"</span>,</span>
<span id="cb640-10"><a href="#cb640-10" aria-hidden="true" tabindex="-1"></a>    filter_lambda<span class="op">=</span><span class="kw">lambda</span> x: x <span class="op">&lt;=</span> <span class="fl">0.01</span>,</span>
<span id="cb640-11"><a href="#cb640-11" aria-hidden="true" tabindex="-1"></a>    top_n<span class="op">=</span><span class="dv">20</span>,</span>
<span id="cb640-12"><a href="#cb640-12" aria-hidden="true" tabindex="-1"></a>    orderby<span class="op">=</span><span class="st">"magnitude_rank"</span>,</span>
<span id="cb640-13"><a href="#cb640-13" aria-hidden="true" tabindex="-1"></a>    orderby_ascending<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb640-14"><a href="#cb640-14" aria-hidden="true" tabindex="-1"></a>    figure_size<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">10</span>),</span>
<span id="cb640-15"><a href="#cb640-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb640-16"><a href="#cb640-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb640-17"><a href="#cb640-17" aria-hidden="true" tabindex="-1"></a>(</span>
<span id="cb640-18"><a href="#cb640-18" aria-hidden="true" tabindex="-1"></a>    a2n_hfd</span>
<span id="cb640-19"><a href="#cb640-19" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb640-20"><a href="#cb640-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change theme</span></span>
<span id="cb640-21"><a href="#cb640-21" aria-hidden="true" tabindex="-1"></a>    p9.theme_tufte()</span>
<span id="cb640-22"><a href="#cb640-22" aria-hidden="true" tabindex="-1"></a>    <span class="op">+</span></span>
<span id="cb640-23"><a href="#cb640-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># modify theme</span></span>
<span id="cb640-24"><a href="#cb640-24" aria-hidden="true" tabindex="-1"></a>    p9.theme(</span>
<span id="cb640-25"><a href="#cb640-25" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adjust facet size</span></span>
<span id="cb640-26"><a href="#cb640-26" aria-hidden="true" tabindex="-1"></a>        strip_text<span class="op">=</span>p9.element_text(size<span class="op">=</span><span class="dv">12</span>)</span>
<span id="cb640-27"><a href="#cb640-27" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb640-28"><a href="#cb640-28" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb640-29"><a href="#cb640-29" aria-hidden="true" tabindex="-1"></a>a2n_hfd.save(filename<span class="op">=</span><span class="st">"liana-dotplot-magnitude_rank-HF-diet-a2n.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>findfont: Font family 'Dejavu Sans' not found.</code></pre>
</div>
</div>
<p>Save data</p>
<div class="cell" data-execution_count="50">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb681"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb681-1"><a href="#cb681-1" aria-hidden="true" tabindex="-1"></a>adata_hfd.write(</span>
<span id="cb681-2"><a href="#cb681-2" aria-hidden="true" tabindex="-1"></a>    <span class="ss">f"/data/</span><span class="sc">{</span>bioproject<span class="sc">}</span><span class="ss">/data/velocity-signalling/</span><span class="sc">{</span>bioproject<span class="sc">}</span><span class="ss">-signalling-treatment.h5ad"</span></span>
<span id="cb681-3"><a href="#cb681-3" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="preprocess-data-for-velocity" class="level3">
<h3 class="anchored" data-anchor-id="preprocess-data-for-velocity">Preprocess data for velocity</h3>
<div class="cell" data-execution_count="51">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb682"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb682-1"><a href="#cb682-1" aria-hidden="true" tabindex="-1"></a>adata1 <span class="op">=</span> scv.read(<span class="st">"../piscem_spliceu/SRR10992871/SRR10992871.h5ad"</span>)</span>
<span id="cb682-2"><a href="#cb682-2" aria-hidden="true" tabindex="-1"></a>adata2 <span class="op">=</span> scv.read(<span class="st">"../piscem_spliceu/SRR10992872/SRR10992872.h5ad"</span>)</span>
<span id="cb682-3"><a href="#cb682-3" aria-hidden="true" tabindex="-1"></a>adata3 <span class="op">=</span> scv.read(<span class="st">"../piscem_spliceu/SRR10992873/SRR10992873.h5ad"</span>)</span>
<span id="cb682-4"><a href="#cb682-4" aria-hidden="true" tabindex="-1"></a>adata4 <span class="op">=</span> scv.read(<span class="st">"../piscem_spliceu/SRR10992874/SRR10992874.h5ad"</span>)</span>
<span id="cb682-5"><a href="#cb682-5" aria-hidden="true" tabindex="-1"></a>adata5 <span class="op">=</span> scv.read(<span class="st">"../piscem_spliceu/SRR10992875/SRR10992875.h5ad"</span>)</span>
<span id="cb682-6"><a href="#cb682-6" aria-hidden="true" tabindex="-1"></a>adata6 <span class="op">=</span> scv.read(<span class="st">"../piscem_spliceu/SRR10992876/SRR10992876.h5ad"</span>)</span>
<span id="cb682-7"><a href="#cb682-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb682-8"><a href="#cb682-8" aria-hidden="true" tabindex="-1"></a><span class="co"># prepare cell names</span></span>
<span id="cb682-9"><a href="#cb682-9" aria-hidden="true" tabindex="-1"></a>cells1 <span class="op">=</span> <span class="st">"SRR10992871_"</span> <span class="op">+</span> adata1.obs_names <span class="op">+</span> <span class="st">"-1"</span></span>
<span id="cb682-10"><a href="#cb682-10" aria-hidden="true" tabindex="-1"></a>adata1.obs_names <span class="op">=</span> cells1</span>
<span id="cb682-11"><a href="#cb682-11" aria-hidden="true" tabindex="-1"></a>adata1.obs[<span class="st">"run"</span>] <span class="op">=</span> <span class="st">"SRR10992871"</span></span>
<span id="cb682-12"><a href="#cb682-12" aria-hidden="true" tabindex="-1"></a>cells2 <span class="op">=</span> <span class="st">"SRR10992872_"</span> <span class="op">+</span> adata2.obs_names <span class="op">+</span> <span class="st">"-1"</span></span>
<span id="cb682-13"><a href="#cb682-13" aria-hidden="true" tabindex="-1"></a>adata2.obs_names <span class="op">=</span> cells2</span>
<span id="cb682-14"><a href="#cb682-14" aria-hidden="true" tabindex="-1"></a>adata2.obs[<span class="st">"run"</span>] <span class="op">=</span> <span class="st">"SRR10992872"</span></span>
<span id="cb682-15"><a href="#cb682-15" aria-hidden="true" tabindex="-1"></a>cells3 <span class="op">=</span> <span class="st">"SRR10992873_"</span> <span class="op">+</span> adata3.obs_names <span class="op">+</span> <span class="st">"-1"</span></span>
<span id="cb682-16"><a href="#cb682-16" aria-hidden="true" tabindex="-1"></a>adata3.obs_names <span class="op">=</span> cells3</span>
<span id="cb682-17"><a href="#cb682-17" aria-hidden="true" tabindex="-1"></a>adata3.obs[<span class="st">"run"</span>] <span class="op">=</span> <span class="st">"SRR10992873"</span></span>
<span id="cb682-18"><a href="#cb682-18" aria-hidden="true" tabindex="-1"></a>cells4 <span class="op">=</span> <span class="st">"SRR10992874_"</span> <span class="op">+</span> adata4.obs_names <span class="op">+</span> <span class="st">"-1"</span></span>
<span id="cb682-19"><a href="#cb682-19" aria-hidden="true" tabindex="-1"></a>adata4.obs_names <span class="op">=</span> cells4</span>
<span id="cb682-20"><a href="#cb682-20" aria-hidden="true" tabindex="-1"></a>adata4.obs[<span class="st">"run"</span>] <span class="op">=</span> <span class="st">"SRR10992874"</span></span>
<span id="cb682-21"><a href="#cb682-21" aria-hidden="true" tabindex="-1"></a>cells5 <span class="op">=</span> <span class="st">"SRR10992875_"</span> <span class="op">+</span> adata5.obs_names <span class="op">+</span> <span class="st">"-1"</span></span>
<span id="cb682-22"><a href="#cb682-22" aria-hidden="true" tabindex="-1"></a>adata5.obs_names <span class="op">=</span> cells5</span>
<span id="cb682-23"><a href="#cb682-23" aria-hidden="true" tabindex="-1"></a>adata5.obs[<span class="st">"run"</span>] <span class="op">=</span> <span class="st">"SRR10992875"</span></span>
<span id="cb682-24"><a href="#cb682-24" aria-hidden="true" tabindex="-1"></a>cells6 <span class="op">=</span> <span class="st">"SRR10992876_"</span> <span class="op">+</span> adata6.obs_names <span class="op">+</span> <span class="st">"-1"</span></span>
<span id="cb682-25"><a href="#cb682-25" aria-hidden="true" tabindex="-1"></a>adata6.obs_names <span class="op">=</span> cells6</span>
<span id="cb682-26"><a href="#cb682-26" aria-hidden="true" tabindex="-1"></a>adata6.obs[<span class="st">"run"</span>] <span class="op">=</span> <span class="st">"SRR10992876"</span></span>
<span id="cb682-27"><a href="#cb682-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb682-28"><a href="#cb682-28" aria-hidden="true" tabindex="-1"></a><span class="co"># subset cleared astrocytes</span></span>
<span id="cb682-29"><a href="#cb682-29" aria-hidden="true" tabindex="-1"></a>adata1 <span class="op">=</span> adata1[adata1.obs_names.isin(adata.obs_names), :]</span>
<span id="cb682-30"><a href="#cb682-30" aria-hidden="true" tabindex="-1"></a>adata2 <span class="op">=</span> adata2[adata2.obs_names.isin(adata.obs_names), :]</span>
<span id="cb682-31"><a href="#cb682-31" aria-hidden="true" tabindex="-1"></a>adata3 <span class="op">=</span> adata3[adata3.obs_names.isin(adata.obs_names), :]</span>
<span id="cb682-32"><a href="#cb682-32" aria-hidden="true" tabindex="-1"></a>adata4 <span class="op">=</span> adata4[adata4.obs_names.isin(adata.obs_names), :]</span>
<span id="cb682-33"><a href="#cb682-33" aria-hidden="true" tabindex="-1"></a>adata5 <span class="op">=</span> adata5[adata5.obs_names.isin(adata.obs_names), :]</span>
<span id="cb682-34"><a href="#cb682-34" aria-hidden="true" tabindex="-1"></a>adata6 <span class="op">=</span> adata6[adata6.obs_names.isin(adata.obs_names), :]</span>
<span id="cb682-35"><a href="#cb682-35" aria-hidden="true" tabindex="-1"></a>velocity_data <span class="op">=</span> ad.concat(</span>
<span id="cb682-36"><a href="#cb682-36" aria-hidden="true" tabindex="-1"></a>    [adata1, adata2, adata3, adata4, adata5, adata6], join<span class="op">=</span><span class="st">"inner"</span></span>
<span id="cb682-37"><a href="#cb682-37" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="52">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb683"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb683-1"><a href="#cb683-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get a list of gene ids from your Scanpy object</span></span>
<span id="cb683-2"><a href="#cb683-2" aria-hidden="true" tabindex="-1"></a>gene_ids <span class="op">=</span> velocity_data.var_names.tolist()</span>
<span id="cb683-3"><a href="#cb683-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb683-4"><a href="#cb683-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert gene symbols to Ensembl IDs using gprofiler</span></span>
<span id="cb683-5"><a href="#cb683-5" aria-hidden="true" tabindex="-1"></a>gp <span class="op">=</span> GProfiler(return_dataframe<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb683-6"><a href="#cb683-6" aria-hidden="true" tabindex="-1"></a>gene_symbols <span class="op">=</span> (</span>
<span id="cb683-7"><a href="#cb683-7" aria-hidden="true" tabindex="-1"></a>    gp.convert(query<span class="op">=</span>gene_ids, organism<span class="op">=</span><span class="st">"mmusculus"</span>, target_namespace<span class="op">=</span><span class="st">"MGI"</span>)</span>
<span id="cb683-8"><a href="#cb683-8" aria-hidden="true" tabindex="-1"></a>    .drop_duplicates([<span class="st">"incoming"</span>])</span>
<span id="cb683-9"><a href="#cb683-9" aria-hidden="true" tabindex="-1"></a>    .set_index(<span class="st">"incoming"</span>, drop<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb683-10"><a href="#cb683-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="53">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb684"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb684-1"><a href="#cb684-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the gene IDs as a new column in your Scanpy object's variable metadata</span></span>
<span id="cb684-2"><a href="#cb684-2" aria-hidden="true" tabindex="-1"></a>velocity_data.var[<span class="st">"ensembl_ids"</span>] <span class="op">=</span> gene_symbols[<span class="st">"incoming"</span>][gene_ids].values</span>
<span id="cb684-3"><a href="#cb684-3" aria-hidden="true" tabindex="-1"></a>velocity_data.var[<span class="st">"mgi_symbol"</span>] <span class="op">=</span> gene_symbols[<span class="st">"converted"</span>][gene_ids].values</span>
<span id="cb684-4"><a href="#cb684-4" aria-hidden="true" tabindex="-1"></a>velocity_data.var_names <span class="op">=</span> velocity_data.var[<span class="st">"mgi_symbol"</span>]</span>
<span id="cb684-5"><a href="#cb684-5" aria-hidden="true" tabindex="-1"></a>velocity_data.var_names_make_unique()</span>
<span id="cb684-6"><a href="#cb684-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb684-7"><a href="#cb684-7" aria-hidden="true" tabindex="-1"></a>velocity_data.obs[<span class="st">"cell_id"</span>] <span class="op">=</span> velocity_data.obs_names</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="54">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb685"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb685-1"><a href="#cb685-1" aria-hidden="true" tabindex="-1"></a>velocity_data <span class="op">=</span> velocity_data[</span>
<span id="cb685-2"><a href="#cb685-2" aria-hidden="true" tabindex="-1"></a>    :,</span>
<span id="cb685-3"><a href="#cb685-3" aria-hidden="true" tabindex="-1"></a>    <span class="op">~</span>velocity_data.var_names.<span class="bu">str</span>.match(</span>
<span id="cb685-4"><a href="#cb685-4" aria-hidden="true" tabindex="-1"></a>        <span class="vs">r"(^Hla-|^Ig[hjkl]|^Rna|^mt-|^Rp[sl]|^Hb[^(p)]|^Gm)"</span></span>
<span id="cb685-5"><a href="#cb685-5" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb685-6"><a href="#cb685-6" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb685-7"><a href="#cb685-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb685-8"><a href="#cb685-8" aria-hidden="true" tabindex="-1"></a>velocity_data <span class="op">=</span> velocity_data[:, <span class="op">~</span>velocity_data.var_names.isin(hk_genes1)]</span>
<span id="cb685-9"><a href="#cb685-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb685-10"><a href="#cb685-10" aria-hidden="true" tabindex="-1"></a>velocity_data.obs <span class="op">=</span> pd.merge(velocity_data.obs, adata.obs, how<span class="op">=</span><span class="st">"left"</span>).set_index(</span>
<span id="cb685-11"><a href="#cb685-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">"cell_id"</span>, drop<span class="op">=</span><span class="va">False</span></span>
<span id="cb685-12"><a href="#cb685-12" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="55">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb686"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb686-1"><a href="#cb686-1" aria-hidden="true" tabindex="-1"></a>scv.pl.proportions(velocity_data, groupby<span class="op">=</span><span class="st">"libname"</span>, save<span class="op">=</span><span class="st">"-libname-proportions.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_proportions_-libname-proportions.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-56-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="56">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb688"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb688-1"><a href="#cb688-1" aria-hidden="true" tabindex="-1"></a>scv.pl.proportions(velocity_data, groupby<span class="op">=</span><span class="st">"k_tree"</span>, save<span class="op">=</span><span class="st">"-k_tree-proportions.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_proportions_-k_tree-proportions.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-57-output-2.png" class="img-fluid"></p>
</div>
</div>
</section>
<section id="rna-velocity" class="level3">
<h3 class="anchored" data-anchor-id="rna-velocity">RNA velocity</h3>
<div class="cell" data-execution_count="57">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb690"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb690-1"><a href="#cb690-1" aria-hidden="true" tabindex="-1"></a><span class="co">## run scvelo dynamic model</span></span>
<span id="cb690-2"><a href="#cb690-2" aria-hidden="true" tabindex="-1"></a>scv.pp.filter_and_normalize(</span>
<span id="cb690-3"><a href="#cb690-3" aria-hidden="true" tabindex="-1"></a>    velocity_data, min_shared_counts<span class="op">=</span>min_shared_umis, n_top_genes<span class="op">=</span>n_top_hvg</span>
<span id="cb690-4"><a href="#cb690-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb690-5"><a href="#cb690-5" aria-hidden="true" tabindex="-1"></a>sc.tl.pca(velocity_data, n_comps<span class="op">=</span>npcs, svd_solver<span class="op">=</span><span class="st">"arpack"</span>, random_state<span class="op">=</span>reseed)</span>
<span id="cb690-6"><a href="#cb690-6" aria-hidden="true" tabindex="-1"></a>sc.pp.neighbors(velocity_data, n_pcs<span class="op">=</span>npcs_selected, n_neighbors<span class="op">=</span>k)</span>
<span id="cb690-7"><a href="#cb690-7" aria-hidden="true" tabindex="-1"></a>scv.pp.moments(velocity_data)  <span class="co">## compute moments</span></span>
<span id="cb690-8"><a href="#cb690-8" aria-hidden="true" tabindex="-1"></a>sc.tl.paga(velocity_data, groups<span class="op">=</span><span class="st">"libname"</span>)</span>
<span id="cb690-9"><a href="#cb690-9" aria-hidden="true" tabindex="-1"></a>sc.pl.paga(velocity_data, plot<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb690-10"><a href="#cb690-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb690-11"><a href="#cb690-11" aria-hidden="true" tabindex="-1"></a>embedding <span class="op">=</span> pacmap.PaCMAP(</span>
<span id="cb690-12"><a href="#cb690-12" aria-hidden="true" tabindex="-1"></a>    n_components<span class="op">=</span><span class="dv">2</span>, n_neighbors<span class="op">=</span><span class="va">None</span>, MN_ratio<span class="op">=</span><span class="fl">0.5</span>, FP_ratio<span class="op">=</span><span class="fl">2.0</span>, apply_pca<span class="op">=</span><span class="va">False</span></span>
<span id="cb690-13"><a href="#cb690-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb690-14"><a href="#cb690-14" aria-hidden="true" tabindex="-1"></a>velocity_data.obsm[<span class="st">"X_pacmap"</span>] <span class="op">=</span> embedding.fit_transform(</span>
<span id="cb690-15"><a href="#cb690-15" aria-hidden="true" tabindex="-1"></a>    velocity_data.obsm[<span class="st">"X_pca"</span>], init<span class="op">=</span><span class="st">"pca"</span></span>
<span id="cb690-16"><a href="#cb690-16" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb690-17"><a href="#cb690-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb690-18"><a href="#cb690-18" aria-hidden="true" tabindex="-1"></a>sc.tl.draw_graph(velocity_data, init_pos<span class="op">=</span><span class="st">"paga"</span>)</span>
<span id="cb690-19"><a href="#cb690-19" aria-hidden="true" tabindex="-1"></a>sc.tl.umap(velocity_data, init_pos<span class="op">=</span><span class="st">"paga"</span>)</span>
<span id="cb690-20"><a href="#cb690-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb690-21"><a href="#cb690-21" aria-hidden="true" tabindex="-1"></a>scv.tl.recover_dynamics(velocity_data, n_jobs<span class="op">=</span>threads)  <span class="co">## model</span></span>
<span id="cb690-22"><a href="#cb690-22" aria-hidden="true" tabindex="-1"></a>scv.tl.velocity(velocity_data, mode<span class="op">=</span><span class="st">"dynamical"</span>)</span>
<span id="cb690-23"><a href="#cb690-23" aria-hidden="true" tabindex="-1"></a>scv.tl.velocity_graph(velocity_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Filtered out 22834 genes that are detected 15 counts (shared).
Normalized count data: X, spliced, unspliced.
Skip filtering by dispersion since number of variables are less than `n_top_genes`.
WARNING: Did not modify X as it looks preprocessed already.
computing PCA</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    with n_comps=50</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    finished (0:00:00)</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>computing neighbors</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    using 'X_pca' with n_pcs = 30</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    finished (0:00:01)</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>computing neighbors
    finished (0:00:00) --&gt; added 
    'distances' and 'connectivities', weighted adjacency matrices (adata.obsp)
computing moments based on connectivities
    finished (0:00:00) --&gt; added 
    'Ms' and 'Mu', moments of un/spliced abundances (adata.layers)
running PAGA</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    finished (0:00:00)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/networkx/convert.py:158: DeprecationWarning: 

The scipy.sparse array containers will be used instead of matrices
in Networkx 3.0. Use `from_scipy_sparse_array` instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>drawing single-cell graph using layout 'fa'</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    finished (0:00:03)</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>computing UMAP</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    finished (0:00:01)</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>recovering dynamics (using 80/256 cores)</code></pre>
</div>
<div class="cell-output cell-output-display">
<script type="application/vnd.jupyter.widget-view+json">
{"model_id":"d35b970892e546a1abf30a1c8c651ea1","version_major":2,"version_minor":0}
</script>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    finished (0:00:10) --&gt; added 
    'fit_pars', fitted parameters for splicing dynamics (adata.var)
computing velocities
    finished (0:00:00) --&gt; added 
    'velocity', velocity vectors for each individual cell (adata.layers)
computing velocity graph (using 1/256 cores)</code></pre>
</div>
<div class="cell-output cell-output-display">
<script type="application/vnd.jupyter.widget-view+json">
{"model_id":"d31b598d82994f42bda505f47fd5b247","version_major":2,"version_minor":0}
</script>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    finished (0:00:00) --&gt; added 
    'velocity_graph', sparse matrix with cosine correlations (adata.uns)</code></pre>
</div>
</div>
<div class="cell" data-execution_count="58">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb707"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb707-1"><a href="#cb707-1" aria-hidden="true" tabindex="-1"></a>save_PRJNA604055 <span class="op">=</span> <span class="ss">f"-PRJNA604055-astrocytes_datasets-velocity.pdf"</span></span>
<span id="cb707-2"><a href="#cb707-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb707-3"><a href="#cb707-3" aria-hidden="true" tabindex="-1"></a>scv.pl.velocity_embedding_stream(</span>
<span id="cb707-4"><a href="#cb707-4" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb707-5"><a href="#cb707-5" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span><span class="st">"k_tree"</span>,</span>
<span id="cb707-6"><a href="#cb707-6" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"umap"</span>,</span>
<span id="cb707-7"><a href="#cb707-7" aria-hidden="true" tabindex="-1"></a>    density<span class="op">=</span>dnsty,</span>
<span id="cb707-8"><a href="#cb707-8" aria-hidden="true" tabindex="-1"></a>    cutoff_perc<span class="op">=</span><span class="dv">0</span>,</span>
<span id="cb707-9"><a href="#cb707-9" aria-hidden="true" tabindex="-1"></a>    n_neighbors<span class="op">=</span>k,</span>
<span id="cb707-10"><a href="#cb707-10" aria-hidden="true" tabindex="-1"></a>    smooth<span class="op">=</span><span class="fl">0.8</span>,</span>
<span id="cb707-11"><a href="#cb707-11" aria-hidden="true" tabindex="-1"></a>    min_mass<span class="op">=</span><span class="dv">4</span>,</span>
<span id="cb707-12"><a href="#cb707-12" aria-hidden="true" tabindex="-1"></a>    title<span class="op">=</span><span class="st">""</span>,</span>
<span id="cb707-13"><a href="#cb707-13" aria-hidden="true" tabindex="-1"></a>    legend_fontoutline<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb707-14"><a href="#cb707-14" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span>pt_size,</span>
<span id="cb707-15"><a href="#cb707-15" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.8</span>,</span>
<span id="cb707-16"><a href="#cb707-16" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb707-17"><a href="#cb707-17" aria-hidden="true" tabindex="-1"></a>    linewidth<span class="op">=</span><span class="fl">2.5</span>,</span>
<span id="cb707-18"><a href="#cb707-18" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb707-19"><a href="#cb707-19" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>plt_size,</span>
<span id="cb707-20"><a href="#cb707-20" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"nipy_spectral"</span>,</span>
<span id="cb707-21"><a href="#cb707-21" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span>save_PRJNA604055,</span>
<span id="cb707-22"><a href="#cb707-22" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb707-23"><a href="#cb707-23" aria-hidden="true" tabindex="-1"></a>plt.show()</span>
<span id="cb707-24"><a href="#cb707-24" aria-hidden="true" tabindex="-1"></a>plt.savefig(<span class="st">"dynamic_velocity.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>computing velocity embedding
    finished (0:00:00) --&gt; added
    'velocity_umap', embedded velocity vectors (adata.obsm)</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-59-output-3.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<pre><code>&lt;Figure size 480x480 with 0 Axes&gt;</code></pre>
</div>
</div>
<div class="cell" data-execution_count="59">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb711"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb711-1"><a href="#cb711-1" aria-hidden="true" tabindex="-1"></a>scv.pl.velocity_embedding_stream(</span>
<span id="cb711-2"><a href="#cb711-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb711-3"><a href="#cb711-3" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span><span class="st">"libname"</span>,</span>
<span id="cb711-4"><a href="#cb711-4" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"umap"</span>,</span>
<span id="cb711-5"><a href="#cb711-5" aria-hidden="true" tabindex="-1"></a>    legend_loc<span class="op">=</span><span class="st">"none"</span>,</span>
<span id="cb711-6"><a href="#cb711-6" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.3</span>,</span>
<span id="cb711-7"><a href="#cb711-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb711-8"><a href="#cb711-8" aria-hidden="true" tabindex="-1"></a>    arrow_color<span class="op">=</span><span class="st">"darkslategrey"</span>,</span>
<span id="cb711-9"><a href="#cb711-9" aria-hidden="true" tabindex="-1"></a>    density<span class="op">=</span><span class="fl">0.9</span>,</span>
<span id="cb711-10"><a href="#cb711-10" aria-hidden="true" tabindex="-1"></a>    linewidth<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb711-11"><a href="#cb711-11" aria-hidden="true" tabindex="-1"></a>    arrow_size<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb711-12"><a href="#cb711-12" aria-hidden="true" tabindex="-1"></a>    min_mass<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb711-13"><a href="#cb711-13" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb711-14"><a href="#cb711-14" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>plt_size,</span>
<span id="cb711-15"><a href="#cb711-15" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"umap-libname"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb711-16"><a href="#cb711-16" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_umap-libname-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-60-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="60">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb713"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb713-1"><a href="#cb713-1" aria-hidden="true" tabindex="-1"></a>scv.pl.velocity_embedding_stream(</span>
<span id="cb713-2"><a href="#cb713-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb713-3"><a href="#cb713-3" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span><span class="st">"k_tree"</span>,</span>
<span id="cb713-4"><a href="#cb713-4" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"pacmap"</span>,</span>
<span id="cb713-5"><a href="#cb713-5" aria-hidden="true" tabindex="-1"></a>    legend_loc<span class="op">=</span><span class="st">"none"</span>,</span>
<span id="cb713-6"><a href="#cb713-6" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.3</span>,</span>
<span id="cb713-7"><a href="#cb713-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb713-8"><a href="#cb713-8" aria-hidden="true" tabindex="-1"></a>    arrow_color<span class="op">=</span><span class="st">"darkslategrey"</span>,</span>
<span id="cb713-9"><a href="#cb713-9" aria-hidden="true" tabindex="-1"></a>    density<span class="op">=</span><span class="fl">0.9</span>,</span>
<span id="cb713-10"><a href="#cb713-10" aria-hidden="true" tabindex="-1"></a>    linewidth<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb713-11"><a href="#cb713-11" aria-hidden="true" tabindex="-1"></a>    arrow_size<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb713-12"><a href="#cb713-12" aria-hidden="true" tabindex="-1"></a>    min_mass<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb713-13"><a href="#cb713-13" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb713-14"><a href="#cb713-14" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>plt_size,</span>
<span id="cb713-15"><a href="#cb713-15" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"pacmap-stream"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb713-16"><a href="#cb713-16" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>computing velocity embedding
    finished (0:00:00) --&gt; added
    'velocity_pacmap', embedded velocity vectors (adata.obsm)</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_pacmap-stream-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-61-output-3.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="61">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb716"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb716-1"><a href="#cb716-1" aria-hidden="true" tabindex="-1"></a>sc.tl.embedding_density(velocity_data, basis<span class="op">=</span><span class="st">"pacmap"</span>, groupby<span class="op">=</span><span class="st">"libname"</span>)</span>
<span id="cb716-2"><a href="#cb716-2" aria-hidden="true" tabindex="-1"></a>sc.pl.embedding_density(</span>
<span id="cb716-3"><a href="#cb716-3" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb716-4"><a href="#cb716-4" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"pacmap"</span>,</span>
<span id="cb716-5"><a href="#cb716-5" aria-hidden="true" tabindex="-1"></a>    key<span class="op">=</span><span class="st">"pacmap_density_libname"</span>,</span>
<span id="cb716-6"><a href="#cb716-6" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb716-7"><a href="#cb716-7" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"pacmap-libname"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb716-8"><a href="#cb716-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>computing density on 'pacmap'</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../output/figures/velocity-signalling/pacmap_density_libname_pacmap-libname-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scanpy/plotting/_tools/__init__.py:1450: MatplotlibDeprecationWarning: The get_cmap function was deprecated in Matplotlib 3.7 and will be removed two minor releases later. Use ``matplotlib.colormaps[name]`` or ``matplotlib.colormaps.get_cmap(obj)`` instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scanpy/plotting/_tools/scatterplots.py:163: MatplotlibDeprecationWarning: The get_cmap function was deprecated in Matplotlib 3.7 and will be removed two minor releases later. Use ``matplotlib.colormaps[name]`` or ``matplotlib.colormaps.get_cmap(obj)`` instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scanpy/plotting/_tools/scatterplots.py:163: MatplotlibDeprecationWarning: The get_cmap function was deprecated in Matplotlib 3.7 and will be removed two minor releases later. Use ``matplotlib.colormaps[name]`` or ``matplotlib.colormaps.get_cmap(obj)`` instead.</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-62-output-4.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="62">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb720"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb720-1"><a href="#cb720-1" aria-hidden="true" tabindex="-1"></a>scv.pl.velocity_embedding_stream(</span>
<span id="cb720-2"><a href="#cb720-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb720-3"><a href="#cb720-3" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span><span class="st">"libname"</span>,</span>
<span id="cb720-4"><a href="#cb720-4" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"pacmap"</span>,</span>
<span id="cb720-5"><a href="#cb720-5" aria-hidden="true" tabindex="-1"></a>    legend_loc<span class="op">=</span><span class="st">"none"</span>,</span>
<span id="cb720-6"><a href="#cb720-6" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.3</span>,</span>
<span id="cb720-7"><a href="#cb720-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb720-8"><a href="#cb720-8" aria-hidden="true" tabindex="-1"></a>    arrow_color<span class="op">=</span><span class="st">"darkslategrey"</span>,</span>
<span id="cb720-9"><a href="#cb720-9" aria-hidden="true" tabindex="-1"></a>    density<span class="op">=</span><span class="fl">0.9</span>,</span>
<span id="cb720-10"><a href="#cb720-10" aria-hidden="true" tabindex="-1"></a>    linewidth<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb720-11"><a href="#cb720-11" aria-hidden="true" tabindex="-1"></a>    arrow_size<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb720-12"><a href="#cb720-12" aria-hidden="true" tabindex="-1"></a>    min_mass<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb720-13"><a href="#cb720-13" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb720-14"><a href="#cb720-14" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>plt_size,</span>
<span id="cb720-15"><a href="#cb720-15" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"pacmap-libname"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb720-16"><a href="#cb720-16" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_pacmap-libname-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-63-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="63">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb722"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb722-1"><a href="#cb722-1" aria-hidden="true" tabindex="-1"></a>scv.pl.velocity_embedding_stream(</span>
<span id="cb722-2"><a href="#cb722-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb722-3"><a href="#cb722-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"umap"</span>,</span>
<span id="cb722-4"><a href="#cb722-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> npep <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names],</span>
<span id="cb722-5"><a href="#cb722-5" aria-hidden="true" tabindex="-1"></a>    legend_loc<span class="op">=</span><span class="st">"none"</span>,</span>
<span id="cb722-6"><a href="#cb722-6" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.6</span>,</span>
<span id="cb722-7"><a href="#cb722-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb722-8"><a href="#cb722-8" aria-hidden="true" tabindex="-1"></a>    arrow_color<span class="op">=</span><span class="st">"darkslategrey"</span>,</span>
<span id="cb722-9"><a href="#cb722-9" aria-hidden="true" tabindex="-1"></a>    density<span class="op">=</span><span class="fl">0.9</span>,</span>
<span id="cb722-10"><a href="#cb722-10" aria-hidden="true" tabindex="-1"></a>    linewidth<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb722-11"><a href="#cb722-11" aria-hidden="true" tabindex="-1"></a>    arrow_size<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb722-12"><a href="#cb722-12" aria-hidden="true" tabindex="-1"></a>    min_mass<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb722-13"><a href="#cb722-13" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb722-14"><a href="#cb722-14" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"gnuplot2"</span>,</span>
<span id="cb722-15"><a href="#cb722-15" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>plt_size,</span>
<span id="cb722-16"><a href="#cb722-16" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="dv">40</span>,</span>
<span id="cb722-17"><a href="#cb722-17" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">4</span>,</span>
<span id="cb722-18"><a href="#cb722-18" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-npep-stream"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb722-19"><a href="#cb722-19" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_-npep-stream-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-64-output-3.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="64">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb725"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb725-1"><a href="#cb725-1" aria-hidden="true" tabindex="-1"></a>scv.pl.velocity_embedding_stream(</span>
<span id="cb725-2"><a href="#cb725-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb725-3"><a href="#cb725-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"umap"</span>,</span>
<span id="cb725-4"><a href="#cb725-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> npr <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names],</span>
<span id="cb725-5"><a href="#cb725-5" aria-hidden="true" tabindex="-1"></a>    legend_loc<span class="op">=</span><span class="st">"none"</span>,</span>
<span id="cb725-6"><a href="#cb725-6" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.6</span>,</span>
<span id="cb725-7"><a href="#cb725-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb725-8"><a href="#cb725-8" aria-hidden="true" tabindex="-1"></a>    arrow_color<span class="op">=</span><span class="st">"darkslategrey"</span>,</span>
<span id="cb725-9"><a href="#cb725-9" aria-hidden="true" tabindex="-1"></a>    density<span class="op">=</span><span class="fl">0.9</span>,</span>
<span id="cb725-10"><a href="#cb725-10" aria-hidden="true" tabindex="-1"></a>    linewidth<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb725-11"><a href="#cb725-11" aria-hidden="true" tabindex="-1"></a>    arrow_size<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb725-12"><a href="#cb725-12" aria-hidden="true" tabindex="-1"></a>    min_mass<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb725-13"><a href="#cb725-13" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb725-14"><a href="#cb725-14" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"gnuplot2"</span>,</span>
<span id="cb725-15"><a href="#cb725-15" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>plt_size,</span>
<span id="cb725-16"><a href="#cb725-16" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="dv">40</span>,</span>
<span id="cb725-17"><a href="#cb725-17" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">4</span>,</span>
<span id="cb725-18"><a href="#cb725-18" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-npr-stream"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb725-19"><a href="#cb725-19" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_-npr-stream-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-65-output-5.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="65">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb730"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb730-1"><a href="#cb730-1" aria-hidden="true" tabindex="-1"></a>scv.pl.velocity_embedding_stream(</span>
<span id="cb730-2"><a href="#cb730-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb730-3"><a href="#cb730-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"umap"</span>,</span>
<span id="cb730-4"><a href="#cb730-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> nmr <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names],</span>
<span id="cb730-5"><a href="#cb730-5" aria-hidden="true" tabindex="-1"></a>    legend_loc<span class="op">=</span><span class="st">"none"</span>,</span>
<span id="cb730-6"><a href="#cb730-6" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.6</span>,</span>
<span id="cb730-7"><a href="#cb730-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb730-8"><a href="#cb730-8" aria-hidden="true" tabindex="-1"></a>    arrow_color<span class="op">=</span><span class="st">"darkslategrey"</span>,</span>
<span id="cb730-9"><a href="#cb730-9" aria-hidden="true" tabindex="-1"></a>    density<span class="op">=</span><span class="fl">0.9</span>,</span>
<span id="cb730-10"><a href="#cb730-10" aria-hidden="true" tabindex="-1"></a>    linewidth<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb730-11"><a href="#cb730-11" aria-hidden="true" tabindex="-1"></a>    arrow_size<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb730-12"><a href="#cb730-12" aria-hidden="true" tabindex="-1"></a>    min_mass<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb730-13"><a href="#cb730-13" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb730-14"><a href="#cb730-14" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"gnuplot2"</span>,</span>
<span id="cb730-15"><a href="#cb730-15" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>plt_size,</span>
<span id="cb730-16"><a href="#cb730-16" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="dv">40</span>,</span>
<span id="cb730-17"><a href="#cb730-17" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">4</span>,</span>
<span id="cb730-18"><a href="#cb730-18" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-nmr-stream"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb730-19"><a href="#cb730-19" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_-nmr-stream-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-66-output-24.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="66">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb754"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb754-1"><a href="#cb754-1" aria-hidden="true" tabindex="-1"></a>scv.pl.velocity_embedding_stream(</span>
<span id="cb754-2"><a href="#cb754-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb754-3"><a href="#cb754-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"umap"</span>,</span>
<span id="cb754-4"><a href="#cb754-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> genes_embed <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names],</span>
<span id="cb754-5"><a href="#cb754-5" aria-hidden="true" tabindex="-1"></a>    legend_loc<span class="op">=</span><span class="st">"none"</span>,</span>
<span id="cb754-6"><a href="#cb754-6" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.6</span>,</span>
<span id="cb754-7"><a href="#cb754-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb754-8"><a href="#cb754-8" aria-hidden="true" tabindex="-1"></a>    arrow_color<span class="op">=</span><span class="st">"darkslategrey"</span>,</span>
<span id="cb754-9"><a href="#cb754-9" aria-hidden="true" tabindex="-1"></a>    density<span class="op">=</span><span class="fl">0.9</span>,</span>
<span id="cb754-10"><a href="#cb754-10" aria-hidden="true" tabindex="-1"></a>    linewidth<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb754-11"><a href="#cb754-11" aria-hidden="true" tabindex="-1"></a>    arrow_size<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb754-12"><a href="#cb754-12" aria-hidden="true" tabindex="-1"></a>    min_mass<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb754-13"><a href="#cb754-13" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb754-14"><a href="#cb754-14" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"gnuplot2"</span>,</span>
<span id="cb754-15"><a href="#cb754-15" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>plt_size,</span>
<span id="cb754-16"><a href="#cb754-16" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="dv">40</span>,</span>
<span id="cb754-17"><a href="#cb754-17" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">4</span>,</span>
<span id="cb754-18"><a href="#cb754-18" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-adgen-stream"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb754-19"><a href="#cb754-19" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_-adgen-stream-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-67-output-49.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="67">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb803"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb803-1"><a href="#cb803-1" aria-hidden="true" tabindex="-1"></a>scv.pl.velocity_embedding_stream(</span>
<span id="cb803-2"><a href="#cb803-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb803-3"><a href="#cb803-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"umap"</span>,</span>
<span id="cb803-4"><a href="#cb803-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> mitochondrial <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names],</span>
<span id="cb803-5"><a href="#cb803-5" aria-hidden="true" tabindex="-1"></a>    legend_loc<span class="op">=</span><span class="st">"none"</span>,</span>
<span id="cb803-6"><a href="#cb803-6" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.6</span>,</span>
<span id="cb803-7"><a href="#cb803-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb803-8"><a href="#cb803-8" aria-hidden="true" tabindex="-1"></a>    arrow_color<span class="op">=</span><span class="st">"darkslategrey"</span>,</span>
<span id="cb803-9"><a href="#cb803-9" aria-hidden="true" tabindex="-1"></a>    density<span class="op">=</span><span class="fl">0.9</span>,</span>
<span id="cb803-10"><a href="#cb803-10" aria-hidden="true" tabindex="-1"></a>    linewidth<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb803-11"><a href="#cb803-11" aria-hidden="true" tabindex="-1"></a>    arrow_size<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb803-12"><a href="#cb803-12" aria-hidden="true" tabindex="-1"></a>    min_mass<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb803-13"><a href="#cb803-13" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb803-14"><a href="#cb803-14" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"gnuplot2"</span>,</span>
<span id="cb803-15"><a href="#cb803-15" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>plt_size,</span>
<span id="cb803-16"><a href="#cb803-16" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="dv">40</span>,</span>
<span id="cb803-17"><a href="#cb803-17" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">4</span>,</span>
<span id="cb803-18"><a href="#cb803-18" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-mitochondrial-stream"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb803-19"><a href="#cb803-19" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_-mitochondrial-stream-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-68-output-16.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="68">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb819"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb819-1"><a href="#cb819-1" aria-hidden="true" tabindex="-1"></a>scv.pl.velocity_embedding_stream(</span>
<span id="cb819-2"><a href="#cb819-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb819-3"><a href="#cb819-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"pacmap"</span>,</span>
<span id="cb819-4"><a href="#cb819-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> npep <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names],</span>
<span id="cb819-5"><a href="#cb819-5" aria-hidden="true" tabindex="-1"></a>    legend_loc<span class="op">=</span><span class="st">"none"</span>,</span>
<span id="cb819-6"><a href="#cb819-6" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.3</span>,</span>
<span id="cb819-7"><a href="#cb819-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb819-8"><a href="#cb819-8" aria-hidden="true" tabindex="-1"></a>    arrow_color<span class="op">=</span><span class="st">"darkslategrey"</span>,</span>
<span id="cb819-9"><a href="#cb819-9" aria-hidden="true" tabindex="-1"></a>    density<span class="op">=</span><span class="fl">0.9</span>,</span>
<span id="cb819-10"><a href="#cb819-10" aria-hidden="true" tabindex="-1"></a>    linewidth<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb819-11"><a href="#cb819-11" aria-hidden="true" tabindex="-1"></a>    arrow_size<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb819-12"><a href="#cb819-12" aria-hidden="true" tabindex="-1"></a>    min_mass<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb819-13"><a href="#cb819-13" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb819-14"><a href="#cb819-14" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"gnuplot2"</span>,</span>
<span id="cb819-15"><a href="#cb819-15" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>plt_size,</span>
<span id="cb819-16"><a href="#cb819-16" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="dv">40</span>,</span>
<span id="cb819-17"><a href="#cb819-17" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">4</span>,</span>
<span id="cb819-18"><a href="#cb819-18" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"pacmap-npep-stream"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb819-19"><a href="#cb819-19" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_pacmap-npep-stream-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-69-output-3.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="69">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb822"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb822-1"><a href="#cb822-1" aria-hidden="true" tabindex="-1"></a>scv.pl.velocity_embedding_stream(</span>
<span id="cb822-2"><a href="#cb822-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb822-3"><a href="#cb822-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"pacmap"</span>,</span>
<span id="cb822-4"><a href="#cb822-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> npr <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names],</span>
<span id="cb822-5"><a href="#cb822-5" aria-hidden="true" tabindex="-1"></a>    legend_loc<span class="op">=</span><span class="st">"none"</span>,</span>
<span id="cb822-6"><a href="#cb822-6" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.3</span>,</span>
<span id="cb822-7"><a href="#cb822-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb822-8"><a href="#cb822-8" aria-hidden="true" tabindex="-1"></a>    arrow_color<span class="op">=</span><span class="st">"darkslategrey"</span>,</span>
<span id="cb822-9"><a href="#cb822-9" aria-hidden="true" tabindex="-1"></a>    density<span class="op">=</span><span class="fl">0.9</span>,</span>
<span id="cb822-10"><a href="#cb822-10" aria-hidden="true" tabindex="-1"></a>    linewidth<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb822-11"><a href="#cb822-11" aria-hidden="true" tabindex="-1"></a>    arrow_size<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb822-12"><a href="#cb822-12" aria-hidden="true" tabindex="-1"></a>    min_mass<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb822-13"><a href="#cb822-13" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb822-14"><a href="#cb822-14" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"gnuplot2"</span>,</span>
<span id="cb822-15"><a href="#cb822-15" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>plt_size,</span>
<span id="cb822-16"><a href="#cb822-16" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="dv">40</span>,</span>
<span id="cb822-17"><a href="#cb822-17" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">4</span>,</span>
<span id="cb822-18"><a href="#cb822-18" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"pacmap-npr-stream"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb822-19"><a href="#cb822-19" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_pacmap-npr-stream-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-70-output-5.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="70">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb827"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb827-1"><a href="#cb827-1" aria-hidden="true" tabindex="-1"></a>scv.pl.velocity_embedding_stream(</span>
<span id="cb827-2"><a href="#cb827-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb827-3"><a href="#cb827-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"pacmap"</span>,</span>
<span id="cb827-4"><a href="#cb827-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> nmr <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names],</span>
<span id="cb827-5"><a href="#cb827-5" aria-hidden="true" tabindex="-1"></a>    legend_loc<span class="op">=</span><span class="st">"none"</span>,</span>
<span id="cb827-6"><a href="#cb827-6" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.3</span>,</span>
<span id="cb827-7"><a href="#cb827-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb827-8"><a href="#cb827-8" aria-hidden="true" tabindex="-1"></a>    arrow_color<span class="op">=</span><span class="st">"darkslategrey"</span>,</span>
<span id="cb827-9"><a href="#cb827-9" aria-hidden="true" tabindex="-1"></a>    density<span class="op">=</span><span class="fl">0.9</span>,</span>
<span id="cb827-10"><a href="#cb827-10" aria-hidden="true" tabindex="-1"></a>    linewidth<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb827-11"><a href="#cb827-11" aria-hidden="true" tabindex="-1"></a>    arrow_size<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb827-12"><a href="#cb827-12" aria-hidden="true" tabindex="-1"></a>    min_mass<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb827-13"><a href="#cb827-13" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb827-14"><a href="#cb827-14" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"gnuplot2"</span>,</span>
<span id="cb827-15"><a href="#cb827-15" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>plt_size,</span>
<span id="cb827-16"><a href="#cb827-16" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="dv">40</span>,</span>
<span id="cb827-17"><a href="#cb827-17" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">4</span>,</span>
<span id="cb827-18"><a href="#cb827-18" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"pacmap-nmr-stream"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb827-19"><a href="#cb827-19" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_pacmap-nmr-stream-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-71-output-24.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="71">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb851"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb851-1"><a href="#cb851-1" aria-hidden="true" tabindex="-1"></a>scv.pl.velocity_embedding_stream(</span>
<span id="cb851-2"><a href="#cb851-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb851-3"><a href="#cb851-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"pacmap"</span>,</span>
<span id="cb851-4"><a href="#cb851-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> genes_embed <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names],</span>
<span id="cb851-5"><a href="#cb851-5" aria-hidden="true" tabindex="-1"></a>    legend_loc<span class="op">=</span><span class="st">"none"</span>,</span>
<span id="cb851-6"><a href="#cb851-6" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.3</span>,</span>
<span id="cb851-7"><a href="#cb851-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb851-8"><a href="#cb851-8" aria-hidden="true" tabindex="-1"></a>    arrow_color<span class="op">=</span><span class="st">"darkslategrey"</span>,</span>
<span id="cb851-9"><a href="#cb851-9" aria-hidden="true" tabindex="-1"></a>    density<span class="op">=</span><span class="fl">0.9</span>,</span>
<span id="cb851-10"><a href="#cb851-10" aria-hidden="true" tabindex="-1"></a>    linewidth<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb851-11"><a href="#cb851-11" aria-hidden="true" tabindex="-1"></a>    arrow_size<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb851-12"><a href="#cb851-12" aria-hidden="true" tabindex="-1"></a>    min_mass<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb851-13"><a href="#cb851-13" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb851-14"><a href="#cb851-14" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"gnuplot2"</span>,</span>
<span id="cb851-15"><a href="#cb851-15" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>plt_size,</span>
<span id="cb851-16"><a href="#cb851-16" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="dv">40</span>,</span>
<span id="cb851-17"><a href="#cb851-17" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">4</span>,</span>
<span id="cb851-18"><a href="#cb851-18" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"pacmap-adgen-stream"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb851-19"><a href="#cb851-19" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_pacmap-adgen-stream-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-72-output-49.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="72">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb900"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb900-1"><a href="#cb900-1" aria-hidden="true" tabindex="-1"></a>scv.pl.velocity_embedding_stream(</span>
<span id="cb900-2"><a href="#cb900-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb900-3"><a href="#cb900-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"pacmap"</span>,</span>
<span id="cb900-4"><a href="#cb900-4" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> mitochondrial <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names],</span>
<span id="cb900-5"><a href="#cb900-5" aria-hidden="true" tabindex="-1"></a>    legend_loc<span class="op">=</span><span class="st">"none"</span>,</span>
<span id="cb900-6"><a href="#cb900-6" aria-hidden="true" tabindex="-1"></a>    alpha<span class="op">=</span><span class="fl">0.3</span>,</span>
<span id="cb900-7"><a href="#cb900-7" aria-hidden="true" tabindex="-1"></a>    legend_fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb900-8"><a href="#cb900-8" aria-hidden="true" tabindex="-1"></a>    arrow_color<span class="op">=</span><span class="st">"darkslategrey"</span>,</span>
<span id="cb900-9"><a href="#cb900-9" aria-hidden="true" tabindex="-1"></a>    density<span class="op">=</span><span class="fl">0.9</span>,</span>
<span id="cb900-10"><a href="#cb900-10" aria-hidden="true" tabindex="-1"></a>    linewidth<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb900-11"><a href="#cb900-11" aria-hidden="true" tabindex="-1"></a>    arrow_size<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb900-12"><a href="#cb900-12" aria-hidden="true" tabindex="-1"></a>    min_mass<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb900-13"><a href="#cb900-13" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb900-14"><a href="#cb900-14" aria-hidden="true" tabindex="-1"></a>    palette<span class="op">=</span><span class="st">"gnuplot2"</span>,</span>
<span id="cb900-15"><a href="#cb900-15" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>plt_size,</span>
<span id="cb900-16"><a href="#cb900-16" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="dv">40</span>,</span>
<span id="cb900-17"><a href="#cb900-17" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">4</span>,</span>
<span id="cb900-18"><a href="#cb900-18" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"pacmap-mitochondrial-stream"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb900-19"><a href="#cb900-19" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_pacmap-mitochondrial-stream-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-73-output-16.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="73">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb916"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb916-1"><a href="#cb916-1" aria-hidden="true" tabindex="-1"></a>velocity_data.write(</span>
<span id="cb916-2"><a href="#cb916-2" aria-hidden="true" tabindex="-1"></a>    <span class="ss">f"/data/</span><span class="sc">{</span>bioproject<span class="sc">}</span><span class="ss">/data/velocity-signalling/</span><span class="sc">{</span>bioproject<span class="sc">}</span><span class="ss">-velocity.h5ad"</span></span>
<span id="cb916-3"><a href="#cb916-3" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="cellrank-analysis" class="level3">
<h3 class="anchored" data-anchor-id="cellrank-analysis">CellRank analysis</h3>
<div class="cell" data-execution_count="74">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb917"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb917-1"><a href="#cb917-1" aria-hidden="true" tabindex="-1"></a>cr.tl.terminal_states(</span>
<span id="cb917-2"><a href="#cb917-2" aria-hidden="true" tabindex="-1"></a>    velocity_data, cluster_key<span class="op">=</span><span class="st">"k_tree"</span>, weight_connectivities<span class="op">=</span><span class="fl">0.2</span>, softmax_scale<span class="op">=</span><span class="dv">4</span></span>
<span id="cb917-3"><a href="#cb917-3" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb917-4"><a href="#cb917-4" aria-hidden="true" tabindex="-1"></a>cr.tl.initial_states(velocity_data, cluster_key<span class="op">=</span><span class="st">"k_tree"</span>)</span>
<span id="cb917-5"><a href="#cb917-5" aria-hidden="true" tabindex="-1"></a>cr.tl.lineages(velocity_data)</span>
<span id="cb917-6"><a href="#cb917-6" aria-hidden="true" tabindex="-1"></a>cr.pl.lineages(velocity_data, same_plot<span class="op">=</span><span class="va">True</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/tmp/ipykernel_46/1769889968.py:1: DeprecationWarning: `cellrank.tl.terminal_states` will be removed in version `2.0`. Please use the `cellrank.kernels` or `cellrank.estimators` interface instead.
/opt/python/3.8.8/lib/python3.8/site-packages/cellrank/tl/_init_term_states.py:156: DeprecationWarning: `cellrank.tl.transition_matrix` will be removed in version `2.0`. Please use the `cellrank.kernels` or `cellrank.estimators` interface instead.</code></pre>
</div>
<div class="cell-output cell-output-display">
<script type="application/vnd.jupyter.widget-view+json">
{"model_id":"905acb3ffbdb4f6190cfe472223c1384","version_major":2,"version_minor":0}
</script>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/tmp/ipykernel_46/1769889968.py:4: DeprecationWarning: `cellrank.tl.initial_states` will be removed in version `2.0`. Please use the `cellrank.kernels` or `cellrank.estimators` interface instead.
/opt/python/3.8.8/lib/python3.8/site-packages/cellrank/tl/_init_term_states.py:156: DeprecationWarning: `cellrank.tl.transition_matrix` will be removed in version `2.0`. Please use the `cellrank.kernels` or `cellrank.estimators` interface instead.</code></pre>
</div>
<div class="cell-output cell-output-display">
<script type="application/vnd.jupyter.widget-view+json">
{"model_id":"3a135c30cf4646e598972ad62b4f4c0d","version_major":2,"version_minor":0}
</script>
</div>
<div class="cell-output cell-output-display">
<script type="application/vnd.jupyter.widget-view+json">
{"model_id":"d0e5e1f50bbe43b282f54f2a18c61104","version_major":2,"version_minor":0}
</script>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: There is only `1` terminal state, all cells will have probability `1` of going there</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: Unable to import petsc4py. For installation, please refer to: https://petsc4py.readthedocs.io/en/stable/install.html.
Defaulting to `'gmres'` solver.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/tmp/ipykernel_46/1769889968.py:5: DeprecationWarning: `cellrank.tl.lineages` will be removed in version `2.0`. Please use the `cellrank.kernels` or `cellrank.estimators` interface instead.</code></pre>
</div>
<div class="cell-output cell-output-display">
<script type="application/vnd.jupyter.widget-view+json">
{"model_id":"5cf43f14bce54546a395790fc0935549","version_major":2,"version_minor":0}
</script>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-75-output-11.png" class="img-fluid"></p>
</div>
</div>
<section id="latent-time" class="level4">
<h4 class="anchored" data-anchor-id="latent-time">Latent time</h4>
<div class="cell" data-execution_count="75">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb924"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb924-1"><a href="#cb924-1" aria-hidden="true" tabindex="-1"></a>scv.tl.latent_time(</span>
<span id="cb924-2"><a href="#cb924-2" aria-hidden="true" tabindex="-1"></a>    velocity_data, root_key<span class="op">=</span><span class="st">"initial_states_probs"</span>, end_key<span class="op">=</span><span class="st">"terminal_states_probs"</span></span>
<span id="cb924-3"><a href="#cb924-3" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb924-4"><a href="#cb924-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb924-5"><a href="#cb924-5" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(</span>
<span id="cb924-6"><a href="#cb924-6" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb924-7"><a href="#cb924-7" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"umap"</span>,</span>
<span id="cb924-8"><a href="#cb924-8" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span><span class="st">"latent_time"</span>,</span>
<span id="cb924-9"><a href="#cb924-9" aria-hidden="true" tabindex="-1"></a>    color_map<span class="op">=</span><span class="st">"gnuplot"</span>,</span>
<span id="cb924-10"><a href="#cb924-10" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="dv">80</span>,</span>
<span id="cb924-11"><a href="#cb924-11" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"umap-latent-time"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb924-12"><a href="#cb924-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb924-13"><a href="#cb924-13" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(</span>
<span id="cb924-14"><a href="#cb924-14" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb924-15"><a href="#cb924-15" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"pacmap"</span>,</span>
<span id="cb924-16"><a href="#cb924-16" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span><span class="st">"latent_time"</span>,</span>
<span id="cb924-17"><a href="#cb924-17" aria-hidden="true" tabindex="-1"></a>    color_map<span class="op">=</span><span class="st">"gnuplot"</span>,</span>
<span id="cb924-18"><a href="#cb924-18" aria-hidden="true" tabindex="-1"></a>    size<span class="op">=</span><span class="dv">80</span>,</span>
<span id="cb924-19"><a href="#cb924-19" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"pacmap-latent-time"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb924-20"><a href="#cb924-20" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>computing latent time using initial_states_probs, terminal_states_probs as prior
    finished (0:00:00) --&gt; added 
    'latent_time', shared time (adata.obs)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_umap-latent-time-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-76-output-4.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_pacmap-latent-time-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-76-output-7.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="76">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb930"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb930-1"><a href="#cb930-1" aria-hidden="true" tabindex="-1"></a>top_genes <span class="op">=</span> velocity_data.var[<span class="st">"fit_likelihood"</span>].sort_values(ascending<span class="op">=</span><span class="va">False</span>).index[:<span class="dv">300</span>]</span>
<span id="cb930-2"><a href="#cb930-2" aria-hidden="true" tabindex="-1"></a>scv.pl.heatmap(</span>
<span id="cb930-3"><a href="#cb930-3" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb930-4"><a href="#cb930-4" aria-hidden="true" tabindex="-1"></a>    var_names<span class="op">=</span>top_genes,</span>
<span id="cb930-5"><a href="#cb930-5" aria-hidden="true" tabindex="-1"></a>    sortby<span class="op">=</span><span class="st">"latent_time"</span>,</span>
<span id="cb930-6"><a href="#cb930-6" aria-hidden="true" tabindex="-1"></a>    col_color<span class="op">=</span><span class="st">"libname"</span>,</span>
<span id="cb930-7"><a href="#cb930-7" aria-hidden="true" tabindex="-1"></a>    n_convolve<span class="op">=</span><span class="dv">100</span>,</span>
<span id="cb930-8"><a href="#cb930-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-77-output-1.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="77">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb931"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb931-1"><a href="#cb931-1" aria-hidden="true" tabindex="-1"></a>genes_embed <span class="op">=</span> [g <span class="cf">for</span> g <span class="kw">in</span> genes_embed <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names]</span>
<span id="cb931-2"><a href="#cb931-2" aria-hidden="true" tabindex="-1"></a>npr <span class="op">=</span> [g <span class="cf">for</span> g <span class="kw">in</span> npr <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names]</span>
<span id="cb931-3"><a href="#cb931-3" aria-hidden="true" tabindex="-1"></a>nmr <span class="op">=</span> [g <span class="cf">for</span> g <span class="kw">in</span> nmr <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names]</span>
<span id="cb931-4"><a href="#cb931-4" aria-hidden="true" tabindex="-1"></a>mitochondrial <span class="op">=</span> [g <span class="cf">for</span> g <span class="kw">in</span> mitochondrial <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="78">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb932"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb932-1"><a href="#cb932-1" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(velocity_data, mitochondrial[:<span class="dv">19</span>], frameon<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb932-2"><a href="#cb932-2" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(velocity_data, x<span class="op">=</span><span class="st">"latent_time"</span>, y<span class="op">=</span>mitochondrial[:<span class="dv">19</span>], frameon<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-79-output-1.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-79-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="79">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb933"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb933-1"><a href="#cb933-1" aria-hidden="true" tabindex="-1"></a><span class="co"># scv.pl.scatter(velocity_data, mitochondrial[20:], frameon=False)</span></span>
<span id="cb933-2"><a href="#cb933-2" aria-hidden="true" tabindex="-1"></a><span class="co"># scv.pl.scatter(velocity_data, x="latent_time", y=mitochondrial[20:], frameon=False)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-execution_count="80">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb934"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb934-1"><a href="#cb934-1" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(velocity_data, genes_embed[:<span class="dv">19</span>], frameon<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb934-2"><a href="#cb934-2" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(velocity_data, x<span class="op">=</span><span class="st">"latent_time"</span>, y<span class="op">=</span>genes_embed[:<span class="dv">19</span>], frameon<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-81-output-1.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-81-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="81">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb935"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb935-1"><a href="#cb935-1" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(velocity_data, genes_embed[<span class="dv">20</span>:<span class="dv">39</span>], frameon<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb935-2"><a href="#cb935-2" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(velocity_data, x<span class="op">=</span><span class="st">"latent_time"</span>, y<span class="op">=</span>genes_embed[<span class="dv">20</span>:<span class="dv">39</span>], frameon<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-82-output-1.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-82-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="82">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb936"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb936-1"><a href="#cb936-1" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(velocity_data, genes_embed[<span class="dv">40</span>:], frameon<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb936-2"><a href="#cb936-2" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(velocity_data, x<span class="op">=</span><span class="st">"latent_time"</span>, y<span class="op">=</span>genes_embed[<span class="dv">40</span>:], frameon<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-83-output-1.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-83-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="83">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb937"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb937-1"><a href="#cb937-1" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(velocity_data, npr[:<span class="dv">19</span>], frameon<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb937-2"><a href="#cb937-2" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(velocity_data, x<span class="op">=</span><span class="st">"latent_time"</span>, y<span class="op">=</span>npr[:<span class="dv">19</span>], frameon<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-84-output-1.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-84-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="84">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb938"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb938-1"><a href="#cb938-1" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(velocity_data, nmr[:<span class="dv">19</span>], frameon<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb938-2"><a href="#cb938-2" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(velocity_data, x<span class="op">=</span><span class="st">"latent_time"</span>, y<span class="op">=</span>nmr[:<span class="dv">19</span>], frameon<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-85-output-1.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-85-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="85">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb939"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb939-1"><a href="#cb939-1" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(velocity_data, nmr[<span class="dv">20</span>:], frameon<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb939-2"><a href="#cb939-2" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(velocity_data, x<span class="op">=</span><span class="st">"latent_time"</span>, y<span class="op">=</span>nmr[<span class="dv">20</span>:], frameon<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-86-output-1.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-86-output-2.png" class="img-fluid"></p>
</div>
</div>
</section>
<section id="top-likelihood-genes" class="level4">
<h4 class="anchored" data-anchor-id="top-likelihood-genes">Top-likelihood genes</h4>
<div class="cell" data-execution_count="86">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb940"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb940-1"><a href="#cb940-1" aria-hidden="true" tabindex="-1"></a>top_genes <span class="op">=</span> velocity_data.var[<span class="st">"fit_likelihood"</span>].sort_values(ascending<span class="op">=</span><span class="va">False</span>).index</span>
<span id="cb940-2"><a href="#cb940-2" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(velocity_data, basis<span class="op">=</span>top_genes[:<span class="dv">15</span>], ncols<span class="op">=</span><span class="dv">5</span>, frameon<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-87-output-1.png" class="img-fluid"></p>
</div>
</div>
</section>
<section id="condition-specific-top-likelihood-genes" class="level4">
<h4 class="anchored" data-anchor-id="condition-specific-top-likelihood-genes">Condition-specific top-likelihood genes</h4>
<div class="cell" data-execution_count="87">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb941"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb941-1"><a href="#cb941-1" aria-hidden="true" tabindex="-1"></a>scv.tl.rank_dynamical_genes(velocity_data, groupby<span class="op">=</span><span class="st">"libname"</span>)</span>
<span id="cb941-2"><a href="#cb941-2" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> scv.get_df(velocity_data, <span class="st">"rank_dynamical_genes/names"</span>)</span>
<span id="cb941-3"><a href="#cb941-3" aria-hidden="true" tabindex="-1"></a>df.head(<span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>ranking genes by cluster-specific likelihoods
    finished (0:00:00) --&gt; added 
    'rank_dynamical_genes', sorted scores by group ids (adata.uns)</code></pre>
</div>
<div class="cell-output cell-output-display" data-execution_count="87">

<div>

<table class="dataframe table table-sm table-striped">
  <thead>
    <tr>
      <th></th>
      <th>control</th>
      <th>hfd</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>Flt1</td>
      <td>Itih3</td>
    </tr>
    <tr>
      <th>1</th>
      <td>Itih3</td>
      <td>Flt1</td>
    </tr>
    <tr>
      <th>2</th>
      <td>Ahi1</td>
      <td>Ahi1</td>
    </tr>
    <tr>
      <th>3</th>
      <td>Ttc3</td>
      <td>Ttc3</td>
    </tr>
    <tr>
      <th>4</th>
      <td>Stmn2</td>
      <td>Mt2</td>
    </tr>
  </tbody>
</table>
</div>
</div>
</div>
<div class="cell" data-execution_count="88">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb943"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb943-1"><a href="#cb943-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> condition <span class="kw">in</span> [<span class="st">"control"</span>, <span class="st">"hfd"</span>]:</span>
<span id="cb943-2"><a href="#cb943-2" aria-hidden="true" tabindex="-1"></a>    scv.pl.scatter(velocity_data, df[condition][:<span class="dv">5</span>], ylabel<span class="op">=</span>condition, frameon<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-89-output-1.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-89-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="89">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb944"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb944-1"><a href="#cb944-1" aria-hidden="true" tabindex="-1"></a>scv.pl.velocity_embedding(</span>
<span id="cb944-2"><a href="#cb944-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb944-3"><a href="#cb944-3" aria-hidden="true" tabindex="-1"></a>    arrow_length<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb944-4"><a href="#cb944-4" aria-hidden="true" tabindex="-1"></a>    arrow_size<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb944-5"><a href="#cb944-5" aria-hidden="true" tabindex="-1"></a>    dpi<span class="op">=</span><span class="dv">120</span>,</span>
<span id="cb944-6"><a href="#cb944-6" aria-hidden="true" tabindex="-1"></a>    show<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb944-7"><a href="#cb944-7" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"embedding"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb944-8"><a href="#cb944-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_embedding-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-90-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="90">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb946"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb946-1"><a href="#cb946-1" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> velocity_data.var</span>
<span id="cb946-2"><a href="#cb946-2" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> df[(df[<span class="st">"fit_likelihood"</span>] <span class="op">&gt;</span> <span class="fl">0.1</span>) <span class="op">&amp;</span> df[<span class="st">"velocity_genes"</span>] <span class="op">==</span> <span class="va">True</span>]</span>
<span id="cb946-3"><a href="#cb946-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb946-4"><a href="#cb946-4" aria-hidden="true" tabindex="-1"></a>kwargs <span class="op">=</span> <span class="bu">dict</span>(xscale<span class="op">=</span><span class="st">"log"</span>, fontsize<span class="op">=</span><span class="dv">16</span>)</span>
<span id="cb946-5"><a href="#cb946-5" aria-hidden="true" tabindex="-1"></a><span class="cf">with</span> scv.GridSpec(ncols<span class="op">=</span><span class="dv">3</span>) <span class="im">as</span> pl:</span>
<span id="cb946-6"><a href="#cb946-6" aria-hidden="true" tabindex="-1"></a>    pl.hist(df[<span class="st">"fit_alpha"</span>], xlabel<span class="op">=</span><span class="st">"transcription rate"</span>, <span class="op">**</span>kwargs)</span>
<span id="cb946-7"><a href="#cb946-7" aria-hidden="true" tabindex="-1"></a>    pl.hist(</span>
<span id="cb946-8"><a href="#cb946-8" aria-hidden="true" tabindex="-1"></a>        df[<span class="st">"fit_beta"</span>] <span class="op">*</span> df[<span class="st">"fit_scaling"</span>],</span>
<span id="cb946-9"><a href="#cb946-9" aria-hidden="true" tabindex="-1"></a>        xlabel<span class="op">=</span><span class="st">"splicing rate"</span>,</span>
<span id="cb946-10"><a href="#cb946-10" aria-hidden="true" tabindex="-1"></a>        xticks<span class="op">=</span>[<span class="fl">0.1</span>, <span class="fl">0.4</span>, <span class="dv">1</span>],</span>
<span id="cb946-11"><a href="#cb946-11" aria-hidden="true" tabindex="-1"></a>        <span class="op">**</span>kwargs,</span>
<span id="cb946-12"><a href="#cb946-12" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb946-13"><a href="#cb946-13" aria-hidden="true" tabindex="-1"></a>    pl.hist(df[<span class="st">"fit_gamma"</span>], xlabel<span class="op">=</span><span class="st">"degradation rate"</span>, xticks<span class="op">=</span>[<span class="fl">0.1</span>, <span class="fl">0.4</span>, <span class="dv">1</span>], <span class="op">**</span>kwargs)</span>
<span id="cb946-14"><a href="#cb946-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb946-15"><a href="#cb946-15" aria-hidden="true" tabindex="-1"></a>scv.get_df(velocity_data, <span class="st">"fit*"</span>, dropna<span class="op">=</span><span class="va">True</span>).head(<span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-91-output-1.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-display" data-execution_count="90">

<div>

<table class="dataframe table table-sm table-striped">
  <thead>
    <tr>
      <th></th>
      <th>fit_r2</th>
      <th>fit_alpha</th>
      <th>fit_beta</th>
      <th>fit_gamma</th>
      <th>fit_t_</th>
      <th>fit_scaling</th>
      <th>fit_std_u</th>
      <th>fit_std_s</th>
      <th>fit_likelihood</th>
      <th>fit_u0</th>
      <th>fit_s0</th>
      <th>fit_pval_steady</th>
      <th>fit_steady_u</th>
      <th>fit_steady_s</th>
      <th>fit_variance</th>
      <th>fit_alignment_scaling</th>
    </tr>
    <tr>
      <th>mgi_symbol</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Dbi</th>
      <td>0.036290</td>
      <td>0.158232</td>
      <td>1.485399</td>
      <td>0.152388</td>
      <td>12.218659</td>
      <td>0.103452</td>
      <td>0.038359</td>
      <td>0.175197</td>
      <td>0.203915</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>2.584603e-01</td>
      <td>0.113017</td>
      <td>0.560284</td>
      <td>1.372637</td>
      <td>3.074815</td>
    </tr>
    <tr>
      <th>Scn1a</th>
      <td>0.022295</td>
      <td>0.010443</td>
      <td>0.081221</td>
      <td>0.121752</td>
      <td>20.571872</td>
      <td>4.234877</td>
      <td>0.056218</td>
      <td>0.020908</td>
      <td>0.162872</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.624862e-01</td>
      <td>0.170626</td>
      <td>0.061767</td>
      <td>1.856030</td>
      <td>4.988267</td>
    </tr>
    <tr>
      <th>Dnm1</th>
      <td>0.053593</td>
      <td>0.034144</td>
      <td>0.482970</td>
      <td>0.281713</td>
      <td>7.332070</td>
      <td>0.694333</td>
      <td>0.021172</td>
      <td>0.026257</td>
      <td>0.187012</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.112517e-01</td>
      <td>0.067623</td>
      <td>0.072274</td>
      <td>1.448264</td>
      <td>2.632057</td>
    </tr>
    <tr>
      <th>Stmn2</th>
      <td>0.188921</td>
      <td>0.043153</td>
      <td>0.529163</td>
      <td>0.200646</td>
      <td>12.435350</td>
      <td>0.656638</td>
      <td>0.030453</td>
      <td>0.053566</td>
      <td>0.209940</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.616186e-01</td>
      <td>0.086368</td>
      <td>0.155111</td>
      <td>1.241808</td>
      <td>3.165753</td>
    </tr>
    <tr>
      <th>Ndufb5</th>
      <td>0.025362</td>
      <td>0.057677</td>
      <td>0.785205</td>
      <td>0.101686</td>
      <td>9.237044</td>
      <td>0.176534</td>
      <td>0.017454</td>
      <td>0.056190</td>
      <td>0.203510</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.486024e-01</td>
      <td>0.054678</td>
      <td>0.194418</td>
      <td>1.231058</td>
      <td>3.769414</td>
    </tr>
    <tr>
      <th>Chd5</th>
      <td>0.132620</td>
      <td>0.079455</td>
      <td>1.230525</td>
      <td>0.216943</td>
      <td>7.379210</td>
      <td>0.644025</td>
      <td>0.023587</td>
      <td>0.056769</td>
      <td>0.200486</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>2.362802e-01</td>
      <td>0.057727</td>
      <td>0.160039</td>
      <td>1.024949</td>
      <td>1.955863</td>
    </tr>
    <tr>
      <th>Flt1</th>
      <td>0.469015</td>
      <td>0.003697</td>
      <td>0.019178</td>
      <td>0.244670</td>
      <td>13.132757</td>
      <td>25.003526</td>
      <td>0.067496</td>
      <td>0.004677</td>
      <td>0.274993</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.837602e-16</td>
      <td>0.151565</td>
      <td>0.011032</td>
      <td>0.624830</td>
      <td>2.321687</td>
    </tr>
    <tr>
      <th>Caly</th>
      <td>0.134080</td>
      <td>0.273853</td>
      <td>5.541302</td>
      <td>0.574985</td>
      <td>4.749272</td>
      <td>0.075596</td>
      <td>0.012763</td>
      <td>0.087231</td>
      <td>0.206636</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>2.705816e-01</td>
      <td>0.038448</td>
      <td>0.223368</td>
      <td>0.974781</td>
      <td>1.233382</td>
    </tr>
    <tr>
      <th>Atp1a3</th>
      <td>0.275698</td>
      <td>0.021252</td>
      <td>0.198042</td>
      <td>0.284806</td>
      <td>11.065362</td>
      <td>2.715145</td>
      <td>0.039520</td>
      <td>0.019614</td>
      <td>0.240880</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>2.275156e-02</td>
      <td>0.094179</td>
      <td>0.059065</td>
      <td>0.856330</td>
      <td>2.505980</td>
    </tr>
    <tr>
      <th>Mt2</th>
      <td>0.011480</td>
      <td>0.223838</td>
      <td>2.861114</td>
      <td>0.159708</td>
      <td>7.489395</td>
      <td>0.074012</td>
      <td>0.028392</td>
      <td>0.181258</td>
      <td>0.192407</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>3.676493e-01</td>
      <td>0.082007</td>
      <td>0.612118</td>
      <td>1.394187</td>
      <td>2.231341</td>
    </tr>
    <tr>
      <th>Ahi1</th>
      <td>0.371978</td>
      <td>0.187694</td>
      <td>0.300541</td>
      <td>0.369701</td>
      <td>15.914042</td>
      <td>0.626090</td>
      <td>0.145734</td>
      <td>0.104746</td>
      <td>0.259145</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>3.356567e-01</td>
      <td>0.574719</td>
      <td>0.372839</td>
      <td>0.823542</td>
      <td>2.391508</td>
    </tr>
    <tr>
      <th>Syt1</th>
      <td>0.044454</td>
      <td>0.036555</td>
      <td>0.029709</td>
      <td>0.116967</td>
      <td>18.588958</td>
      <td>6.547850</td>
      <td>0.461092</td>
      <td>0.064551</td>
      <td>0.126273</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>4.019544e-01</td>
      <td>1.998606</td>
      <td>0.198615</td>
      <td>2.847291</td>
      <td>4.888983</td>
    </tr>
    <tr>
      <th>Camk2b</th>
      <td>0.022313</td>
      <td>0.033671</td>
      <td>0.376858</td>
      <td>0.127972</td>
      <td>5.912110</td>
      <td>2.543035</td>
      <td>0.036532</td>
      <td>0.024888</td>
      <td>0.157463</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>2.974212e-01</td>
      <td>0.096872</td>
      <td>0.078892</td>
      <td>1.200822</td>
      <td>1.807772</td>
    </tr>
    <tr>
      <th>Npm1</th>
      <td>0.089143</td>
      <td>0.036156</td>
      <td>0.580834</td>
      <td>0.089243</td>
      <td>25.857934</td>
      <td>0.124873</td>
      <td>0.020094</td>
      <td>0.076031</td>
      <td>0.198988</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.077602e-01</td>
      <td>0.062990</td>
      <td>0.253901</td>
      <td>1.387474</td>
      <td>6.204285</td>
    </tr>
    <tr>
      <th>Gh</th>
      <td>0.880139</td>
      <td>1.288597</td>
      <td>57.239481</td>
      <td>1.038821</td>
      <td>12.024057</td>
      <td>0.005396</td>
      <td>0.003191</td>
      <td>0.266101</td>
      <td>0.197255</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>4.479904e-01</td>
      <td>0.020024</td>
      <td>1.047751</td>
      <td>0.683810</td>
      <td>1.457041</td>
    </tr>
    <tr>
      <th>Myt1l</th>
      <td>0.054103</td>
      <td>0.013504</td>
      <td>0.034542</td>
      <td>0.143774</td>
      <td>12.077064</td>
      <td>7.309314</td>
      <td>0.151786</td>
      <td>0.017882</td>
      <td>0.141830</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>2.666798e-01</td>
      <td>0.516206</td>
      <td>0.053576</td>
      <td>2.521730</td>
      <td>3.581158</td>
    </tr>
    <tr>
      <th>Itih3</th>
      <td>0.079256</td>
      <td>2.078945</td>
      <td>1.654160</td>
      <td>0.344545</td>
      <td>22.307307</td>
      <td>0.087814</td>
      <td>0.227253</td>
      <td>1.222773</td>
      <td>0.332370</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>4.717263e-01</td>
      <td>0.934926</td>
      <td>4.441294</td>
      <td>0.470476</td>
      <td>3.252815</td>
    </tr>
    <tr>
      <th>Ttc3</th>
      <td>0.395891</td>
      <td>0.127191</td>
      <td>0.118017</td>
      <td>0.088061</td>
      <td>30.389360</td>
      <td>0.440643</td>
      <td>0.182404</td>
      <td>0.224568</td>
      <td>0.247037</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>3.362706e-01</td>
      <td>0.752675</td>
      <td>0.948286</td>
      <td>0.914819</td>
      <td>9.935296</td>
    </tr>
    <tr>
      <th>Gap43</th>
      <td>0.108115</td>
      <td>0.021445</td>
      <td>0.159717</td>
      <td>0.139596</td>
      <td>9.299651</td>
      <td>3.894822</td>
      <td>0.062725</td>
      <td>0.027901</td>
      <td>0.191815</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>1.661141e-01</td>
      <td>0.182126</td>
      <td>0.077091</td>
      <td>1.335419</td>
      <td>2.924317</td>
    </tr>
    <tr>
      <th>Hsp90ab1</th>
      <td>0.029964</td>
      <td>0.128020</td>
      <td>2.302985</td>
      <td>0.094453</td>
      <td>17.550118</td>
      <td>0.036476</td>
      <td>0.015304</td>
      <td>0.198240</td>
      <td>0.196903</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>3.223215e-01</td>
      <td>0.049360</td>
      <td>0.741837</td>
      <td>1.444626</td>
      <td>5.356896</td>
    </tr>
  </tbody>
</table>
</div>
</div>
</div>
<div class="cell" data-execution_count="91">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb947"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb947-1"><a href="#cb947-1" aria-hidden="true" tabindex="-1"></a>scv.tl.velocity_confidence(velocity_data)</span>
<span id="cb947-2"><a href="#cb947-2" aria-hidden="true" tabindex="-1"></a>keys <span class="op">=</span> [<span class="st">"velocity_length"</span>, <span class="st">"velocity_confidence"</span>]</span>
<span id="cb947-3"><a href="#cb947-3" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(</span>
<span id="cb947-4"><a href="#cb947-4" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb947-5"><a href="#cb947-5" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>keys,</span>
<span id="cb947-6"><a href="#cb947-6" aria-hidden="true" tabindex="-1"></a>    cmap<span class="op">=</span><span class="st">"coolwarm"</span>,</span>
<span id="cb947-7"><a href="#cb947-7" aria-hidden="true" tabindex="-1"></a>    perc<span class="op">=</span>[<span class="dv">5</span>, <span class="dv">95</span>],</span>
<span id="cb947-8"><a href="#cb947-8" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"umap-velocity_confidence"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb947-9"><a href="#cb947-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb947-10"><a href="#cb947-10" aria-hidden="true" tabindex="-1"></a>scv.pl.scatter(</span>
<span id="cb947-11"><a href="#cb947-11" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb947-12"><a href="#cb947-12" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"pacmap"</span>,</span>
<span id="cb947-13"><a href="#cb947-13" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>keys,</span>
<span id="cb947-14"><a href="#cb947-14" aria-hidden="true" tabindex="-1"></a>    cmap<span class="op">=</span><span class="st">"coolwarm"</span>,</span>
<span id="cb947-15"><a href="#cb947-15" aria-hidden="true" tabindex="-1"></a>    perc<span class="op">=</span>[<span class="dv">5</span>, <span class="dv">95</span>],</span>
<span id="cb947-16"><a href="#cb947-16" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"pacmap-velocity_confidence"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb947-17"><a href="#cb947-17" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>--&gt; added 'velocity_length' (adata.obs)
--&gt; added 'velocity_confidence' (adata.obs)
--&gt; added 'velocity_confidence_transition' (adata.obs)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_umap-velocity_confidence-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-92-output-4.png" class="img-fluid"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_pacmap-velocity_confidence-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-92-output-7.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="92">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb953"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb953-1"><a href="#cb953-1" aria-hidden="true" tabindex="-1"></a>scv.tl.rank_velocity_genes(velocity_data, groupby<span class="op">=</span><span class="st">"libname"</span>, min_corr<span class="op">=</span><span class="fl">0.3</span>)</span>
<span id="cb953-2"><a href="#cb953-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb953-3"><a href="#cb953-3" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> scv.DataFrame(velocity_data.uns[<span class="st">"rank_velocity_genes"</span>][<span class="st">"names"</span>])</span>
<span id="cb953-4"><a href="#cb953-4" aria-hidden="true" tabindex="-1"></a>df.head()</span>
<span id="cb953-5"><a href="#cb953-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb953-6"><a href="#cb953-6" aria-hidden="true" tabindex="-1"></a>df.to_csv(</span>
<span id="cb953-7"><a href="#cb953-7" aria-hidden="true" tabindex="-1"></a>    <span class="ss">f"/data/</span><span class="sc">{</span>bioproject<span class="sc">}</span><span class="ss">/data/velocity-signalling/</span><span class="sc">{</span>bioproject<span class="sc">}</span><span class="ss">-rank_velocity_genes.csv"</span></span>
<span id="cb953-8"><a href="#cb953-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>ranking velocity genes</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/tools/utils.py:501: DeprecationWarning: Please use `rankdata` from the `scipy.stats` namespace, the `scipy.stats.stats` namespace is deprecated.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    finished (0:00:00) --&gt; added 
    'rank_velocity_genes', sorted scores by group ids (adata.uns) 
    'spearmans_score', spearmans correlation scores (adata.var)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/tmp/ipykernel_46/2487383092.py:3: DeprecationWarning: `scvelo.read_load.get_df` is deprecated since scVelo v0.2.4 and will be removed in a future version. Please use `scvelo.core.get_df` instead.</code></pre>
</div>
</div>
</section>
<section id="get-lineage-drivers" class="level4">
<h4 class="anchored" data-anchor-id="get-lineage-drivers">Get lineage drivers</h4>
<div class="cell" data-execution_count="93">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb958"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb958-1"><a href="#cb958-1" aria-hidden="true" tabindex="-1"></a>scv.tl.paga(</span>
<span id="cb958-2"><a href="#cb958-2" aria-hidden="true" tabindex="-1"></a>    adata<span class="op">=</span>velocity_data,</span>
<span id="cb958-3"><a href="#cb958-3" aria-hidden="true" tabindex="-1"></a>    groups<span class="op">=</span><span class="st">"k_tree"</span>,</span>
<span id="cb958-4"><a href="#cb958-4" aria-hidden="true" tabindex="-1"></a>    root_key<span class="op">=</span><span class="st">"initial_states_probs"</span>,</span>
<span id="cb958-5"><a href="#cb958-5" aria-hidden="true" tabindex="-1"></a>    end_key<span class="op">=</span><span class="st">"terminal_states_probs"</span>,</span>
<span id="cb958-6"><a href="#cb958-6" aria-hidden="true" tabindex="-1"></a>    use_time_prior<span class="op">=</span><span class="st">"latent_time"</span>,</span>
<span id="cb958-7"><a href="#cb958-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>running PAGA using priors: ['latent_time', 'initial_states_probs', 'terminal_states_probs']</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    finished (0:00:00) --&gt; added
    'paga/connectivities', connectivities adjacency (adata.uns)
    'paga/connectivities_tree', connectivities subtree (adata.uns)
    'paga/transitions_confidence', velocity transitions (adata.uns)</code></pre>
</div>
</div>
<div class="cell" data-execution_count="94">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb961"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb961-1"><a href="#cb961-1" aria-hidden="true" tabindex="-1"></a>cr.tl.lineage_drivers(velocity_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: There is only 1 lineage present. Using stationary distribution instead</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/tmp/ipykernel_46/3714307425.py:1: DeprecationWarning: `cellrank.tl.lineage_drivers` will be removed in version `2.0`. Please use the `cellrank.kernels` or `cellrank.estimators` interface instead.</code></pre>
</div>
<div class="cell-output cell-output-display" data-execution_count="94">

<div>

<table class="dataframe table table-sm table-striped">
  <thead>
    <tr>
      <th></th>
      <th>13_corr</th>
      <th>13_pval</th>
      <th>13_qval</th>
      <th>13_ci_low</th>
      <th>13_ci_high</th>
    </tr>
    <tr>
      <th>mgi_symbol</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Itih3</th>
      <td>0.399612</td>
      <td>4.587418e-38</td>
      <td>6.082916e-35</td>
      <td>0.344230</td>
      <td>0.452223</td>
    </tr>
    <tr>
      <th>Stk39</th>
      <td>0.234920</td>
      <td>2.953630e-13</td>
      <td>1.958256e-10</td>
      <td>0.173318</td>
      <td>0.294690</td>
    </tr>
    <tr>
      <th>Scrg1</th>
      <td>0.223977</td>
      <td>3.799722e-12</td>
      <td>1.679477e-09</td>
      <td>0.162093</td>
      <td>0.284107</td>
    </tr>
    <tr>
      <th>Ewsr1</th>
      <td>0.220006</td>
      <td>9.287716e-12</td>
      <td>3.078878e-09</td>
      <td>0.158023</td>
      <td>0.280263</td>
    </tr>
    <tr>
      <th>Gria4</th>
      <td>0.208338</td>
      <td>1.160741e-10</td>
      <td>3.078286e-08</td>
      <td>0.146076</td>
      <td>0.268956</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>Ncam1</th>
      <td>-0.067124</td>
      <td>4.046427e-02</td>
      <td>5.312438e-01</td>
      <td>-0.130776</td>
      <td>-0.002920</td>
    </tr>
    <tr>
      <th>Kif1b</th>
      <td>-0.068542</td>
      <td>3.640322e-02</td>
      <td>4.976358e-01</td>
      <td>-0.132176</td>
      <td>-0.004345</td>
    </tr>
    <tr>
      <th>App</th>
      <td>-0.072143</td>
      <td>2.761605e-02</td>
      <td>4.024052e-01</td>
      <td>-0.135730</td>
      <td>-0.007964</td>
    </tr>
    <tr>
      <th>Ahi1</th>
      <td>-0.072266</td>
      <td>2.734967e-02</td>
      <td>4.024052e-01</td>
      <td>-0.135852</td>
      <td>-0.008088</td>
    </tr>
    <tr>
      <th>Slc1a2</th>
      <td>-0.081855</td>
      <td>1.240194e-02</td>
      <td>2.081642e-01</td>
      <td>-0.145307</td>
      <td>-0.017732</td>
    </tr>
  </tbody>
</table>
<p>1326 rows  5 columns</p>
</div>
</div>
</div>
<div class="cell" data-execution_count="95">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb964"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb964-1"><a href="#cb964-1" aria-hidden="true" tabindex="-1"></a>model <span class="op">=</span> cr.ul.models.GAM(velocity_data)</span>
<span id="cb964-2"><a href="#cb964-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb964-3"><a href="#cb964-3" aria-hidden="true" tabindex="-1"></a>cr.pl.gene_trends(</span>
<span id="cb964-4"><a href="#cb964-4" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb964-5"><a href="#cb964-5" aria-hidden="true" tabindex="-1"></a>    model<span class="op">=</span>model,</span>
<span id="cb964-6"><a href="#cb964-6" aria-hidden="true" tabindex="-1"></a>    data_key<span class="op">=</span><span class="st">"X"</span>,</span>
<span id="cb964-7"><a href="#cb964-7" aria-hidden="true" tabindex="-1"></a>    genes<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> genes_embed <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names],</span>
<span id="cb964-8"><a href="#cb964-8" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb964-9"><a href="#cb964-9" aria-hidden="true" tabindex="-1"></a>    time_key<span class="op">=</span><span class="st">"latent_time"</span>,</span>
<span id="cb964-10"><a href="#cb964-10" aria-hidden="true" tabindex="-1"></a>    same_plot<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb964-11"><a href="#cb964-11" aria-hidden="true" tabindex="-1"></a>    hide_cells<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb964-12"><a href="#cb964-12" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">12</span>, <span class="dv">12</span>),</span>
<span id="cb964-13"><a href="#cb964-13" aria-hidden="true" tabindex="-1"></a>    n_test_points<span class="op">=</span><span class="dv">200</span>,</span>
<span id="cb964-14"><a href="#cb964-14" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-adgen-gene_trends"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb964-15"><a href="#cb964-15" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<script type="application/vnd.jupyter.widget-view+json">
{"model_id":"da2d34906d0d4d358d61ee36b88794b4","version_major":2,"version_minor":0}
</script>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-96-output-12.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="96">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb975"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb975-1"><a href="#cb975-1" aria-hidden="true" tabindex="-1"></a>cr.pl.gene_trends(</span>
<span id="cb975-2"><a href="#cb975-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb975-3"><a href="#cb975-3" aria-hidden="true" tabindex="-1"></a>    model<span class="op">=</span>model,</span>
<span id="cb975-4"><a href="#cb975-4" aria-hidden="true" tabindex="-1"></a>    data_key<span class="op">=</span><span class="st">"X"</span>,</span>
<span id="cb975-5"><a href="#cb975-5" aria-hidden="true" tabindex="-1"></a>    genes<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> npep <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names],</span>
<span id="cb975-6"><a href="#cb975-6" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb975-7"><a href="#cb975-7" aria-hidden="true" tabindex="-1"></a>    time_key<span class="op">=</span><span class="st">"latent_time"</span>,</span>
<span id="cb975-8"><a href="#cb975-8" aria-hidden="true" tabindex="-1"></a>    same_plot<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb975-9"><a href="#cb975-9" aria-hidden="true" tabindex="-1"></a>    hide_cells<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb975-10"><a href="#cb975-10" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">4</span>),</span>
<span id="cb975-11"><a href="#cb975-11" aria-hidden="true" tabindex="-1"></a>    n_test_points<span class="op">=</span><span class="dv">200</span>,</span>
<span id="cb975-12"><a href="#cb975-12" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-npep-gene_trends"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb975-13"><a href="#cb975-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<script type="application/vnd.jupyter.widget-view+json">
{"model_id":"3dbe6a9233cb4af8befc489665e5a368","version_major":2,"version_minor":0}
</script>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-97-output-3.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="97">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb977"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb977-1"><a href="#cb977-1" aria-hidden="true" tabindex="-1"></a>cr.pl.gene_trends(</span>
<span id="cb977-2"><a href="#cb977-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb977-3"><a href="#cb977-3" aria-hidden="true" tabindex="-1"></a>    model<span class="op">=</span>model,</span>
<span id="cb977-4"><a href="#cb977-4" aria-hidden="true" tabindex="-1"></a>    data_key<span class="op">=</span><span class="st">"X"</span>,</span>
<span id="cb977-5"><a href="#cb977-5" aria-hidden="true" tabindex="-1"></a>    genes<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> npr <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names],</span>
<span id="cb977-6"><a href="#cb977-6" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb977-7"><a href="#cb977-7" aria-hidden="true" tabindex="-1"></a>    time_key<span class="op">=</span><span class="st">"latent_time"</span>,</span>
<span id="cb977-8"><a href="#cb977-8" aria-hidden="true" tabindex="-1"></a>    same_plot<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb977-9"><a href="#cb977-9" aria-hidden="true" tabindex="-1"></a>    hide_cells<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb977-10"><a href="#cb977-10" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">4</span>),</span>
<span id="cb977-11"><a href="#cb977-11" aria-hidden="true" tabindex="-1"></a>    n_test_points<span class="op">=</span><span class="dv">200</span>,</span>
<span id="cb977-12"><a href="#cb977-12" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-npr-gene_trends"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb977-13"><a href="#cb977-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<script type="application/vnd.jupyter.widget-view+json">
{"model_id":"3640de3ad9e943adadf6e85da7f267b9","version_major":2,"version_minor":0}
</script>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-98-output-3.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="98">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb979"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb979-1"><a href="#cb979-1" aria-hidden="true" tabindex="-1"></a>cr.pl.gene_trends(</span>
<span id="cb979-2"><a href="#cb979-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb979-3"><a href="#cb979-3" aria-hidden="true" tabindex="-1"></a>    model<span class="op">=</span>model,</span>
<span id="cb979-4"><a href="#cb979-4" aria-hidden="true" tabindex="-1"></a>    data_key<span class="op">=</span><span class="st">"X"</span>,</span>
<span id="cb979-5"><a href="#cb979-5" aria-hidden="true" tabindex="-1"></a>    genes<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> nmr <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names],</span>
<span id="cb979-6"><a href="#cb979-6" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb979-7"><a href="#cb979-7" aria-hidden="true" tabindex="-1"></a>    time_key<span class="op">=</span><span class="st">"latent_time"</span>,</span>
<span id="cb979-8"><a href="#cb979-8" aria-hidden="true" tabindex="-1"></a>    same_plot<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb979-9"><a href="#cb979-9" aria-hidden="true" tabindex="-1"></a>    hide_cells<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb979-10"><a href="#cb979-10" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">12</span>, <span class="dv">18</span>),</span>
<span id="cb979-11"><a href="#cb979-11" aria-hidden="true" tabindex="-1"></a>    n_test_points<span class="op">=</span><span class="dv">200</span>,</span>
<span id="cb979-12"><a href="#cb979-12" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-nmr-gene_trends"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb979-13"><a href="#cb979-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<script type="application/vnd.jupyter.widget-view+json">
{"model_id":"42878bd6dc364530b3a365299f723f44","version_major":2,"version_minor":0}
</script>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-99-output-5.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="99">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb983"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb983-1"><a href="#cb983-1" aria-hidden="true" tabindex="-1"></a>cr.pl.gene_trends(</span>
<span id="cb983-2"><a href="#cb983-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb983-3"><a href="#cb983-3" aria-hidden="true" tabindex="-1"></a>    model<span class="op">=</span>model,</span>
<span id="cb983-4"><a href="#cb983-4" aria-hidden="true" tabindex="-1"></a>    data_key<span class="op">=</span><span class="st">"X"</span>,</span>
<span id="cb983-5"><a href="#cb983-5" aria-hidden="true" tabindex="-1"></a>    genes<span class="op">=</span>[g <span class="cf">for</span> g <span class="kw">in</span> mitochondrial <span class="cf">if</span> g <span class="kw">in</span> velocity_data.var_names],</span>
<span id="cb983-6"><a href="#cb983-6" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb983-7"><a href="#cb983-7" aria-hidden="true" tabindex="-1"></a>    time_key<span class="op">=</span><span class="st">"latent_time"</span>,</span>
<span id="cb983-8"><a href="#cb983-8" aria-hidden="true" tabindex="-1"></a>    same_plot<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb983-9"><a href="#cb983-9" aria-hidden="true" tabindex="-1"></a>    hide_cells<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb983-10"><a href="#cb983-10" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">12</span>, <span class="dv">18</span>),</span>
<span id="cb983-11"><a href="#cb983-11" aria-hidden="true" tabindex="-1"></a>    n_test_points<span class="op">=</span><span class="dv">200</span>,</span>
<span id="cb983-12"><a href="#cb983-12" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-mitochondrial-gene_trends"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb983-13"><a href="#cb983-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<script type="application/vnd.jupyter.widget-view+json">
{"model_id":"1468e1593493481bb34907d7c1240af6","version_major":2,"version_minor":0}
</script>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:649: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/opt/python/3.8.8/lib/python3.8/site-packages/pygam/utils.py:650: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-100-output-4.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="100">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb986"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb986-1"><a href="#cb986-1" aria-hidden="true" tabindex="-1"></a>cr.pl.lineage_drivers(</span>
<span id="cb986-2"><a href="#cb986-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb986-3"><a href="#cb986-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"umap"</span>,</span>
<span id="cb986-4"><a href="#cb986-4" aria-hidden="true" tabindex="-1"></a>    lineage<span class="op">=</span><span class="dv">13</span>,</span>
<span id="cb986-5"><a href="#cb986-5" aria-hidden="true" tabindex="-1"></a>    n_genes<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb986-6"><a href="#cb986-6" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"umap-lineage-drivers"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb986-7"><a href="#cb986-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-101-output-4.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="101">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb990"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb990-1"><a href="#cb990-1" aria-hidden="true" tabindex="-1"></a>cr.pl.lineage_drivers(</span>
<span id="cb990-2"><a href="#cb990-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb990-3"><a href="#cb990-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"pacmap"</span>,</span>
<span id="cb990-4"><a href="#cb990-4" aria-hidden="true" tabindex="-1"></a>    lineage<span class="op">=</span><span class="dv">13</span>,</span>
<span id="cb990-5"><a href="#cb990-5" aria-hidden="true" tabindex="-1"></a>    n_genes<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb990-6"><a href="#cb990-6" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"pacmap-lineage-drivers"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb990-7"><a href="#cb990-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.
/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-102-output-5.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="102">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb995"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb995-1"><a href="#cb995-1" aria-hidden="true" tabindex="-1"></a>cr.pl.initial_states(</span>
<span id="cb995-2"><a href="#cb995-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb995-3"><a href="#cb995-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"umap"</span>,</span>
<span id="cb995-4"><a href="#cb995-4" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"umap-initial-states"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb995-5"><a href="#cb995-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb995-6"><a href="#cb995-6" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_umap-initial-states-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-103-output-3.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="103">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb998"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb998-1"><a href="#cb998-1" aria-hidden="true" tabindex="-1"></a>cr.pl.initial_states(</span>
<span id="cb998-2"><a href="#cb998-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb998-3"><a href="#cb998-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"pacmap"</span>,</span>
<span id="cb998-4"><a href="#cb998-4" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"pacmap-initial-states"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb998-5"><a href="#cb998-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb998-6"><a href="#cb998-6" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_pacmap-initial-states-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-104-output-3.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="104">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1001"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1001-1"><a href="#cb1001-1" aria-hidden="true" tabindex="-1"></a>cr.pl.terminal_states(</span>
<span id="cb1001-2"><a href="#cb1001-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb1001-3"><a href="#cb1001-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"pacmap"</span>,</span>
<span id="cb1001-4"><a href="#cb1001-4" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"pacmap-terminal-states"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb1001-5"><a href="#cb1001-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1001-6"><a href="#cb1001-6" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_pacmap-terminal-states-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-105-output-3.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="105">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1004"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1004-1"><a href="#cb1004-1" aria-hidden="true" tabindex="-1"></a>cr.pl.terminal_states(</span>
<span id="cb1004-2"><a href="#cb1004-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb1004-3"><a href="#cb1004-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"umap"</span>,</span>
<span id="cb1004-4"><a href="#cb1004-4" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"umap-terminal-states"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb1004-5"><a href="#cb1004-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1004-6"><a href="#cb1004-6" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_umap-terminal-states-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-106-output-3.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="106">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1007"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1007-1"><a href="#cb1007-1" aria-hidden="true" tabindex="-1"></a>cr.pl.lineages(</span>
<span id="cb1007-2"><a href="#cb1007-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb1007-3"><a href="#cb1007-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"umap"</span>,</span>
<span id="cb1007-4"><a href="#cb1007-4" aria-hidden="true" tabindex="-1"></a>    same_plot<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb1007-5"><a href="#cb1007-5" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"umap-lineages"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb1007-6"><a href="#cb1007-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1007-7"><a href="#cb1007-7" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_umap-lineages-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-107-output-3.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="107">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1010"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1010-1"><a href="#cb1010-1" aria-hidden="true" tabindex="-1"></a>cr.pl.lineages(</span>
<span id="cb1010-2"><a href="#cb1010-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb1010-3"><a href="#cb1010-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"pacmap"</span>,</span>
<span id="cb1010-4"><a href="#cb1010-4" aria-hidden="true" tabindex="-1"></a>    same_plot<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb1010-5"><a href="#cb1010-5" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"pacmap-lineages"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb1010-6"><a href="#cb1010-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1010-7"><a href="#cb1010-7" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_pacmap-lineages-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-108-output-3.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="108">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1013"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1013-1"><a href="#cb1013-1" aria-hidden="true" tabindex="-1"></a>cr.pl.lineages(</span>
<span id="cb1013-2"><a href="#cb1013-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb1013-3"><a href="#cb1013-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"umap"</span>,</span>
<span id="cb1013-4"><a href="#cb1013-4" aria-hidden="true" tabindex="-1"></a>    same_plot<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb1013-5"><a href="#cb1013-5" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"umap-lineages-same"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb1013-6"><a href="#cb1013-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1013-7"><a href="#cb1013-7" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_umap-lineages-same-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-109-output-3.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="109">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1016"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1016-1"><a href="#cb1016-1" aria-hidden="true" tabindex="-1"></a>cr.pl.lineages(</span>
<span id="cb1016-2"><a href="#cb1016-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb1016-3"><a href="#cb1016-3" aria-hidden="true" tabindex="-1"></a>    basis<span class="op">=</span><span class="st">"pacmap"</span>,</span>
<span id="cb1016-4"><a href="#cb1016-4" aria-hidden="true" tabindex="-1"></a>    same_plot<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb1016-5"><a href="#cb1016-5" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"pacmap-lineages-same"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb1016-6"><a href="#cb1016-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1016-7"><a href="#cb1016-7" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>/opt/python/3.8.8/lib/python3.8/site-packages/scvelo/plotting/utils.py:869: MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>saving figure to file ./figures/scvelo_pacmap-lineages-same-PRJNA604055-astrocytes_datasets-velocity.pdf</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-110-output-3.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="110">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1019"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1019-1"><a href="#cb1019-1" aria-hidden="true" tabindex="-1"></a>cr.pl.heatmap(</span>
<span id="cb1019-2"><a href="#cb1019-2" aria-hidden="true" tabindex="-1"></a>    velocity_data,</span>
<span id="cb1019-3"><a href="#cb1019-3" aria-hidden="true" tabindex="-1"></a>    model,</span>
<span id="cb1019-4"><a href="#cb1019-4" aria-hidden="true" tabindex="-1"></a>    genes<span class="op">=</span>velocity_data.varm[<span class="st">"terminal_lineage_drivers"</span>][<span class="st">"13_corr"</span>]</span>
<span id="cb1019-5"><a href="#cb1019-5" aria-hidden="true" tabindex="-1"></a>    .sort_values(ascending<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb1019-6"><a href="#cb1019-6" aria-hidden="true" tabindex="-1"></a>    .index[:<span class="dv">100</span>],</span>
<span id="cb1019-7"><a href="#cb1019-7" aria-hidden="true" tabindex="-1"></a>    show_absorption_probabilities<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb1019-8"><a href="#cb1019-8" aria-hidden="true" tabindex="-1"></a>    lineages<span class="op">=</span>[<span class="st">"13"</span>],</span>
<span id="cb1019-9"><a href="#cb1019-9" aria-hidden="true" tabindex="-1"></a>    n_jobs<span class="op">=</span>threads,</span>
<span id="cb1019-10"><a href="#cb1019-10" aria-hidden="true" tabindex="-1"></a>    backend<span class="op">=</span><span class="st">"loky"</span>,</span>
<span id="cb1019-11"><a href="#cb1019-11" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"-terminal_0-heatmap"</span> <span class="op">+</span> save_PRJNA604055,</span>
<span id="cb1019-12"><a href="#cb1019-12" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display">
<script type="application/vnd.jupyter.widget-view+json">
{"model_id":"09e4213f46cd4747a3318a14cb146f86","version_major":2,"version_minor":0}
</script>
</div>
<div class="cell-output cell-output-display">
<p><img src="02-velocity-signalling_hfd_files/figure-html/cell-111-output-2.png" class="img-fluid"></p>
</div>
</div>
<div class="cell" data-execution_count="111">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1020"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1020-1"><a href="#cb1020-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(watermark())</span>
<span id="cb1020-2"><a href="#cb1020-2" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(watermark(iversions<span class="op">=</span><span class="va">True</span>, globals_<span class="op">=</span><span class="bu">globals</span>()))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Last updated: 2023-07-22T11:37:29.865792+00:00

Python implementation: CPython
Python version       : 3.8.8
IPython version      : 8.12.1

Compiler    : GCC 9.3.0
OS          : Linux
Release     : 5.15.0-76-generic
Machine     : x86_64
Processor   : x86_64
CPU cores   : 256
Architecture: 64bit

scvelo    : 0.2.5
umap      : 0.5.3
numpy     : 1.23.5
pacmap    : 0.7.0
pandas    : 1.5.3
anndata   : 0.9.1
scanpy    : 1.9.3
liana     : 0.1.8
sklearn   : 1.2.2
matplotlib: 3.7.1
cellrank  : 1.5.1
scipy     : 1.10.1
plotnine  : 0.10.1
</code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script type="application/vnd.jupyter.widget-state+json">
{"state":{"00a5478cfe1b4272ad468f7a8d1c34d0":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"ProgressStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"ProgressStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","bar_color":null,"description_width":""}},"00d4e590aad0406abaebfa4791ec96a4":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"01c1be6e6e7f4aef8e6737e667ef48d3":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_e8a24da5a655442d92a466eb3428073b","placeholder":"","style":"IPY_MODEL_3bc094335ddd4952b56e45526005be01","tabbable":null,"tooltip":null,"value":"100%"}},"03f259a97902428ca422798df141c47b":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"04d44152c5e041a6b3e29f2b69e8d47b":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_a790ffb527b345d2bcd3c0b0c6014f55","placeholder":"","style":"IPY_MODEL_5b2b797b21c140ee9c39ed2bf146e63f","tabbable":null,"tooltip":null,"value":"100%"}},"09e4213f46cd4747a3318a14cb146f86":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HBoxModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HBoxModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HBoxView","box_style":"","children":["IPY_MODEL_1212decf8e5644eab9461dc2d52208a0","IPY_MODEL_84f7306367d3459aa65cd6f98dbd03e3","IPY_MODEL_4df157a6dd274fcb9c40659caef66b41"],"layout":"IPY_MODEL_de2019cc45b6468ca8608cd4d273c777","tabbable":null,"tooltip":null}},"0b5bcc560caf4a44bab2aba1a9b4e4f9":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"0ca9b7daa7844250ae075047bb534cb1":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"0dffd5cbab1749d29861704878910367":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_6699196c6ec54119b0afa965be98ad1e","placeholder":"","style":"IPY_MODEL_03f259a97902428ca422798df141c47b","tabbable":null,"tooltip":null,"value":"100%"}},"0e729df1329e4afa8eae0b8296ca8024":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"0ea59a6c3443433dab34ffedec115b05":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"0f7b524f3fed42d3b21061c68316c7f2":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_2f65989e2840424eacd71c75e3e6252d","placeholder":"","style":"IPY_MODEL_4eac8803c14a446f8e1d0050f7fa12a5","tabbable":null,"tooltip":null,"value":" 14/14 [00:00&lt;00:00, 32.56gene/s]"}},"1203690b02e348edaeb34104ab8c51be":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"FloatProgressModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"FloatProgressModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"ProgressView","bar_style":"success","description":"","description_allow_html":false,"layout":"IPY_MODEL_c99ba60d9ad144c5ae2ee8f23b285b4d","max":3,"min":0,"orientation":"horizontal","style":"IPY_MODEL_6c581c2660274f99a252db6dab4166bf","tabbable":null,"tooltip":null,"value":3}},"1212decf8e5644eab9461dc2d52208a0":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_6a01330b98bf484d8e2233a456441f55","placeholder":"","style":"IPY_MODEL_f697c315391c4609a5099e9ab34d9688","tabbable":null,"tooltip":null,"value":"100%"}},"1468e1593493481bb34907d7c1240af6":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HBoxModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HBoxModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HBoxView","box_style":"","children":["IPY_MODEL_01c1be6e6e7f4aef8e6737e667ef48d3","IPY_MODEL_59b7221ef3354de29f32dc31ce612229","IPY_MODEL_0f7b524f3fed42d3b21061c68316c7f2"],"layout":"IPY_MODEL_0ca9b7daa7844250ae075047bb534cb1","tabbable":null,"tooltip":null}},"17d8a3c6826c422fbd2abb23bd02fd70":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"19675504ac264f1ca518a33dca2a3a0f":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"1b472f67817e4437ae1b33015ee168e2":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"1d15b10626484d9e9efb012b13afe832":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_334c160c98ef4ab9ac7752834a6b5510","placeholder":"","style":"IPY_MODEL_5ba01ad8df8c442bbc6c813b5a46fcdf","tabbable":null,"tooltip":null,"value":"100%"}},"1f710c779e1046f8a41fe95a846d0f4e":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"2176be109fae4b90ac40d9b266280689":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"22969b49420241c0bbc0801b900c0a32":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"23b00ae6b403437caa096279c4a71491":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"253a7bb41e734057a956b1a71aca4e57":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"29ea5d0eab1b4e8dbad9050fcc8d7090":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"2c6890d2a85946a4919d3ad677c7aeae":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"2d2b3c94c91e42e78a183f2224d8d453":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"2f65989e2840424eacd71c75e3e6252d":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"30b3ec3d1cd848fb9db7c051df6e2c3e":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"3137689692dd4c54b7efdabd001a5270":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"334c160c98ef4ab9ac7752834a6b5510":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"335d7bcdd5704083b2ae1f869e320352":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"3640de3ad9e943adadf6e85da7f267b9":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HBoxModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HBoxModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HBoxView","box_style":"","children":["IPY_MODEL_1d15b10626484d9e9efb012b13afe832","IPY_MODEL_1203690b02e348edaeb34104ab8c51be","IPY_MODEL_8980e35cae07486d8f2cdfe609d31c20"],"layout":"IPY_MODEL_29ea5d0eab1b4e8dbad9050fcc8d7090","tabbable":null,"tooltip":null}},"3a135c30cf4646e598972ad62b4f4c0d":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HBoxModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HBoxModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HBoxView","box_style":"","children":["IPY_MODEL_9274c42d16e14c6a890da2543c83fc42","IPY_MODEL_ebd1bb44392043f0934877a7986a9a79","IPY_MODEL_47652789c69647f88c8a7e441213a1d1"],"layout":"IPY_MODEL_d560525f32994a0f8d934697797708c1","tabbable":null,"tooltip":null}},"3a464c732313434c9c64c17533430c57":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"ProgressStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"ProgressStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","bar_color":null,"description_width":""}},"3bc094335ddd4952b56e45526005be01":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"3c045df9d05c4322b73eb6b6145df953":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"3dbe6a9233cb4af8befc489665e5a368":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HBoxModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HBoxModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HBoxView","box_style":"","children":["IPY_MODEL_4ee572f069c54781b15939b524fcc22b","IPY_MODEL_b5b22d89dbfb4a87bdb60b03d646953c","IPY_MODEL_7aed94c82f834701aacd750729bebb44"],"layout":"IPY_MODEL_bfccaf13fdf741dea2fa96111d605fcf","tabbable":null,"tooltip":null}},"3f1a1a87252a43358ae2bd6c77aa31ae":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"42878bd6dc364530b3a365299f723f44":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HBoxModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HBoxModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HBoxView","box_style":"","children":["IPY_MODEL_7e39af36e36f499ea212dc0ec7e06b49","IPY_MODEL_448d18a03ea14cbe86fbc1a2cb2fdc7a","IPY_MODEL_6fe2f0e95ef740d9902eb6a66adc84ea"],"layout":"IPY_MODEL_62649691cd374ab09a25f44e7a9fa777","tabbable":null,"tooltip":null}},"428fe09f10a84b5a978b7b8e0453b906":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"445a3718f7cb4b64b0680fea2eb19ba1":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_93c9e91a64de4d2ea5e3cd966650f594","placeholder":"","style":"IPY_MODEL_30b3ec3d1cd848fb9db7c051df6e2c3e","tabbable":null,"tooltip":null,"value":"100%"}},"447963c8801445a996e48d2d5eb03d72":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"448d18a03ea14cbe86fbc1a2cb2fdc7a":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"FloatProgressModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"FloatProgressModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"ProgressView","bar_style":"success","description":"","description_allow_html":false,"layout":"IPY_MODEL_2c6890d2a85946a4919d3ad677c7aeae","max":22,"min":0,"orientation":"horizontal","style":"IPY_MODEL_de0a151370fa4c05a9c5f722f7880940","tabbable":null,"tooltip":null,"value":22}},"451f0cd4d0cb41b195b1c302b810b1c0":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_df154a702e874eada63812fbc76c40d5","placeholder":"","style":"IPY_MODEL_253a7bb41e734057a956b1a71aca4e57","tabbable":null,"tooltip":null,"value":" 932/932 [00:00&lt;00:00, 2075.20cells/s]"}},"47652789c69647f88c8a7e441213a1d1":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_b205c1a477c2418bb50b9019240e5041","placeholder":"","style":"IPY_MODEL_8d0764460d0b479083488d7bc43718cd","tabbable":null,"tooltip":null,"value":" 932/932 [00:02&lt;00:00, 372.90cell/s]"}},"4df157a6dd274fcb9c40659caef66b41":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_bb7db0e540c2439cb21c3274582b5e35","placeholder":"","style":"IPY_MODEL_6a3b0b3b69d44ca2a96b284956abf798","tabbable":null,"tooltip":null,"value":" 100/100 [00:06&lt;00:00, 40.45gene/s]"}},"4eac8803c14a446f8e1d0050f7fa12a5":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"4ee572f069c54781b15939b524fcc22b":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_c3435779ea7244bfb804871d00d2a455","placeholder":"","style":"IPY_MODEL_23b00ae6b403437caa096279c4a71491","tabbable":null,"tooltip":null,"value":"100%"}},"532a8fa82f094981b9b9e8d0e1835556":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_19675504ac264f1ca518a33dca2a3a0f","placeholder":"","style":"IPY_MODEL_c163222be987476e937eb0e00f746bb2","tabbable":null,"tooltip":null,"value":"100%"}},"59b7221ef3354de29f32dc31ce612229":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"FloatProgressModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"FloatProgressModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"ProgressView","bar_style":"success","description":"","description_allow_html":false,"layout":"IPY_MODEL_bfea5ccf418a44098d7bbc4be6f99aec","max":14,"min":0,"orientation":"horizontal","style":"IPY_MODEL_8903ffcc76f841a79e70b657c457d29b","tabbable":null,"tooltip":null,"value":14}},"5b2b797b21c140ee9c39ed2bf146e63f":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"5ba01ad8df8c442bbc6c813b5a46fcdf":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"5cf43f14bce54546a395790fc0935549":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HBoxModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HBoxModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HBoxView","box_style":"","children":["IPY_MODEL_445a3718f7cb4b64b0680fea2eb19ba1","IPY_MODEL_b6edb8f0ae664ca6a990f588e79ef283","IPY_MODEL_73045067a51347c2b2c8254074e18606"],"layout":"IPY_MODEL_c22b8cbbac1248b799b60a837fbb7640","tabbable":null,"tooltip":null}},"62649691cd374ab09a25f44e7a9fa777":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"646241f894ac40419ce474241126e49b":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"FloatProgressModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"FloatProgressModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"ProgressView","bar_style":"success","description":"","description_allow_html":false,"layout":"IPY_MODEL_de68a0c14a0a4269a5db2eb2cfac9c53","max":932,"min":0,"orientation":"horizontal","style":"IPY_MODEL_8fc638bb55054b91af1de47a6e60e44b","tabbable":null,"tooltip":null,"value":932}},"6699196c6ec54119b0afa965be98ad1e":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"6880a2525a844d82bb96e800c4ba93cb":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"6a01330b98bf484d8e2233a456441f55":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"6a3b0b3b69d44ca2a96b284956abf798":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"6c581c2660274f99a252db6dab4166bf":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"ProgressStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"ProgressStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","bar_color":null,"description_width":""}},"6fe2f0e95ef740d9902eb6a66adc84ea":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_1f710c779e1046f8a41fe95a846d0f4e","placeholder":"","style":"IPY_MODEL_8850ba8775dc47849467476079b548b8","tabbable":null,"tooltip":null,"value":" 22/22 [00:00&lt;00:00, 46.29gene/s]"}},"73045067a51347c2b2c8254074e18606":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_335d7bcdd5704083b2ae1f869e320352","placeholder":"","style":"IPY_MODEL_f243d8311f524318baeb1f35da0853d6","tabbable":null,"tooltip":null,"value":" 1/1 [00:00&lt;00:00, 10.06/s]"}},"783cf01bfcec4d8883665db28c0dba9f":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"7aed94c82f834701aacd750729bebb44":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_f8a02b40cd1e4be2973515819011862a","placeholder":"","style":"IPY_MODEL_22969b49420241c0bbc0801b900c0a32","tabbable":null,"tooltip":null,"value":" 1/1 [00:00&lt;00:00,  7.67gene/s]"}},"7b6df7d0774440499da4615bc4549b96":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_783cf01bfcec4d8883665db28c0dba9f","placeholder":"","style":"IPY_MODEL_a30ebda966954ef9bd61b3db4c0208d1","tabbable":null,"tooltip":null,"value":" 47/47 [00:01&lt;00:00, 52.69gene/s]"}},"7d0ee262b3a64919ad4c5c6c48fdac4f":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"FloatProgressModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"FloatProgressModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"ProgressView","bar_style":"success","description":"","description_allow_html":false,"layout":"IPY_MODEL_2176be109fae4b90ac40d9b266280689","max":22,"min":0,"orientation":"horizontal","style":"IPY_MODEL_de715ebedd22404da7730002b28ce22f","tabbable":null,"tooltip":null,"value":22}},"7dbdd15a627041c488fb22fb8da09529":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"7e39af36e36f499ea212dc0ec7e06b49":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_0e729df1329e4afa8eae0b8296ca8024","placeholder":"","style":"IPY_MODEL_94fcaf29ece546a386d555fece9c9592","tabbable":null,"tooltip":null,"value":"100%"}},"7faed283dd8e49cf8cc330258347300a":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"80d067b5171f4e2eb48d8555bdb059d0":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"ProgressStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"ProgressStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","bar_color":null,"description_width":""}},"84f7306367d3459aa65cd6f98dbd03e3":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"FloatProgressModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"FloatProgressModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"ProgressView","bar_style":"success","description":"","description_allow_html":false,"layout":"IPY_MODEL_e2c62a8abbd5421ba389c951374be324","max":100,"min":0,"orientation":"horizontal","style":"IPY_MODEL_80d067b5171f4e2eb48d8555bdb059d0","tabbable":null,"tooltip":null,"value":100}},"8816bdc942a84638b1d8d4397dc5b84c":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_abba31073aa845b99c31382affb4ccaf","placeholder":"","style":"IPY_MODEL_eba7c3e1318448cd81db6edebf278c6c","tabbable":null,"tooltip":null,"value":"100%"}},"8850ba8775dc47849467476079b548b8":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"8903ffcc76f841a79e70b657c457d29b":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"ProgressStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"ProgressStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","bar_color":null,"description_width":""}},"8980e35cae07486d8f2cdfe609d31c20":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_0ea59a6c3443433dab34ffedec115b05","placeholder":"","style":"IPY_MODEL_3137689692dd4c54b7efdabd001a5270","tabbable":null,"tooltip":null,"value":" 3/3 [00:00&lt;00:00,  7.14gene/s]"}},"8a02abe21af64dadaef7d359f40c9cf0":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"8d0764460d0b479083488d7bc43718cd":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"8fc638bb55054b91af1de47a6e60e44b":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"ProgressStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"ProgressStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","bar_color":null,"description_width":""}},"905acb3ffbdb4f6190cfe472223c1384":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HBoxModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HBoxModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HBoxView","box_style":"","children":["IPY_MODEL_8816bdc942a84638b1d8d4397dc5b84c","IPY_MODEL_e66b2b8f1c0d4bfa930711fc8fd1c774","IPY_MODEL_eb64def955c4445ab34dde9b213e4cce"],"layout":"IPY_MODEL_b91be9d855e044219c1a6f59f349fc06","tabbable":null,"tooltip":null}},"9274c42d16e14c6a890da2543c83fc42":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_c525024285e841e389780e1ae5b034fd","placeholder":"","style":"IPY_MODEL_17d8a3c6826c422fbd2abb23bd02fd70","tabbable":null,"tooltip":null,"value":"100%"}},"93c9e91a64de4d2ea5e3cd966650f594":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"94d78b8a0f164637984ffd87a4204ac1":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"94fcaf29ece546a386d555fece9c9592":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"986c7b9304b6456abc1ca86ee7610150":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"9bcebd18addc44e08983706f7f866024":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_447963c8801445a996e48d2d5eb03d72","placeholder":"","style":"IPY_MODEL_1b472f67817e4437ae1b33015ee168e2","tabbable":null,"tooltip":null,"value":" 932/932 [00:00&lt;00:00, 2166.06cell/s]"}},"a30ebda966954ef9bd61b3db4c0208d1":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"a331ad933cef48699e20d44e20b24269":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"a790ffb527b345d2bcd3c0b0c6014f55":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"aa3e79591a324074a2ea32213f45b842":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"ProgressStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"ProgressStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","bar_color":null,"description_width":""}},"abba31073aa845b99c31382affb4ccaf":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"b205c1a477c2418bb50b9019240e5041":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"b5b22d89dbfb4a87bdb60b03d646953c":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"FloatProgressModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"FloatProgressModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"ProgressView","bar_style":"success","description":"","description_allow_html":false,"layout":"IPY_MODEL_8a02abe21af64dadaef7d359f40c9cf0","max":1,"min":0,"orientation":"horizontal","style":"IPY_MODEL_00a5478cfe1b4272ad468f7a8d1c34d0","tabbable":null,"tooltip":null,"value":1}},"b6edb8f0ae664ca6a990f588e79ef283":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"FloatProgressModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"FloatProgressModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"ProgressView","bar_style":"success","description":"","description_allow_html":false,"layout":"IPY_MODEL_a331ad933cef48699e20d44e20b24269","max":1,"min":0,"orientation":"horizontal","style":"IPY_MODEL_f98fde330ec14729ac1a0480f6200438","tabbable":null,"tooltip":null,"value":1}},"b91be9d855e044219c1a6f59f349fc06":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"bb7db0e540c2439cb21c3274582b5e35":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"bfccaf13fdf741dea2fa96111d605fcf":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"bfea5ccf418a44098d7bbc4be6f99aec":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"c163222be987476e937eb0e00f746bb2":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"c22b8cbbac1248b799b60a837fbb7640":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"c2a8f630197346ad84cc948a0d01d6b4":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"FloatProgressModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"FloatProgressModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"ProgressView","bar_style":"success","description":"","description_allow_html":false,"layout":"IPY_MODEL_cb17aa11ebf048a6a3194015e2df29a0","max":47,"min":0,"orientation":"horizontal","style":"IPY_MODEL_3a464c732313434c9c64c17533430c57","tabbable":null,"tooltip":null,"value":47}},"c3435779ea7244bfb804871d00d2a455":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"c525024285e841e389780e1ae5b034fd":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"c99ba60d9ad144c5ae2ee8f23b285b4d":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"ca8cdcccf6bd408ba971408deb20db72":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"cb17aa11ebf048a6a3194015e2df29a0":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"d0e5e1f50bbe43b282f54f2a18c61104":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HBoxModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HBoxModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HBoxView","box_style":"","children":["IPY_MODEL_d21e1ba7a9f84610ba2533e1fb0843cd","IPY_MODEL_d29a4fd48c074a62adecd841ed8a5289","IPY_MODEL_9bcebd18addc44e08983706f7f866024"],"layout":"IPY_MODEL_6880a2525a844d82bb96e800c4ba93cb","tabbable":null,"tooltip":null}},"d21e1ba7a9f84610ba2533e1fb0843cd":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_7dbdd15a627041c488fb22fb8da09529","placeholder":"","style":"IPY_MODEL_428fe09f10a84b5a978b7b8e0453b906","tabbable":null,"tooltip":null,"value":"100%"}},"d29a4fd48c074a62adecd841ed8a5289":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"FloatProgressModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"FloatProgressModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"ProgressView","bar_style":"success","description":"","description_allow_html":false,"layout":"IPY_MODEL_0b5bcc560caf4a44bab2aba1a9b4e4f9","max":932,"min":0,"orientation":"horizontal","style":"IPY_MODEL_aa3e79591a324074a2ea32213f45b842","tabbable":null,"tooltip":null,"value":932}},"d29e85919485415eb38ef76f22b94cb7":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_986c7b9304b6456abc1ca86ee7610150","placeholder":"","style":"IPY_MODEL_00d4e590aad0406abaebfa4791ec96a4","tabbable":null,"tooltip":null,"value":" 22/22 [00:10&lt;00:00,  4.21gene/s]"}},"d31b598d82994f42bda505f47fd5b247":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HBoxModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HBoxModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HBoxView","box_style":"","children":["IPY_MODEL_532a8fa82f094981b9b9e8d0e1835556","IPY_MODEL_646241f894ac40419ce474241126e49b","IPY_MODEL_451f0cd4d0cb41b195b1c302b810b1c0"],"layout":"IPY_MODEL_3c045df9d05c4322b73eb6b6145df953","tabbable":null,"tooltip":null}},"d35b970892e546a1abf30a1c8c651ea1":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HBoxModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HBoxModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HBoxView","box_style":"","children":["IPY_MODEL_04d44152c5e041a6b3e29f2b69e8d47b","IPY_MODEL_7d0ee262b3a64919ad4c5c6c48fdac4f","IPY_MODEL_d29e85919485415eb38ef76f22b94cb7"],"layout":"IPY_MODEL_94d78b8a0f164637984ffd87a4204ac1","tabbable":null,"tooltip":null}},"d560525f32994a0f8d934697797708c1":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"da2d34906d0d4d358d61ee36b88794b4":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HBoxModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HBoxModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HBoxView","box_style":"","children":["IPY_MODEL_0dffd5cbab1749d29861704878910367","IPY_MODEL_c2a8f630197346ad84cc948a0d01d6b4","IPY_MODEL_7b6df7d0774440499da4615bc4549b96"],"layout":"IPY_MODEL_3f1a1a87252a43358ae2bd6c77aa31ae","tabbable":null,"tooltip":null}},"ddf76590f97e42afaef4f9ae021ad4b8":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"ProgressStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"ProgressStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","bar_color":null,"description_width":""}},"de0a151370fa4c05a9c5f722f7880940":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"ProgressStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"ProgressStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","bar_color":null,"description_width":""}},"de2019cc45b6468ca8608cd4d273c777":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"de68a0c14a0a4269a5db2eb2cfac9c53":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"de715ebedd22404da7730002b28ce22f":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"ProgressStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"ProgressStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","bar_color":null,"description_width":""}},"df154a702e874eada63812fbc76c40d5":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"e2c62a8abbd5421ba389c951374be324":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"e66b2b8f1c0d4bfa930711fc8fd1c774":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"FloatProgressModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"FloatProgressModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"ProgressView","bar_style":"success","description":"","description_allow_html":false,"layout":"IPY_MODEL_f505a063f6c64b83ada614f1586dd59f","max":932,"min":0,"orientation":"horizontal","style":"IPY_MODEL_e973b8f969384b7692692110f6567452","tabbable":null,"tooltip":null,"value":932}},"e8a24da5a655442d92a466eb3428073b":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"e973b8f969384b7692692110f6567452":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"ProgressStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"ProgressStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","bar_color":null,"description_width":""}},"eb64def955c4445ab34dde9b213e4cce":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"HTMLView","description":"","description_allow_html":false,"layout":"IPY_MODEL_2d2b3c94c91e42e78a183f2224d8d453","placeholder":"","style":"IPY_MODEL_ca8cdcccf6bd408ba971408deb20db72","tabbable":null,"tooltip":null,"value":" 932/932 [00:04&lt;00:00, 212.08cell/s]"}},"eba7c3e1318448cd81db6edebf278c6c":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"ebd1bb44392043f0934877a7986a9a79":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"FloatProgressModel","state":{"_dom_classes":[],"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"FloatProgressModel","_view_count":null,"_view_module":"@jupyter-widgets/controls","_view_module_version":"2.0.0","_view_name":"ProgressView","bar_style":"success","description":"","description_allow_html":false,"layout":"IPY_MODEL_7faed283dd8e49cf8cc330258347300a","max":932,"min":0,"orientation":"horizontal","style":"IPY_MODEL_ddf76590f97e42afaef4f9ae021ad4b8","tabbable":null,"tooltip":null,"value":932}},"f243d8311f524318baeb1f35da0853d6":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"f505a063f6c64b83ada614f1586dd59f":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"f697c315391c4609a5099e9ab34d9688":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"HTMLStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"HTMLStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","background":null,"description_width":"","font_size":null,"text_color":null}},"f8a02b40cd1e4be2973515819011862a":{"model_module":"@jupyter-widgets/base","model_module_version":"2.0.0","model_name":"LayoutModel","state":{"_model_module":"@jupyter-widgets/base","_model_module_version":"2.0.0","_model_name":"LayoutModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border_bottom":null,"border_left":null,"border_right":null,"border_top":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"f98fde330ec14729ac1a0480f6200438":{"model_module":"@jupyter-widgets/controls","model_module_version":"2.0.0","model_name":"ProgressStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"2.0.0","_model_name":"ProgressStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"2.0.0","_view_name":"StyleView","bar_color":null,"description_width":""}}},"version_major":2,"version_minor":0}
</script>
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>